diff -u -r org.glite.ce.blahp/src/classad_binary_op_unwind.C org.glite.ce.blahp.patched/src/classad_binary_op_unwind.C
--- org.glite.ce.blahp/src/classad_binary_op_unwind.C	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/classad_binary_op_unwind.C	2010-05-26 16:21:48.000000000 -0500
@@ -39,10 +39,10 @@
 
 #include <strings.h> // strcasecmp
 
-#include "classad_distribution.h"
+#include "classad/classad_distribution.h"
 #include "classad_binary_op_unwind.h"
 
-#ifdef WANT_NAMESPACES
+#if 1
 namespace classad {
 #endif
 
@@ -221,6 +221,6 @@
   return;
 }
 
-#ifdef WANT_NAMESPACES
+#if 1
 } // end of classad namespace
 #endif
diff -u -r org.glite.ce.blahp/src/classad_binary_op_unwind.h org.glite.ce.blahp.patched/src/classad_binary_op_unwind.h
--- org.glite.ce.blahp/src/classad_binary_op_unwind.h	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/classad_binary_op_unwind.h	2010-05-26 16:21:48.000000000 -0500
@@ -33,12 +33,12 @@
 #
 */
 
-#include "classad_distribution.h"
+#include "classad/classad_distribution.h"
 
 #ifndef __CLASSAD_BINARY_OP_UNWIND_H__
 #define __CLASSAD_BINARY_OP_UNWIND_H__
 
-#ifdef WANT_NAMESPACES
+#if 1
 using namespace classad;
 
 namespace classad {
@@ -64,7 +64,7 @@
         std::vector<std::string> m_unwind_output;
 };
 
-#ifdef WANT_NAMESPACES
+#if 1
 } // end of classad namespace
 #endif
 
diff -u -r org.glite.ce.blahp/src/classad_c_helper.C org.glite.ce.blahp.patched/src/classad_c_helper.C
--- org.glite.ce.blahp/src/classad_c_helper.C	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/classad_c_helper.C	2010-05-26 16:21:48.000000000 -0500
@@ -38,10 +38,10 @@
 */
 
 #include <string.h>
-#include "classad_distribution.h"
+#include "classad/classad_distribution.h"
 #include "classad_binary_op_unwind.h"
 
-#ifdef WANT_NAMESPACES
+#if 1
 using namespace classad;
 #endif
 
