--- nscd_helper.c.orig	Fri May 30 13:32:04 2008
+++ glibc-2.5-20061008T1257-x86_64/nscd/nscd_helper.c	Fri May 30 13:39:34 2008
@@ -98,7 +98,15 @@
 static int
 open_socket (void)
 {
-  int sock = __socket (PF_UNIX, SOCK_STREAM, 0);
+  int sock;
+
+  /* patched by psilord 05-29-08 to turn off connecting to the nscd via a unix
+  	domain socket which appears to be left permanently open. This open
+	socket defers checkpointing forever. */
+  errno = ENOENT;
+  return -1;
+  
+  sock = __socket (PF_UNIX, SOCK_STREAM, 0);
   if (sock < 0)
     return -1;
 
