diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp.patched/src/server.c
--- org.glite.ce.blahp/src/server.c	2008-10-17 11:11:42.000000000 -0500
+++ org.glite.ce.blahp.patched/src/server.c	2009-03-19 23:34:19.000000000 -0500
@@ -1181,7 +1181,7 @@
 		resultLine = make_message("%s 8 no\\ jobId\\ in\\ submission\\ script's\\ output\\ (stdout:%s)\\ (stderr:%s) N/A", reqId, escpd_cmd_out, escpd_cmd_err);
 		if (escpd_cmd_out != blah_omem_msg) free(escpd_cmd_out);
 		if (escpd_cmd_err != blah_omem_msg) free(escpd_cmd_err);
-		goto cleanup_cmd_out;
+		goto cleanup_regbuf;
 	}
 
 	cmd_out[pmatch[2].rm_eo] = '\000';
@@ -1196,10 +1196,11 @@
 
 	/* Free up all arguments and exit (exit point in case of error is the label
 	   pointing to last successfully allocated variable) */
+cleanup_regbuf:
+	regfree(&regbuf);
 cleanup_cmd_out:
 	free(cmd_out);
 	if (cmd_err) free(cmd_err);
-	regfree(&regbuf);
 cleanup_command:
 	free(command);
 cleanup_proxyname:
