diff -u -r gt4.2.1-all-source-installer/source-trees/xio/drivers/gsi/globus_xio_gsi.c gt4.2.1-all-source-installer.patched/source-trees/xio/drivers/gsi/globus_xio_gsi.c
--- gt4.2.1-all-source-installer/source-trees/xio/drivers/gsi/globus_xio_gsi.c	2008-09-18 15:24:06.000000000 -0500
+++ gt4.2.1-all-source-installer.patched/source-trees/xio/drivers/gsi/globus_xio_gsi.c	2009-01-20 14:40:58.000000000 -0600
@@ -1,3 +1,4 @@
+#if 0
 /*
  * Copyright 1999-2006 University of Chicago
  * 
@@ -4100,3 +4101,4 @@
     GlobusXIOGSIDebugInternalExitWithError();    
     return result;
 }
+#endif
diff -u -r gt4.2.1-all-source-installer/source-trees/xio/drivers/popen/source/globus_xio_popen_driver.c gt4.2.1-all-source-installer.patched/source-trees/xio/drivers/popen/source/globus_xio_popen_driver.c
--- gt4.2.1-all-source-installer/source-trees/xio/drivers/popen/source/globus_xio_popen_driver.c	2008-09-18 01:16:35.000000000 -0500
+++ gt4.2.1-all-source-installer.patched/source-trees/xio/drivers/popen/source/globus_xio_popen_driver.c	2009-01-20 14:41:14.000000000 -0600
@@ -1,3 +1,4 @@
+#if 0
 /*
  * Copyright 1999-2006 University of Chicago
  *
@@ -781,3 +782,4 @@
     
     return GLOBUS_SUCCESS;
 }
+#endif
