diff -ruN glibc-2.3.2.27.9.7/configure.in glibc-2.3.2.27.9.7.patched/configure.in
--- glibc-2.3.2.27.9.7/configure.in	2003-04-01 08:53:27.000000000 +0200
+++ glibc-2.3.2.27.9.7.patched/configure.in	2006-10-24 17:25:07.000000000 +0200
@@ -700,7 +700,7 @@
 # These programs are version sensitive.
 AC_CHECK_TOOL_PREFIX
 AC_CHECK_PROG_VER(CC, ${ac_tool_prefix}gcc ${ac_tool_prefix}cc, -v,
-  [version \([egcygnustpi-]*[0-9.]*\)], [3.[2-9]*],
+  [version \([egcygnustpi-]*[0-9.]*\)], [3.[2-9]* | 4.*],
   critic_missing="$critic_missing gcc")
 AC_CHECK_PROG_VER(MAKE, gnumake gmake make, --version,
   [GNU Make[^0-9]*\([0-9][0-9.]*\)],
diff -ru glibc-2.3.2.27.9.7.orig/csu/Makefile glibc-2.3.2.27.9.7/csu/Makefile
--- glibc-2.3.2.27.9.7.orig/csu/Makefile	2003-03-23 00:00:17.000000000 +0100
+++ glibc-2.3.2.27.9.7/csu/Makefile	2006-10-25 02:31:24.000000000 +0200
@@ -101,7 +101,7 @@
 $(objpfx)crt%.o: $(objpfx)crt%.S $(objpfx)defs.h
 	$(compile.S) -g0 $(ASFLAGS-.os) -o $@
 
-CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions
+CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions -fno-unit-at-a-time
 
 vpath initfini.c $(full_config_sysdirs)

diff -ru glibc-2.3.2.27.9.7.orig/linuxthreads/sysdeps/i386/Makefile glibc-2.3.2.27.9.7/linuxthreads/sysdeps/i386/Makefile
--- glibc-2.3.2.27.9.7.orig/linuxthreads/sysdeps/i386/Makefile	2002-12-28 23:06:49.000000000 +0100
+++ glibc-2.3.2.27.9.7/linuxthreads/sysdeps/i386/Makefile	2006-10-25 02:28:44.000000000 +0200
@@ -1,7 +1,7 @@
 ifeq ($(subdir),linuxthreads)
 # On i686 we must avoid generating the trampoline functions generated
 # to get the GOT pointer.
-CFLAGS-pt-initfini.s += -march=i386 -mcpu=i386
+CFLAGS-pt-initfini.s += -march=i386 -mcpu=i386 -fno-unit-at-a-time
 
 # Most files must not be compiled without frame pointer since we need
 # the frame base address which is stored in %ebp unless the frame pointer
