--- gsoap-2.7.10-p1/gsoap/Makefile.in	2008-01-27 15:02:36.000000000 -0600
+++ gsoap-2.7.10-p1-mf/gsoap/Makefile.in	2008-06-25 19:06:06.000000000 -0500
@@ -104,6 +104,9 @@ CLEANFILES = soapcpp2 stdsoap2_cpp.cpp s
 AM_CXXFLAGS = $(SOAPCPP2_DEBUG) -D$(platform)
 AM_CFLAGS = $(SOAPCPP2_DEBUG) -D$(platform)
 
+# Install all soapcpp2 and wsdl2h files into /usr/share/gsoap
+nobase_pkgdata_DATA = import/* plugin/* WS/* custom/* extras/*
+
 lib_LIBRARIES = libgsoap.a libgsoap++.a libgsoapck.a libgsoapck++.a libgsoapssl.a libgsoapssl++.a
 libgsoap_a_SOURCES = stdsoap2.c
 libgsoap_a_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform)
@@ -179,6 +182,8 @@ CXXFLAGS = @CXXFLAGS@
 DIST_SOURCES = $(libgsoap___a_SOURCES) $(libgsoap_a_SOURCES) \
 	$(libgsoapck___a_SOURCES) $(libgsoapck_a_SOURCES) \
 	$(libgsoapssl___a_SOURCES) $(libgsoapssl_a_SOURCES)
+DATA = $(nobase_pkgdata_DATA)
+
 HEADERS = $(include_HEADERS)
 
 
@@ -377,6 +382,24 @@ libgsoapssl___a-stdsoap2_ssl_cpp.obj: st
 	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgsoapssl___a_CXXFLAGS) $(CXXFLAGS) -c -o libgsoapssl___a-stdsoap2_ssl_cpp.obj `cygpath -w stdsoap2_ssl_cpp.cpp`
 CXXDEPMODE = @CXXDEPMODE@
 uninstall-info-am:
+nobase_pkgdataDATA_INSTALL = $(install_sh_DATA)
+install-nobase_pkgdataDATA: $(nobase_pkgdata_DATA)
+	@$(NORMAL_INSTALL)
+	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)
+	@list='$(nobase_pkgdata_DATA)'; for p in $$list; do \
+	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
+	  f="$$p"; \
+	  echo " $(nobase_pkgdataDATA_INSTALL) $$d$$p $(DESTDIR)$(pkgdatadir)/$$f"; \
+	  $(nobase_pkgdataDATA_INSTALL) $$d$$p $(DESTDIR)$(pkgdatadir)/$$f; \
+	done
+
+uninstall-nobase_pkgdataDATA:
+	@$(NORMAL_UNINSTALL)
+	@list='$(nobase_pkgdata_DATA)'; for p in $$list; do \
+	  f="$$p"; \
+	  echo " rm -f $(DESTDIR)$(pkgdatadir)/$$f"; \
+	  rm -f $(DESTDIR)$(pkgdatadir)/$$f; \
+	done
 includeHEADERS_INSTALL = $(INSTALL_HEADER)
 install-includeHEADERS: $(include_HEADERS)
 	@$(NORMAL_INSTALL)
@@ -532,10 +555,10 @@ distdir: $(DISTFILES)
 	done
 check-am: all-am
 check: check-recursive
-all-am: Makefile $(LIBRARIES) $(HEADERS)
+all-am: Makefile $(LIBRARIES) $(DATA) $(HEADERS)
 installdirs: installdirs-recursive
 installdirs-am:
-	$(mkinstalldirs) $(DESTDIR)$(libdir) $(DESTDIR)$(includedir)
+	$(mkinstalldirs) $(DESTDIR)$(libdir) $(DESTDIR)$(pkgdatadir) $(DESTDIR)$(includedir)
 
 install: install-recursive
 install-exec: install-exec-recursive
@@ -580,7 +603,7 @@ info: info-recursive
 
 info-am:
 
-install-data-am: install-includeHEADERS
+install-data-am: install-includeHEADERS install-nobase_pkgdataDATA
 	@$(NORMAL_INSTALL)
 	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 
@@ -601,7 +624,7 @@ mostlyclean: mostlyclean-recursive
 mostlyclean-am: mostlyclean-compile mostlyclean-generic
 
 uninstall-am: uninstall-includeHEADERS uninstall-info-am \
-	uninstall-libLIBRARIES
+	uninstall-libLIBRARIES uninstall-nobase_pkgdataDATA
 
 uninstall-info: uninstall-info-recursive
 
@@ -614,14 +637,15 @@ uninstall-info: uninstall-info-recursive
 	install-exec install-exec-am install-exec-recursive \
 	install-includeHEADERS install-info install-info-am \
 	install-info-recursive install-libLIBRARIES install-man \
-	install-recursive install-strip installcheck installcheck-am \
-	installdirs installdirs-am installdirs-recursive \
-	maintainer-clean maintainer-clean-generic \
+	install-nobase_pkgdataDATA install-recursive install-strip \
+	installcheck installcheck-am installdirs installdirs-am \
+	installdirs-recursive maintainer-clean maintainer-clean-generic \
 	maintainer-clean-recursive mostlyclean mostlyclean-compile \
 	mostlyclean-generic mostlyclean-recursive tags tags-recursive \
 	uninstall uninstall-am uninstall-includeHEADERS \
 	uninstall-info-am uninstall-info-recursive \
-	uninstall-libLIBRARIES uninstall-recursive
+	uninstall-libLIBRARIES uninstall-nobase_pkgdataDATA \
+	uninstall-recursive
 
 
 stdsoap2_ck.c: stdsoap2.cpp
