diff -u -r gsoap-2.7.10/gsoap/wsdl/Makefile.in gsoap-2.7.10-patched/gsoap/wsdl/Makefile.in
--- gsoap-2.7.10/gsoap/wsdl/Makefile.in	2008-01-27 15:02:58.000000000 -0600
+++ gsoap-2.7.10-patched/gsoap/wsdl/Makefile.in	2016-11-04 10:12:07.000000000 -0500
@@ -102,7 +102,7 @@
 AUTOMAKE_OPTIONS = foreign 1.4
 
 INCLUDES = -I$(top_srcdir)/gsoap
-LDFLAGS = $(INCLUDES) -L$(srcdir)
+LDFLAGS = @LDFLAGS@ $(INCLUDES) -L$(srcdir)
 SOAP = $(top_builddir)/gsoap/src/soapcpp2
 SOAP_CPP_LIB_SSL = $(top_builddir)/gsoap/libgsoapssl++.a
 SOAP_CPP_SRC = wsdlC.cpp
