diff -u -r org.glite.ce.blahp/config/Makefile.am org.glite.ce.blahp-patched/config/Makefile.am
--- org.glite.ce.blahp/config/Makefile.am	Wed May  3 16:09:23 2006
+++ org.glite.ce.blahp-patched/config/Makefile.am	Wed May  3 16:14:39 2006
@@ -15,12 +15,12 @@
 #
 
 wletcwmdir = $(sysconfdir)
-wletcwm_DATA = blah.config blparser.conf baparser.conf
+wletcwm_DATA = batch_gahp.config blparser.conf baparser.conf
 
 install-data-hook:
-	/bin/cp $(DESTDIR)$(sysconfdir)/blah.config $(DESTDIR)$(sysconfdir)/blah.config.template
+	/bin/cp $(DESTDIR)$(sysconfdir)/batch_gahp.config $(DESTDIR)$(sysconfdir)/batch_gahp.config.template
 
-EXTRA_DIST = blah.config \
+EXTRA_DIST = batch_gahp.config \
              blparser.conf \
              baparser.conf \
              glite-ce-blparser \
diff -u -r org.glite.ce.blahp/config/Makefile.in org.glite.ce.blahp-patched/config/Makefile.in
--- org.glite.ce.blahp/config/Makefile.in	Wed Apr  5 20:37:54 2006
+++ org.glite.ce.blahp-patched/config/Makefile.in	Wed May  3 16:16:51 2006
@@ -131,9 +131,9 @@
 install_sh = @install_sh@
 
 wletcwmdir = $(sysconfdir)
-wletcwm_DATA = blah.config blparser.conf baparser.conf
+wletcwm_DATA = batch_gahp.config blparser.conf baparser.conf
 
-EXTRA_DIST = blah.config \
+EXTRA_DIST = batch_gahp.config \
              blparser.conf \
              baparser.conf \
              glite-ce-blparser \
@@ -324,7 +324,7 @@
 
 
 install-data-hook:
-	/bin/cp $(DESTDIR)$(sysconfdir)/blah.config $(DESTDIR)$(sysconfdir)/blah.config.template
+	/bin/cp $(DESTDIR)$(sysconfdir)/batch_gahp.config $(DESTDIR)$(sysconfdir)/batch_gahp.config.template
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -u -r org.glite.ce.blahp/src/Makefile.am org.glite.ce.blahp-patched/src/Makefile.am
--- org.glite.ce.blahp/src/Makefile.am	Wed May  3 16:09:23 2006
+++ org.glite.ce.blahp-patched/src/Makefile.am	Wed May  3 16:14:53 2006
@@ -18,17 +18,17 @@
 
 SUBDIRS = scripts
 
-bin_PROGRAMS = blahpd blahpd_daemon BPRclient BPRserver BLClient BLParserLSF BLParserPBS BAParser BAClient
+bin_PROGRAMS = batch_gahp batch_gahp_daemon BPRclient BPRserver BLClient BLParserLSF BLParserPBS BAParser BAClient
 
 common_sources = console.c job_status.c resbuffer.c server.c commands.c classad_binary_op_unwind.C classad_c_helper.C mtsafe_popen.c
 
-blahpd_SOURCES = main.c $(common_sources)
+batch_gahp_SOURCES = main.c $(common_sources)
 
-blahpd_daemon_SOURCES = main_daemon.c $(common_sources)
+batch_gahp_daemon_SOURCES = main_daemon.c $(common_sources)
 
-blahpd_LDADD = $(CLASSAD_LIBS)
+batch_gahp_LDADD = $(CLASSAD_LIBS)
 
-blahpd_daemon_LDADD = $(blahpd_LDADD)
+batch_gahp_daemon_LDADD = $(batch_gahp_LDADD)
 
 BPRclient_SOURCES = BPRclient.c BPRcomm.c tokens.c
 
diff -u -r org.glite.ce.blahp/src/Makefile.in org.glite.ce.blahp-patched/src/Makefile.in
--- org.glite.ce.blahp/src/Makefile.in	Wed Apr  5 20:37:55 2006
+++ org.glite.ce.blahp-patched/src/Makefile.in	Wed May  3 16:25:34 2006
@@ -134,17 +134,17 @@
 
 SUBDIRS = scripts
 
-bin_PROGRAMS = blahpd blahpd_daemon BPRclient BPRserver BLClient BLParserLSF BLParserPBS BAParser BAClient
+bin_PROGRAMS = batch_gahp batch_gahp_daemon BPRclient BPRserver BLClient BLParserLSF BLParserPBS BAParser BAClient
 
 common_sources = console.c job_status.c resbuffer.c server.c commands.c classad_binary_op_unwind.C classad_c_helper.C mtsafe_popen.c
 
-blahpd_SOURCES = main.c $(common_sources)
+batch_gahp_SOURCES = main.c $(common_sources)
 
-blahpd_daemon_SOURCES = main_daemon.c $(common_sources)
+batch_gahp_daemon_SOURCES = main_daemon.c $(common_sources)
 
-blahpd_LDADD = $(CLASSAD_LIBS)
+batch_gahp_LDADD = $(CLASSAD_LIBS)
 
-blahpd_daemon_LDADD = $(blahpd_LDADD)
+batch_gahp_daemon_LDADD = $(batch_gahp_LDADD)
 
 BPRclient_SOURCES = BPRclient.c BPRcomm.c tokens.c
 
@@ -186,7 +186,7 @@
 subdir = src
 mkinstalldirs = $(SHELL) $(top_srcdir)/./project/mkinstalldirs
 CONFIG_CLEAN_FILES =
-bin_PROGRAMS = blahpd$(EXEEXT) blahpd_daemon$(EXEEXT) BPRclient$(EXEEXT) \
+bin_PROGRAMS = batch_gahp$(EXEEXT) batch_gahp_daemon$(EXEEXT) BPRclient$(EXEEXT) \
 	BPRserver$(EXEEXT) BLClient$(EXEEXT) BLParserLSF$(EXEEXT) \
 	BLParserPBS$(EXEEXT) BAParser$(EXEEXT) BAClient$(EXEEXT)
 PROGRAMS = $(bin_PROGRAMS)
@@ -227,14 +227,14 @@
 	resbuffer.$(OBJEXT) server.$(OBJEXT) commands.$(OBJEXT) \
 	classad_binary_op_unwind.$(OBJEXT) classad_c_helper.$(OBJEXT) \
 	mtsafe_popen.$(OBJEXT)
-am_blahpd_OBJECTS = main.$(OBJEXT) $(am__objects_1)
-blahpd_OBJECTS = $(am_blahpd_OBJECTS)
-blahpd_DEPENDENCIES =
-blahpd_LDFLAGS =
-am_blahpd_daemon_OBJECTS = main_daemon.$(OBJEXT) $(am__objects_1)
-blahpd_daemon_OBJECTS = $(am_blahpd_daemon_OBJECTS)
-blahpd_daemon_DEPENDENCIES =
-blahpd_daemon_LDFLAGS =
+am_batch_gahp_OBJECTS = main.$(OBJEXT) $(am__objects_1)
+batch_gahp_OBJECTS = $(am_batch_gahp_OBJECTS)
+batch_gahp_DEPENDENCIES =
+batch_gahp_LDFLAGS =
+am_batch_gahp_daemon_OBJECTS = main_daemon.$(OBJEXT) $(am__objects_1)
+batch_gahp_daemon_OBJECTS = $(am_batch_gahp_daemon_OBJECTS)
+batch_gahp_daemon_DEPENDENCIES =
+batch_gahp_daemon_LDFLAGS =
 
 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir)
@@ -276,7 +276,7 @@
 DIST_SOURCES = $(BAClient_SOURCES) $(BAParser_SOURCES) \
 	$(BLClient_SOURCES) $(BLParserLSF_SOURCES) \
 	$(BLParserPBS_SOURCES) $(BPRclient_SOURCES) \
-	$(BPRserver_SOURCES) $(blahpd_SOURCES) $(blahpd_daemon_SOURCES)
+	$(BPRserver_SOURCES) $(batch_gahp_SOURCES) $(batch_gahp_daemon_SOURCES)
 HEADERS = $(noinst_HEADERS)
 
 
@@ -286,7 +286,7 @@
 	uninstall-recursive check-recursive installcheck-recursive
 DIST_COMMON = $(noinst_HEADERS) Makefile.am Makefile.in
 DIST_SUBDIRS = $(SUBDIRS)
-SOURCES = $(BAClient_SOURCES) $(BAParser_SOURCES) $(BLClient_SOURCES) $(BLParserLSF_SOURCES) $(BLParserPBS_SOURCES) $(BPRclient_SOURCES) $(BPRserver_SOURCES) $(blahpd_SOURCES) $(blahpd_daemon_SOURCES)
+SOURCES = $(BAClient_SOURCES) $(BAParser_SOURCES) $(BLClient_SOURCES) $(BLParserLSF_SOURCES) $(BLParserPBS_SOURCES) $(BPRclient_SOURCES) $(BPRserver_SOURCES) $(batch_gahp_SOURCES) $(batch_gahp_daemon_SOURCES)
 
 all: all-recursive
 
@@ -347,12 +347,12 @@
 BPRserver$(EXEEXT): $(BPRserver_OBJECTS) $(BPRserver_DEPENDENCIES) 
 	@rm -f BPRserver$(EXEEXT)
 	$(LINK) $(BPRserver_LDFLAGS) $(BPRserver_OBJECTS) $(BPRserver_LDADD) $(LIBS)
-blahpd$(EXEEXT): $(blahpd_OBJECTS) $(blahpd_DEPENDENCIES) 
-	@rm -f blahpd$(EXEEXT)
-	$(CXXLINK) $(blahpd_LDFLAGS) $(blahpd_OBJECTS) $(blahpd_LDADD) $(LIBS)
-blahpd_daemon$(EXEEXT): $(blahpd_daemon_OBJECTS) $(blahpd_daemon_DEPENDENCIES) 
-	@rm -f blahpd_daemon$(EXEEXT)
-	$(CXXLINK) $(blahpd_daemon_LDFLAGS) $(blahpd_daemon_OBJECTS) $(blahpd_daemon_LDADD) $(LIBS)
+batch_gahp$(EXEEXT): $(batch_gahp_OBJECTS) $(batch_gahp_DEPENDENCIES) 
+	@rm -f batch_gahp$(EXEEXT)
+	$(CXXLINK) $(batch_gahp_LDFLAGS) $(batch_gahp_OBJECTS) $(batch_gahp_LDADD) $(LIBS)
+batch_gahp_daemon$(EXEEXT): $(batch_gahp_daemon_OBJECTS) $(batch_gahp_daemon_DEPENDENCIES) 
+	@rm -f batch_gahp_daemon$(EXEEXT)
+	$(CXXLINK) $(batch_gahp_daemon_LDFLAGS) $(batch_gahp_daemon_OBJECTS) $(batch_gahp_daemon_LDADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT) core *.core
diff -u -r org.glite.ce.blahp/src/scripts/lsf_cancel.sh org.glite.ce.blahp-patched/src/scripts/lsf_cancel.sh
--- org.glite.ce.blahp/src/scripts/lsf_cancel.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/lsf_cancel.sh	Wed May  3 16:15:20 2006
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep lsf_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 
 jnr=0
diff -u -r org.glite.ce.blahp/src/scripts/lsf_hold.sh org.glite.ce.blahp-patched/src/scripts/lsf_hold.sh
--- org.glite.ce.blahp/src/scripts/lsf_hold.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/lsf_hold.sh	Wed May  3 16:15:20 2006
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep lsf_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 
 requested=`echo $1 | sed 's/^.*\///'`
diff -u -r org.glite.ce.blahp/src/scripts/lsf_resume.sh org.glite.ce.blahp-patched/src/scripts/lsf_resume.sh
--- org.glite.ce.blahp/src/scripts/lsf_resume.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/lsf_resume.sh	Wed May  3 16:15:20 2006
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep lsf_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 
 requested=`echo $1 | sed 's/^.*\///'`
diff -u -r org.glite.ce.blahp/src/scripts/lsf_status.sh org.glite.ce.blahp-patched/src/scripts/lsf_status.sh
--- org.glite.ce.blahp/src/scripts/lsf_status.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/lsf_status.sh	Wed May  3 16:15:20 2006
@@ -20,7 +20,7 @@
 #  See http://grid.infn.it/grid/license.html for license details.
 #
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep lsf_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 fallback=`grep lsf_fallback $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 BLParser=`grep lsf_BLParser $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
diff -u -r org.glite.ce.blahp/src/scripts/lsf_submit.sh org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh
--- org.glite.ce.blahp/src/scripts/lsf_submit.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh	Wed May  3 16:15:20 2006
@@ -33,7 +33,7 @@
 #
 #
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep lsf_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 confpath=`grep lsf_confpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 fallback=`grep lsf_fallback $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
diff -u -r org.glite.ce.blahp/src/scripts/pbs_cancel.sh org.glite.ce.blahp-patched/src/scripts/pbs_cancel.sh
--- org.glite.ce.blahp/src/scripts/pbs_cancel.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/pbs_cancel.sh	Wed May  3 16:15:20 2006
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep pbs_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 
 jnr=0
diff -u -r org.glite.ce.blahp/src/scripts/pbs_hold.sh org.glite.ce.blahp-patched/src/scripts/pbs_hold.sh
--- org.glite.ce.blahp/src/scripts/pbs_hold.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/pbs_hold.sh	Wed May  3 16:15:20 2006
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep pbs_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 
 requested=`echo $1 | sed 's/^.*\///'`
diff -u -r org.glite.ce.blahp/src/scripts/pbs_resume.sh org.glite.ce.blahp-patched/src/scripts/pbs_resume.sh
--- org.glite.ce.blahp/src/scripts/pbs_resume.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/pbs_resume.sh	Wed May  3 16:15:20 2006
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep pbs_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
 
 requested=`echo $1 | sed 's/^.*\///'`
diff -u -r org.glite.ce.blahp/src/scripts/pbs_status.sh org.glite.ce.blahp-patched/src/scripts/pbs_status.sh
--- org.glite.ce.blahp/src/scripts/pbs_status.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/pbs_status.sh	Wed May  3 16:15:20 2006
@@ -20,7 +20,7 @@
 #  See http://grid.infn.it/grid/license.html for license details.
 #
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 pbsbinpath=`grep pbs_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 spoolpath=`grep pbs_spoolpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 fallback=`grep pbs_fallback $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
diff -u -r org.glite.ce.blahp/src/scripts/pbs_submit.sh org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh
--- org.glite.ce.blahp/src/scripts/pbs_submit.sh	Wed May  3 16:09:24 2006
+++ org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh	Wed May  3 16:15:20 2006
@@ -31,7 +31,7 @@
 #
 #
 
-blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/blah.config"
+blahconffile="${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config"
 binpath=`grep pbs_binpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 spoolpath=`grep pbs_spoolpath $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`/
 fallback=`grep pbs_fallback $blahconffile|grep -v \#|awk -F"=" '{ print $2}'|sed -e 's/ //g'|sed -e 's/\"//g'`
diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp-patched/src/server.c
--- org.glite.ce.blahp/src/server.c	Wed May  3 16:09:23 2006
+++ org.glite.ce.blahp-patched/src/server.c	Wed May  3 16:15:06 2006
@@ -193,7 +193,7 @@
         {
         	gloc = DEFAULT_GLEXEC_COMMAND;
         }
-	conffilestr = make_message("%s/etc/blah.config",result);
+	conffilestr = make_message("%s/etc/batch_gahp.config",result);
 	if((conffile = fopen(conffilestr,"r")) != NULL)
 	{
 		while(fgets(buffer, 128, conffile))
@@ -1810,7 +1810,7 @@
         {
                 glite_loc = DEFAULT_GLITE_LOCATION;
         }
-        blah_conf=make_message("%s/etc/blah.config",glite_loc);
+        blah_conf=make_message("%s/etc/batch_gahp.config",glite_loc);
 
         if(blah_conf!=NULL) conf_file= fopen(blah_conf,"r");
         else return 1;
