diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp-patched/src/server.c
--- org.glite.ce.blahp/src/server.c	Mon Mar 27 09:04:49 2006
+++ org.glite.ce.blahp-patched/src/server.c	Fri Apr  7 13:46:56 2006
@@ -611,8 +611,10 @@
 	}else if(result != C_CLASSAD_NO_ERROR) /* job must be refused */
         {
         	/* PUSH A FAILURE */
+/*
         	resultLine = make_message("%s 1 Proxy\\ not\\ found N/A", reqId);
         	goto cleanup_command;   
+*/
 	}
 
 	/* Cmd attribute is mandatory: stop on any error */
