--- org.glite.ce.blahp/src/server.c	Tue May 30 14:14:09 2006
+++ org.glite.ce.blahp-patched/src/server.c	Wed Jun  7 00:22:45 2006
@@ -188,7 +188,7 @@
          	 setenv("LD_LIBRARY_PATH",needed_libs,1);
 	
 	blah_script_location = make_message(BINDIR_LOCATION, result);
-	blah_version = make_message(RCSID_VERSION, VERSION, "poly");
+	blah_version = make_message(RCSID_VERSION, VERSION, "poly,new_esc_format");
         if ((gloc=getenv("GLEXEC_COMMAND")) == NULL)
         {
         	gloc = DEFAULT_GLEXEC_COMMAND;
@@ -696,12 +696,12 @@
                                 free(conv_environment);
                                 goto cleanup_command;
                         }
+		        free(environment);
+		        free(conv_environment);
+                        /* Swap new command in */
+                        free(command);
+                        command = command_ext;
                 }
-		free(environment);
-		free(conv_environment);
-               /* Swap new command in */
-               free(command);
-               command = command_ext;
         }else
         if(set_cmd_string_option(&command, cad, "Env","-v", SINGLE_QUOTE) == C_CLASSAD_OUT_OF_MEMORY)
         {
@@ -729,12 +729,12 @@
                                 free(conv_arguments);
                                 goto cleanup_command;
                         }
+                        free(arguments);
+                        free(conv_arguments);
+                        /* Swap new command in */
+                        free(command);
+                        command = command_ext;
                 }
-                free(arguments);
-                free(conv_arguments);
-               /* Swap new command in */
-               free(command);
-               command = command_ext;
         }else
         if(set_cmd_string_option(&command, cad, "Args","--", NO_QUOTE) == C_CLASSAD_OUT_OF_MEMORY)
         {
@@ -2345,6 +2345,7 @@
 	sprintf(quoted_sep, CONVARG_QUOTSEP, separator);
 
 	orig_len = strlen(original);
+	if (orig_len == 0) return NULL;
 
         /* Worst case is <a> --> <"\"a\""> */
 	result = (char *)malloc(orig_len * 10);
