--- condor_config.generic	2023-10-06 13:27:28.225551803 -0500
+++ condor_config.generic	2024-05-21 15:47:46.312454837 -0500
@@ -19,27 +19,25 @@
 ######################################################################
 
 ##  Where have you installed the bin, sbin and lib condor directories?
-RELEASE_DIR = /usr/local/condor
+RELEASE_DIR = /usr
 
 ##  Where is the local condor directory for each host?  This is where the local config file(s), logs and
 ##  spool/execute directories are located. this is the default for Linux and Unix systems.
-#LOCAL_DIR = $(TILDE)
-##  this is the default on Windows sytems
-#LOCAL_DIR = $(RELEASE_DIR)
+LOCAL_DIR = /var
 
 ##  Where is the machine-specific local config file for each host?
-LOCAL_CONFIG_FILE = $(LOCAL_DIR)/condor_config.local
+LOCAL_CONFIG_FILE = /etc/condor/condor_config.local
 ##  If your configuration is on a shared file system, then this might be a better default
 #LOCAL_CONFIG_FILE = $(RELEASE_DIR)/etc/$(HOSTNAME).local
 ##  If the local config file is not present, is it an error? (WARNING: This is a potential security issue.)
-#REQUIRE_LOCAL_CONFIG_FILE = true
+REQUIRE_LOCAL_CONFIG_FILE = false
 
 ##  The normal way to do configuration with RPM and Debian packaging is to read all of the
 ##  files in a given directory that don't match a regex as configuration files.
 ##  Config files are read in lexicographic order.
 ##  Multiple directories may be specified, separated by commas; directories
 ##  are read in left-to-right order.
-LOCAL_CONFIG_DIR = $(LOCAL_DIR)/config
+LOCAL_CONFIG_DIR = /usr/share/condor/config.d,/etc/condor/config.d
 #LOCAL_CONFIG_DIR_EXCLUDE_REGEXP = ^((\..*)|(.*~)|(#.*)|(.*\.rpmsave)|(.*\.rpmnew))$
 
 ##
@@ -68,5 +66,28 @@
 #FLOCK_TO = condor.cs.wisc.edu, cm.example.edu
 
 ##--------------------------------------------------------------------
-## Values set by the condor_configure script:
+## Values set by the rpm patch script:
 ##--------------------------------------------------------------------
+
+## For Unix machines, the path and file name of the file containing
+## the pool password for password authentication.
+#SEC_PASSWORD_FILE = $(LOCAL_DIR)/lib/condor/pool_password
+
+##  Pathnames
+RUN     = $(LOCAL_DIR)/run/condor
+LOG     = $(LOCAL_DIR)/log/condor
+LOCK    = $(LOCAL_DIR)/lock/condor
+SPOOL   = $(LOCAL_DIR)/lib/condor/spool
+EXECUTE = $(LOCAL_DIR)/lib/condor/execute
+BIN     = $(RELEASE_DIR)/bin
+LIB     = $(RELEASE_DIR)/lib64/condor
+INCLUDE = $(RELEASE_DIR)/include/condor
+SBIN    = $(RELEASE_DIR)/sbin
+LIBEXEC = $(RELEASE_DIR)/libexec/condor
+SHARE   = $(RELEASE_DIR)/share/condor
+
+JAVA_CLASSPATH_DEFAULT = $(SHARE) .
+
+SINGULARITY = /usr/bin/singularity
+
+##  Install the minicondor package to run HTCondor on a single node
