--- condor_config.generic	2015-05-29 15:49:25.000804760 -0500
+++ condor_config.patched.X86_64	2015-05-29 15:49:14.849710094 -0500
@@ -62,7 +62,7 @@
 SPOOL   = $(LOCAL_DIR)/lib/condor/spool
 EXECUTE = $(LOCAL_DIR)/lib/condor/execute
 BIN     = $(RELEASE_DIR)/bin
-LIB     = $(RELEASE_DIR)/lib/condor
+LIB     = $(RELEASE_DIR)/lib64/condor
 INCLUDE = $(RELEASE_DIR)/include/condor
 SBIN    = $(RELEASE_DIR)/sbin
 LIBEXEC = $(RELEASE_DIR)/libexec/condor
