--- condor_config.generic	2021-05-01 07:31:58.090876247 -0500
+++ condor_config.patched	2021-05-01 07:33:10.903564239 -0500
@@ -19,27 +19,25 @@
 ######################################################################
 
 ##  Where have you installed the bin, sbin and lib condor directories?
-RELEASE_DIR = /usr/local/condor
+RELEASE_DIR = /usr
 
 ##  Where is the local condor directory for each host?  This is where the local config file(s), logs and
 ##  spool/execute directories are located. this is the default for Linux and Unix systems.
-#LOCAL_DIR = $(TILDE)
-##  this is the default on Windows sytems
-#LOCAL_DIR = $(RELEASE_DIR)
+LOCAL_DIR = /var
 
 ##  Where is the machine-specific local config file for each host?
-LOCAL_CONFIG_FILE = $(LOCAL_DIR)/condor_config.local
+LOCAL_CONFIG_FILE = /etc/condor/condor_config.local
 ##  If your configuration is on a shared file system, then this might be a better default
 #LOCAL_CONFIG_FILE = $(RELEASE_DIR)/etc/$(HOSTNAME).local
 ##  If the local config file is not present, is it an error? (WARNING: This is a potential security issue.)
-#REQUIRE_LOCAL_CONFIG_FILE = true
+REQUIRE_LOCAL_CONFIG_FILE = false
 
 ##  The normal way to do configuration with RPM and Debian packaging is to read all of the
 ##  files in a given directory that don't match a regex as configuration files.
 ##  Config files are read in lexicographic order.
 ##  Multiple directories may be specified, separated by commas; directories
 ##  are read in left-to-right order.
-LOCAL_CONFIG_DIR = $(LOCAL_DIR)/config
+LOCAL_CONFIG_DIR = /usr/share/condor/config.d,/etc/condor/config.d
 #LOCAL_CONFIG_DIR_EXCLUDE_REGEXP = ^((\..*)|(.*~)|(#.*)|(.*\.rpmsave)|(.*\.rpmnew))$
 
 ##
@@ -68,5 +66,36 @@
 #FLOCK_TO = condor.cs.wisc.edu, cm.example.edu
 
 ##--------------------------------------------------------------------
-## Values set by the condor_configure script:
+## Values set by the debian patch script:
 ##--------------------------------------------------------------------
+
+## For Unix machines, the path and file name of the file containing
+## the pool password for password authentication.
+#SEC_PASSWORD_FILE = $(LOCAL_DIR)/lib/condor/pool_password
+
+##  Pathnames
+RUN     = $(LOCAL_DIR)/run/condor
+LOG     = $(LOCAL_DIR)/log/condor
+LOCK    = $(LOCAL_DIR)/lock/condor
+SPOOL   = $(LOCAL_DIR)/spool/condor
+EXECUTE = $(LOCAL_DIR)/lib/condor/execute
+CRED_STORE_DIR = $(LOCAL_DIR)/lib/condor/cred_dir
+ETC     = /etc/condor
+BIN     = $(RELEASE_DIR)/bin
+LIB     = $(RELEASE_DIR)/lib/condor
+INCLUDE = $(RELEASE_DIR)/include/condor
+SBIN    = $(RELEASE_DIR)/sbin
+LIBEXEC = $(RELEASE_DIR)/lib/condor/libexec
+SHARE   = $(RELEASE_DIR)/share/condor
+MAIL    = /usr/bin/mail
+GANGLIA_LIB64_PATH = /lib,/usr/lib,/usr/local/lib
+
+PROCD_ADDRESS = $(RUN)/procd_pipe
+
+# Account for different pki locations for Debian
+AUTH_SSL_SERVER_CAFILE   = /etc/ssl/certs/ca-certificates.crt
+AUTH_SSL_CLIENT_CAFILE   = /etc/ssl/certs/ca-certificates.crt
+AUTH_SSL_SERVER_CERTFILE = /etc/ssl/certs/localhost.crt
+AUTH_SSL_SERVER_KEYFILE  = /etc/ssl/private/localhost.key
+
+##  Install the minihtcondor package to run HTCondor on a single node
