--- Makeconfig	2006-07-10 16:42:27.000000000 -0500
+++ glibc-2.5-20061008T1257/Makeconfig	2007-12-11 11:50:44.000000000 -0600
@@ -115,6 +115,14 @@
 +sysdep_dirs := $(objdir) $(+sysdep_dirs)
 endif
 
+# If using static nss, then create some references to those objects.
+ifeq (yes,$(build-static-nss))
+nssobjdir := $(patsubst ../$(subdir),.,$(common-objpfx)nss)
+resolvobjdir := $(patsubst ../$(subdir),.,$(common-objpfx)resolv)
+otherlibs += $(nssobjdir)/libnss_files.a $(resolvobjdir)/libnss_dns.a \
+			$(resolvobjdir)/libresolv.a
+endif
+
 # Run config.status to update config.make and config.h.  We don't show the
 # dependence of config.h to Make, because it is only touched when it
 # changes and so config.status would be run every time; the dependence of
@@ -510,7 +518,8 @@
 
 # The static libraries.
 ifeq (yes,$(build-static))
-link-libc-static = $(common-objpfx)libc.a $(static-gnulib) $(common-objpfx)libc.a
+# Add in the static nss libraries too!
+link-libc-static = $(common-objpfx)libc.a $(static-gnulib) $(otherlibs) $(common-objpfx)libc.a
 else
 ifeq (yes,$(build-shared))
 # We can try to link the programs with lib*_pic.a...
