diff -u -r org.gridsite.core/interface/gridsite.h org.gridsite.core.patched/interface/gridsite.h
--- org.gridsite.core/interface/gridsite.h	2007-11-15 07:04:11.000000000 -0600
+++ org.gridsite.core.patched/interface/gridsite.h	2013-05-08 16:38:52.000000000 -0500
@@ -84,7 +84,7 @@
 
 #define GRSTerrorLog(GRSTerrorLevel, ...) if (GRSTerrorLogFunc != NULL) (GRSTerrorLogFunc)(__FILE__, __LINE__, GRSTerrorLevel, __VA_ARGS__)
 
-void (*GRSTerrorLogFunc)(char *, int, int, char *, ...);
+extern void (*GRSTerrorLogFunc)(char *, int, int, char *, ...);
 
 /* these levels are the same as Unix syslog() and Apache ap_log_error() */
 
