diff -ru condor-7.2.0/src/condor_examples/condor.init condor-7.2.0mf/src/condor_examples/condor.init
--- condor-7.2.0/src/condor_examples/condor.init	2008-12-19 14:10:08.000000000 -0600
+++ condor-7.2.0mf/src/condor_examples/condor.init	2009-01-13 19:31:15.000000000 -0600
@@ -2,7 +2,7 @@
 #
 # condor	This script allows for starting and stopping Condor.
 #
-# chkconfig: 2345 98 10
+# chkconfig: - 98 10
 # description: Condor is a high throughput computing batch processing
 # 	       platform.
 # processname: condor_master
