--- ./src/condor_tools/cmake_install.cmake.orig	2013-08-22 00:01:37.175200137 -0500
+++ ./src/condor_tools/cmake_install.cmake	2013-08-22 00:31:16.668240765 -0500
@@ -271,17 +271,17 @@
 ENDIF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
 
 IF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
-  execute_process(COMMAND mkdir -p /usr/lib/condor)
+  execute_process(COMMAND mkdir -p $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib/condor)
 ENDIF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
 
 IF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
   execute_process(COMMAND /builddir/build/BUILD/condor-8.1.2/src/condor_scripts/print-libs /builddir/build/BUILD/condor-8.1.2/src/condor_tools/condor_version /builddir/build/BUILD/condor-8.1.2/src/condor_scripts/syslib-patterns
-		 COMMAND xargs -isrc cp src /usr/lib/condor
+		 COMMAND xargs -isrc cp src $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib/condor
 		 )
 ENDIF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
 
 IF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
-  execute_process(COMMAND chmod -R u+w /usr/lib/condor/)
+  execute_process(COMMAND chmod -R u+w $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib/condor/)
 ENDIF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
 
 IF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
