diff --git a/src/condor_examples/condor.init b/src/condor_examples/condor.init
index 97be80d..13831b4 100644
 --- a/src/condor_examples/condor.init
+++ b/src/condor_examples/condor.init
@@ -71,6 +87,12 @@
     return $RETVAL
 }
 
+peaceful-off() {
+    condor_off -peaceful
+    rc=$?
+    exit $rc
+}
+
 #
 # Determine if a process is running only by looking in a pidfile.
 # There is no use of pidof, which can find processes that are not
@@ -226,8 +247,14 @@
 	status -p $pidfile $prog
 	RETVAL=$?
 	;;
+
+    peaceful-off)
+	[ $running -eq 0 ] || exit 0
+	peaceful-off
+	RETVAL=$?
+       ;;
     *)
-	echo $"Usage: $0 {start|stop|restart|condrestart|try-restart|reload|force-reload|status}"
+	echo $"Usage: $0 {start|stop|restart|condrestart|try-restart|reload|force-reload|status|peaceful-off}"
 	RETVAL=2
 esac

