diff --git a/src/condor_utils/tests/ring_buffer_tests.cpp b/src/condor_utils/tests/ring_buffer_tests.cpp
index 1267092..9e3bf57 100644
--- a/src/condor_utils/tests/ring_buffer_tests.cpp
+++ b/src/condor_utils/tests/ring_buffer_tests.cpp
@@ -31,5 +20,6 @@
 #define BOOST_TEST_MAIN
 #define BOOST_AUTO_TEST_MAIN
-#include <boost/test/unit_test.hpp>
+#include <boost141/boost/test/auto_unit_test.hpp>
+#include <boost141/boost/test/included/unit_test.hpp>

 #define BOOST_TEST_MODULE ring_buffer

 using std::deque;
