--- condor-7.9.6/src/condor_syscall_lib/CMakeLists.txt.orig	2013-05-08 16:14:37.000000000 -0500
+++ condor-7.9.6/src/condor_syscall_lib/CMakeLists.txt	2013-07-18 15:12:30.735142223 -0500
@@ -125,13 +125,14 @@
 	# part 2:
 	# set dependencies to ensure build dep order for parallel builds
 	##################################################
-	add_dependencies( condorsyscall stub_gen )
+	add_dependencies( condorsyscall senders switches )
 	if (DOES_COMPRESS_CKPT)
-		add_dependencies( condorzsyscall stub_gen )
+		add_dependencies( condorzsyscall senders switches )
 	endif()
 	add_dependencies( switches syscall_numbers stub_gen )
 	add_dependencies( senders syscall_numbers stub_gen )
 	add_dependencies( receivers syscall_numbers stub_gen )
+	add_dependencies( std_local_ref senders receivers )
 	add_dependencies( sys-list syscall_numbers stub_gen )
 
 	#################################################
