diff --git a/src/condor_examples/condor.init b/src/condor_examples/condor.init
index 97be80d..81b4e00 100644
--- a/src/condor_examples/condor.init
+++ b/src/condor_examples/condor.init
@@ -35,6 +38,7 @@ pidfile=/var/run/condor/$prog.pid
 # Check that networking is up
 [ "${NETWORKING}" = "no" ] && exit 1
 
+ulimit $ULIMIT_FLAGS > /dev/null 2>&1
 
 start() {
     echo -n $"Starting Condor daemons: "
