--- glibc-2.17-36-x86_64/nss/Makefile	2012-12-24 21:02:13.000000000 -0600
+++ new/nss/Makefile	2014-01-21 11:42:30.017933492 -0600
@@ -70,8 +70,8 @@
 install-others		+= $(inst_vardbdir)/Makefile
 
 # Build static module into libc if requested
-libnss_files-inhibit-o	= $(filter-out .os,$(object-suffixes))
-libnss_db-inhibit-o	= $(filter-out .os,$(object-suffixes))
+#libnss_files-inhibit-o	= $(filter-out .os,$(object-suffixes))
+#libnss_db-inhibit-o	= $(filter-out .os,$(object-suffixes))
 ifeq ($(build-static-nss),yes)
 routines                += $(libnss_files-routines)
 static-only-routines    += $(libnss_files-routines)
--- glibc-2.17-36-x86_64/resolv/Makefile	2012-12-24 21:02:13.000000000 -0600
+++ new/resolv/Makefile	2014-01-21 11:40:39.241972791 -0600
@@ -55,7 +55,7 @@
 vpath %.c nss_dns
 
 libnss_dns-routines	:= dns-host dns-network dns-canon
-libnss_dns-inhibit-o	= $(filter-out .os,$(object-suffixes))
+# libnss_dns-inhibit-o	= $(filter-out .os,$(object-suffixes))
 ifeq ($(build-static-nss),yes)
 routines                += $(libnss_dns-routines) $(libresolv-routines)
 static-only-routines    += $(libnss_dns-routines) $(libresolv-routines)
