--- libc-symbols.h.orig	2011-12-01 13:04:30.000000000 -0600
+++ glibc-2.12-2-x86_64/include/libc-symbols.h	2011-12-01 13:04:44.000000000 -0600
@@ -283,15 +283,9 @@
 		"warning: " #name " is not implemented and will always fail")
 
 /* Warning for linking functions calling dlopen into static binaries.  */
-#ifdef SHARED
-#define static_link_warning(name)
-#else
-#define static_link_warning(name) static_link_warning1(name)
-#define static_link_warning1(name) \
-  link_warning(name, "Using '" #name "' in statically linked applications \
-requires at runtime the shared libraries from the glibc version used \
-for linking")
-#endif
+/* GGT disable linker warning about deprecated functions, because
+	we're always static-nss */
+#define static_link_warning(name) 
 
 /* Declare SYMBOL to be TYPE (`function' or `object') of SIZE bytes
    alias to ORIGINAL, when the assembler supports such declarations
