Index: voms-2.0.2/src/include/gssapi_compat.h
===================================================================
--- voms-2.0.2.orig/src/include/gssapi_compat.h	2011-09-12 18:09:40.000000000 -0500
+++ voms-2.0.2/src/include/gssapi_compat.h	2011-09-12 19:20:28.000000000 -0500
@@ -72,7 +72,7 @@
     globus_gsi_cred_handle_t            cred_handle;
 } gss2_cred_id_desc;
 
-#if !defined(GLOBUS_INCLUDE_GLOBUS_THREAD)
+#if !(defined(GLOBUS_INCLUDE_GLOBUS_THREAD) || defined(GLOBUS_INCLUDE_GLOBUS_THREAD_H))
 typedef int globus_mutex_t;
 #endif
 
