--- configure.orig	2008-01-28 08:50:17.000000000 -0600
+++ configure	2008-02-01 15:51:45.127935904 -0600
@@ -21153,12 +21153,8 @@
 { echo "$as_me:$LINENO: result: $ac_cv_type_long_long" >&5
 echo "${ECHO_T}$ac_cv_type_long_long" >&6; }
 if test $ac_cv_type_long_long = yes; then
-
-cat >>confdefs.h <<_ACEOF
-#define HAVE_LONG_LONG 1
-_ACEOF
-
-pcre_have_long_long="1"
+  echo CONDOR PATCH: forcing HAVE_LONG_LONG to false
+pcre_have_long_long="0"
 else
   pcre_have_long_long="0"
 fi
@@ -21217,12 +21213,8 @@
 { echo "$as_me:$LINENO: result: $ac_cv_type_unsigned_long_long" >&5
 echo "${ECHO_T}$ac_cv_type_unsigned_long_long" >&6; }
 if test $ac_cv_type_unsigned_long_long = yes; then
-
-cat >>confdefs.h <<_ACEOF
-#define HAVE_UNSIGNED_LONG_LONG 1
-_ACEOF
-
-pcre_have_ulong_long="1"
+  echo CONDOR PATCH: forcing HAVE_UNSIGNED_LONG_LONG to false
+pcre_have_ulong_long="0"
 else
   pcre_have_ulong_long="0"
 fi
