--- RunGrepTest.orig	2007-12-05 14:40:51.000000000 -0600
+++ RunGrepTest	2008-02-07 22:29:50.562943714 -0600
@@ -14,7 +14,7 @@
 echo "Testing pcregrep"
 $pcregrep -V
 
-cf="diff -ub"
+cf="diff -b"
 valgrind=
 
 while [ $# -gt 0 ] ; do
--- RunTest.orig	2007-09-17 04:35:29.000000000 -0500
+++ RunTest	2008-02-07 22:29:46.727784628 -0600
@@ -2,7 +2,7 @@
 
 # Run PCRE tests
 
-cf="diff -u"
+cf="diff"
 valgrind=
 testdata=testdata
 
