--- libvirt-0.6.2/configure	2009-04-03 09:48:52.000000000 -0500
+++ libvirt-0.6.2-patched/configure	2009-08-28 13:12:17.000000000 -0500
@@ -46184,7 +46184,7 @@
   fail=1
 fi
 
-
+fail=0
   test $fail = 1 &&
     { { echo "$as_me:$LINENO: error: You must install the GnuTLS library in order to compile and run libvirt" >&5
 echo "$as_me: error: You must install the GnuTLS library in order to compile and run libvirt" >&2;}
