diff -u -r voms-1.8.9/configure voms-1.8.9-patched/configure
--- voms-1.8.9/configure	2008-12-23 09:27:19.000000000 -0600
+++ voms-1.8.9-patched/configure	2009-08-17 16:20:43.000000000 -0500
@@ -19615,6 +19615,7 @@
   if test "x$found" = "xyes"; then
     NO_GLOBUS_FLAGS="-I$with_openssl_prefix/include"
     OPENSSL_LIBS="-L$with_openssl_prefix/lib -lcrypto -lssl"
+    OPENSSL_HACK="-L$with_openssl_prefix/lib"
 
 
     echo "$as_me:$LINENO: checking for system OpenSSL version" >&5
@@ -20105,50 +20106,50 @@
       echo "$as_me:$LINENO: result: found flavor=$FLVR" >&5
 echo "${ECHO_T}found flavor=$FLVR" >&6
       if test "x$flavor" = "x$with_globus_flavor" ; then
-      	GLOBUS_CFLAGS="-I$with_globus_prefix/include/$flavor"
+      	GLOBUS_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32" ; then
-	      GLOBUS_GCC32_CFLAGS="-I$with_globus_prefix/include/$flavor"
+	      GLOBUS_GCC32_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC32_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC32_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC32_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32dbg" ; then
-        GLOBUS_GCC32DBG_CFLAGS="-I$with_globus_prefix/include/$flavor"
+        GLOBUS_GCC32DBG_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC32DBG_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC32DBG_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC32DBG_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32dbgpthr" ; then
-        GLOBUS_GCC32DBGPTHR_CFLAGS="-I$with_globus_prefix/include/$flavor"
+        GLOBUS_GCC32DBGPTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC32DBGPTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC32DPBPTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC32DPBPTHR_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32pthr" ; then
-        GLOBUS_GCC32PTHR_CFLAGS="-I$with_globus_prefix/include/$flavor"
+        GLOBUS_GCC32PTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC32PTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC32PTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC32PTHR_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
 
       if test "x$flavor" = "xgcc64" ; then
-	      GLOBUS_GCC64_CFLAGS="-I$with_globus_prefix/include/$flavor"
+	      GLOBUS_GCC64_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC64_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC64_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC64_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc64dbg" ; then
-        GLOBUS_GCC64DBG_CFLAGS="-I$with_globus_prefix/include/$flavor"
+        GLOBUS_GCC64DBG_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC64DBG_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC64DBG_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC64DBG_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc64dbgpthr" ; then
-        GLOBUS_GCC64DBGPTHR_CFLAGS="-I$with_globus_prefix/include/$flavor"
+        GLOBUS_GCC64DBGPTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC64DBGPTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC64DBGPTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC64DBGPTHR_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc64pthr" ; then
-        GLOBUS_GCC64PTHR_CFLAGS="-I$with_globus_prefix/include/$flavor"
+        GLOBUS_GCC64PTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
         GLOBUS_GCC64PTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
-        GLOBUS_GCC64PTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
+        GLOBUS_GCC64PTHR_GSS_API_LIBS="$ac_globus_ldlib $OPENSSL_HACK -lcrypto$FLVR -lssl$FLVR"
       fi
     done
 
