--- org.glite.ce.blahp/src/main.c	2006-11-30 11:36:11.000000000 -0600
+++ org.glite.ce.blahp-patched/src/main.c	2009-02-12 23:30:29.000000000 -0600
@@ -30,7 +30,6 @@
 #include <sys/select.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <mcheck.h>
 
 #include "blahpd.h"
 #include "server.h"
--- org.glite.ce.blahp/src/main_daemon.c	2004-09-14 10:20:54.000000000 -0500
+++ org.glite.ce.blahp-patched/src/main_daemon.c	2009-02-12 23:30:04.000000000 -0600
@@ -30,7 +30,6 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <syslog.h>
-#include <mcheck.h>
 
 #include "blahpd.h"
 #include "server.h"
@@ -51,8 +50,6 @@
     int exit_program = 0;
     pid_t pid;
 
-    mtrace();
-    
     openlog("blahpd", LOG_PID, LOG_DAEMON);
     syslog(LOG_DAEMON | LOG_INFO, "Starting blah server (%s)", RCSID_VERSION);
     
