diff -u -r gt5.2.5-all-source-installer/source-trees/gram/client/source/dirt.sh gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/dirt.sh
--- gt5.2.5-all-source-installer/source-trees/gram/client/source/dirt.sh	2012-02-13 08:36:02.000000000 -0600
+++ gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/dirt.sh	2015-03-30 16:36:25.000000000 -0500
@@ -1,2 +1,2 @@
-DIRT_TIMESTAMP=1329143762
+DIRT_TIMESTAMP=1414000480
 DIRT_BRANCH_ID=83
diff -u -r gt5.2.5-all-source-installer/source-trees/gram/client/source/globus_gram_client.c gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/globus_gram_client.c
--- gt5.2.5-all-source-installer/source-trees/gram/client/source/globus_gram_client.c	2011-11-15 14:15:13.000000000 -0600
+++ gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/globus_gram_client.c	2015-03-30 16:39:13.000000000 -0500
@@ -603,14 +603,9 @@
     {
         goto destroy_2;
     }
-    /* HACK: To support 4.0.8 which VDT widely redistributed, and
-     * which doesn't handle TLSv1 correctly when exporting/importing
-     * security contexts from the gatekeeper to job manager, we
-     * must use SSLv3 instead
-     */
     res = globus_io_attr_set_secure_channel_mode(
         attrp,
-        GLOBUS_IO_SECURE_CHANNEL_MODE_GSI_WRAP_SSL3);
+        GLOBUS_IO_SECURE_CHANNEL_MODE_GSI_WRAP);
     if (res != GLOBUS_SUCCESS)
     {
         goto destroy_2;
diff -u -r gt5.2.5-all-source-installer/source-trees/gram/client/source/gptdata.sh gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/gptdata.sh
--- gt5.2.5-all-source-installer/source-trees/gram/client/source/gptdata.sh	2013-10-30 11:19:32.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/gptdata.sh	2015-03-30 16:39:38.000000000 -0500
@@ -1,6 +1,6 @@
 GPT_NAME='globus_gram_client'
 GPT_MAJOR_VERSION='12'
-GPT_MINOR_VERSION='4'
+GPT_MINOR_VERSION='5'
 GPT_AGE_VERSION='9'
 GPT_CFLAGS='@GPT_PKG_CFLAGS@'
 GPT_INCLUDES='@GPT_EXTERNAL_INCLUDES@'
diff -u -r gt5.2.5-all-source-installer/source-trees/gram/client/source/pkgdata/pkg_data_src.gpt.in gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/pkgdata/pkg_data_src.gpt.in
--- gt5.2.5-all-source-installer/source-trees/gram/client/source/pkgdata/pkg_data_src.gpt.in	2012-02-13 08:36:02.000000000 -0600
+++ gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/pkgdata/pkg_data_src.gpt.in	2015-03-30 16:40:05.000000000 -0500
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE gpt_package_metadata SYSTEM "package.dtd">
 <gpt_package_metadata Format_Version="0.02" Name="globus_gram_client">
-    <Aging_Version Major="12" Minor="4" Age="9"/>
+    <Aging_Version Major="12" Minor="5" Age="9"/>
     <Description>GRAM Client Library</Description>
     <Functional_Group>ResourceManagement</Functional_Group>
     <Version_Stability Release="Experimental" Dirt="@DIRT_TIMESTAMP@-@DIRT_BRANCH_ID@"/>
diff -u -r gt5.2.5-all-source-installer/source-trees/gram/client/source/pkgdata/pkg_data_src.pc.in gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/pkgdata/pkg_data_src.pc.in
--- gt5.2.5-all-source-installer/source-trees/gram/client/source/pkgdata/pkg_data_src.pc.in	2013-10-30 11:19:26.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gram/client/source/pkgdata/pkg_data_src.pc.in	2015-03-30 16:40:20.000000000 -0500
@@ -7,7 +7,7 @@
 
 Name: globus-gram-client
 Description: Globus Toolkit - GRAM Client Library
-Version: 12.4
+Version: 12.5
 Requires.private: globus-gram-protocol >= 11 globus-common >= 14 @GPT_PKGCONFIG_DEPENDENCIES@ globus-gss-assist >= 8 globus-rsl >= 9 globus-io >= 9 globus-core >= 8
 Libs: -L${libdir} -lglobus_gram_client
 Libs.private: @GPT_EXTERNAL_LIBS@
