diff -u -r gt5.2.5-all-source-installer/source-trees/gass/cache/source/configure gt5.2.5-all-source-installer-patched/source-trees/gass/cache/source/configure
--- gt5.2.5-all-source-installer/source-trees/gass/cache/source/configure	2013-10-30 11:09:16.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gass/cache/source/configure	2014-01-23 11:25:29.000000000 -0600
@@ -18397,6 +18397,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gass/copy/source/configure gt5.2.5-all-source-installer-patched/source-trees/gass/copy/source/configure
--- gt5.2.5-all-source-installer/source-trees/gass/copy/source/configure	2013-10-30 11:20:32.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gass/copy/source/configure	2014-01-23 11:25:58.000000000 -0600
@@ -19095,6 +19095,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gatekeeper/source/configure gt5.2.5-all-source-installer-patched/source-trees/gatekeeper/source/configure
--- gt5.2.5-all-source-installer/source-trees/gatekeeper/source/configure	2013-10-30 11:14:24.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gatekeeper/source/configure	2014-01-23 11:26:25.000000000 -0600
@@ -19117,6 +19117,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gram/jobmanager/source/configure gt5.2.5-all-source-installer-patched/source-trees/gram/jobmanager/source/configure
--- gt5.2.5-all-source-installer/source-trees/gram/jobmanager/source/configure	2013-10-30 11:21:04.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gram/jobmanager/source/configure	2014-01-23 11:26:46.000000000 -0600
@@ -19344,6 +19344,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gridftp/client/source/configure gt5.2.5-all-source-installer-patched/source-trees/gridftp/client/source/configure
--- gt5.2.5-all-source-installer/source-trees/gridftp/client/source/configure	2013-10-30 11:19:12.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gridftp/client/source/configure	2014-01-23 11:30:46.000000000 -0600
@@ -19093,6 +19093,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gridftp/server/src/configure gt5.2.5-all-source-installer-patched/source-trees/gridftp/server/src/configure
--- gt5.2.5-all-source-installer/source-trees/gridftp/server/src/configure	2013-10-30 11:19:58.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gridftp/server/src/configure	2014-01-23 11:36:54.000000000 -0600
@@ -19097,6 +19097,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/callback/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/callback/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/callback/source/configure	2013-10-30 11:13:07.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/callback/source/configure	2014-01-23 11:27:05.000000000 -0600
@@ -19248,6 +19248,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/cert_utils/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/cert_utils/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/cert_utils/source/configure	2013-10-30 11:12:46.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/cert_utils/source/configure	2014-01-23 11:27:19.000000000 -0600
@@ -19338,6 +19338,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/credential/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/credential/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/credential/source/configure	2013-10-30 11:13:18.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/credential/source/configure	2014-01-23 11:27:35.000000000 -0600
@@ -19092,6 +19092,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/gridmap_eppn_callout/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/gridmap_eppn_callout/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/gridmap_eppn_callout/configure	2013-10-30 11:18:00.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/gridmap_eppn_callout/configure	2014-01-23 11:37:43.000000000 -0600
@@ -19088,6 +19088,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/gridmap_verify_myproxy_callout/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/gridmap_verify_myproxy_callout/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/gridmap_verify_myproxy_callout/configure	2013-10-30 11:18:10.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/gridmap_verify_myproxy_callout/configure	2014-01-23 11:38:07.000000000 -0600
@@ -19088,6 +19088,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/gssapi/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/gssapi/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/gssapi/source/configure	2013-10-30 11:13:38.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/gssapi/source/configure	2014-01-23 11:27:52.000000000 -0600
@@ -19093,6 +19093,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/gssapi/test/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/gssapi/test/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/gssapi/test/configure	2013-10-30 11:17:17.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/gssapi/test/configure	2014-01-23 11:28:13.000000000 -0600
@@ -18391,6 +18391,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/openssl_error/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/openssl_error/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/openssl_error/source/configure	2013-10-30 11:09:36.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/openssl_error/source/configure	2014-01-23 11:28:28.000000000 -0600
@@ -19334,6 +19334,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/openssl_error/test/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/openssl_error/test/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/openssl_error/test/configure	2013-10-30 11:10:32.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/openssl_error/test/configure	2014-01-23 11:28:40.000000000 -0600
@@ -19088,6 +19088,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/openssl_module/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/openssl_module/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/openssl_module/source/configure	2013-10-30 11:10:43.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/openssl_module/source/configure	2014-01-23 11:28:57.000000000 -0600
@@ -19335,6 +19335,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_core/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_core/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_core/source/configure	2013-10-30 11:13:28.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_core/source/configure	2014-01-23 11:29:23.000000000 -0600
@@ -19092,6 +19092,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_ssl/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_ssl/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_ssl/source/configure	2013-10-30 11:07:46.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_ssl/source/configure	2014-01-23 11:29:40.000000000 -0600
@@ -19333,6 +19333,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_ssl/test/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_ssl/test/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_ssl/test/configure	2013-10-30 11:09:46.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_ssl/test/configure	2014-01-23 11:29:53.000000000 -0600
@@ -18391,6 +18391,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_utils/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_utils/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/proxy/proxy_utils/source/configure	2013-10-30 11:14:55.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/proxy/proxy_utils/source/configure	2014-01-23 11:30:09.000000000 -0600
@@ -19425,6 +19425,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
diff -u -r gt5.2.5-all-source-installer/source-trees/gsi/sysconfig/source/configure gt5.2.5-all-source-installer-patched/source-trees/gsi/sysconfig/source/configure
--- gt5.2.5-all-source-installer/source-trees/gsi/sysconfig/source/configure	2013-10-30 11:12:56.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/gsi/sysconfig/source/configure	2014-01-23 11:30:29.000000000 -0600
@@ -19092,6 +19092,9 @@
             esac
 
 fi;
+if test `uname` = "Darwin" ; then
+    openssl="/usr"
+fi
 
 if test "${openssl}" != ""; then
     PKG_CONFIG_PATH=${openssl}/lib/pkgconfig
