diff -u -r gt5.2.5-all-source-installer/source-trees/xio/src/configure gt5.2.5-all-source-installer-patched/source-trees/xio/src/configure
--- gt5.2.5-all-source-installer/source-trees/xio/src/configure	2013-10-30 11:10:12.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/xio/src/configure	2014-01-23 11:55:35.000000000 -0600
@@ -19616,7 +19616,7 @@
 
 
 
-                                                                                                                                                                          ac_config_files="$ac_config_files Makefile pkgdata/Makefile pkgdata/pkg_data_src.gpt builtins/Makefile doxygen/Makefile doxygen/Doxyfile doxygen/Doxyfile-internal version.h extensions.h builtins/file/Makefile builtins/http/Makefile builtins/mode_e/Makefile builtins/ordering/Makefile builtins/queue/Makefile builtins/tcp/Makefile builtins/telnet/Makefile builtins/udp/Makefile"
+                                                                                                                                                                          ac_config_files="$ac_config_files Makefile pkgdata/Makefile pkgdata/pkg_data_src.gpt builtins/Makefile doxygen/Makefile doxygen/Doxyfile doxygen/Doxyfile-internal version.h extensions.h builtins/file/Makefile builtins/http/Makefile builtins/mode_e/Makefile builtins/ordering/Makefile builtins/queue/Makefile builtins/tcp/Makefile builtins/telnet/Makefile builtins/udp/Makefile builtins/gsi/Makefile builtins/popen/Makefile"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
 # tests run on this system so they can be shared between configure
@@ -20212,6 +20212,8 @@
   "builtins/tcp/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/tcp/Makefile" ;;
   "builtins/telnet/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/telnet/Makefile" ;;
   "builtins/udp/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/udp/Makefile" ;;
+  "builtins/gsi/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/gsi/Makefile" ;;
+  "builtins/popen/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/popen/Makefile" ;;
   "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
   "config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
diff -u -r gt5.2.5-all-source-installer/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in gt5.2.5-all-source-installer-patched/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in
--- gt5.2.5-all-source-installer/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	2013-10-10 12:29:15.000000000 -0500
+++ gt5.2.5-all-source-installer-patched/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	2014-01-23 11:57:53.000000000 -0600
@@ -8,6 +8,21 @@
     <src_pkg>
         <With_Flavors build="yes"/>
         <Source_Dependencies Type="compile">
+            <Dependency Name="globus_gss_assist">
+                <Version>
+                    <Simple_Version Major="9"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_error">
+                <Version>
+                    <Simple_Version Major="4"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_gsi">
+                <Version>
+                    <Simple_Version Major="10"/>
+                </Version>
+            </Dependency>
             <Dependency Name="globus_common">
                 <Version>
                     <Simple_Version Major="14"/>
@@ -20,6 +35,21 @@
             </Dependency>
         </Source_Dependencies>
         <Source_Dependencies Type="lib_link">
+            <Dependency Name="globus_gss_assist">
+                <Version>
+                    <Simple_Version Major="9"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_error">
+                <Version>
+                    <Simple_Version Major="4"/>
+                </Version>
+            </Dependency>
+            <Dependency Name="globus_gssapi_gsi">
+                <Version>
+                    <Simple_Version Major="10"/>
+                </Version>
+            </Dependency>
             <Dependency Name="globus_common">
                 <Version>
                     <Simple_Version Major="14"/>
