diff --git a/samples/condor/Makefile b/samples/condor/Makefile
index 19e66d7..50d2849 100644
--- a/samples/condor/Makefile
+++ b/samples/condor/Makefile
@@ -7,7 +7,7 @@ distclean: clean
 
 distclean-recursive: clean
 
-boinc_gahp: boinc_gahp.cpp ../../lib/remote_submit.h ../../lib/remote_submit.cpp ../../svn_version.h
+boinc_gahp: boinc_gahp.cpp ../../lib/remote_submit.h ../../lib/remote_submit.cpp ../../svn_version.h ../../version.h
 	g++ -g -O0 -I../../lib -I../.. \
 	-o boinc_gahp boinc_gahp.cpp ../../lib/remote_submit.cpp \
 	-L../../lib -lboinc -lpthread -lcurl
diff --git a/samples/condor/boinc_gahp.cpp b/samples/condor/boinc_gahp.cpp
index 970ec31..4fe1758 100644
--- a/samples/condor/boinc_gahp.cpp
+++ b/samples/condor/boinc_gahp.cpp
@@ -35,8 +35,9 @@
 #include "parse.h"
 #include "remote_submit.h"
 #include "svn_version.h"
+#include "version.h"
 
-#define BOINC_GAHP_VERSION "1.0.1"
+#define BOINC_GAHP_VERSION "1.0.2"
 
 using std::map;
 using std::pair;
@@ -723,7 +724,7 @@ int COMMAND::parse_command() {
 }
 
 void print_version(bool startup) {
-    BPRINTF("%s$GahpVersion: %s %s BOINC\\ GAHP\\ GIT:%x $\n", startup ? "" : "S ", BOINC_GAHP_VERSION, __DATE__, GIT_REVISION);
+    BPRINTF("%s$GahpVersion: %s %s BOINC\\ %s\\ GAHP $\n", startup ? "" : "S ", BOINC_GAHP_VERSION, __DATE__, BOINC_VERSION_STRING);
 }
 
 int n_results() {
