diff -u -r openssl-0.9.8h/ssl/s3_lib.c openssl-0.9.8h.patched/ssl/s3_lib.c
--- openssl-0.9.8h/ssl/s3_lib.c	2008-04-30 11:11:32.000000000 -0500
+++ openssl-0.9.8h.patched/ssl/s3_lib.c	2009-12-11 10:57:50.000000000 -0600
@@ -2592,6 +2592,9 @@
 	if (s->s3->flags & SSL3_FLAGS_NO_RENEGOTIATE_CIPHERS)
 		return(0);
 
+	if (!(s->s3->flags & SSL3_FLAGS_ALLOW_UNSAFE_LEGACY_RENEGOTIATION))
+		return(0);
+
 	s->s3->renegotiate=1;
 	return(1);
 	}
diff -u -r openssl-0.9.8h/ssl/s3_pkt.c openssl-0.9.8h.patched/ssl/s3_pkt.c
--- openssl-0.9.8h/ssl/s3_pkt.c	2006-11-29 08:45:14.000000000 -0600
+++ openssl-0.9.8h.patched/ssl/s3_pkt.c	2009-12-11 11:00:32.000000000 -0600
@@ -978,6 +978,7 @@
 
 		if (SSL_is_init_finished(s) &&
 			!(s->s3->flags & SSL3_FLAGS_NO_RENEGOTIATE_CIPHERS) &&
+			(s->s3->flags & SSL3_FLAGS_ALLOW_UNSAFE_LEGACY_RENEGOTIATION) &&
 			!s->s3->renegotiate)
 			{
 			ssl3_renegotiate(s);
@@ -1110,7 +1111,8 @@
 	if ((s->s3->handshake_fragment_len >= 4) &&	!s->in_handshake)
 		{
 		if (((s->state&SSL_ST_MASK) == SSL_ST_OK) &&
-			!(s->s3->flags & SSL3_FLAGS_NO_RENEGOTIATE_CIPHERS))
+			!(s->s3->flags & SSL3_FLAGS_NO_RENEGOTIATE_CIPHERS) &&
+			(s->s3->flags & SSL3_FLAGS_ALLOW_UNSAFE_LEGACY_RENEGOTIATION))
 			{
 #if 0 /* worked only because C operator preferences are not as expected (and
        * because this is not really needed for clients except for detecting
diff -u -r openssl-0.9.8h/ssl/s3_srvr.c openssl-0.9.8h.patched/ssl/s3_srvr.c
--- openssl-0.9.8h/ssl/s3_srvr.c	2008-04-30 11:11:32.000000000 -0500
+++ openssl-0.9.8h.patched/ssl/s3_srvr.c	2009-12-11 11:01:58.000000000 -0600
@@ -718,6 +718,14 @@
 #endif
 	STACK_OF(SSL_CIPHER) *ciphers=NULL;
 
+	if (s->new_session
+	    && !(s->s3->flags&SSL3_FLAGS_ALLOW_UNSAFE_LEGACY_RENEGOTIATION))
+		{
+		al=SSL_AD_HANDSHAKE_FAILURE;
+		SSLerr(SSL_F_SSL3_GET_CLIENT_HELLO, ERR_R_INTERNAL_ERROR);
+		goto f_err;
+		}
+
 	/* We do this so that we will respond with our native type.
 	 * If we are TLSv1 and we get SSLv3, we will respond with TLSv1,
 	 * This down switching should be handled by a different method.
diff -u -r openssl-0.9.8h/ssl/ssl.h openssl-0.9.8h.patched/ssl/ssl.h
--- openssl-0.9.8h/ssl/ssl.h	2008-04-30 11:11:32.000000000 -0500
+++ openssl-0.9.8h.patched/ssl/ssl.h	2009-12-11 11:03:01.000000000 -0600
@@ -1942,6 +1942,7 @@
 #define SSL_R_NO_PRIVATE_KEY_ASSIGNED			 190
 #define SSL_R_NO_PROTOCOLS_AVAILABLE			 191
 #define SSL_R_NO_PUBLICKEY				 192
+#define SSL_R_NO_RENEGOTIATION				 318
 #define SSL_R_NO_SHARED_CIPHER				 193
 #define SSL_R_NO_VERIFY_CALLBACK			 194
 #define SSL_R_NULL_SSL_CTX				 195
diff -u -r openssl-0.9.8h/ssl/ssl3.h openssl-0.9.8h.patched/ssl/ssl3.h
--- openssl-0.9.8h/ssl/ssl3.h	2007-10-11 19:00:30.000000000 -0500
+++ openssl-0.9.8h.patched/ssl/ssl3.h	2009-12-11 11:03:40.000000000 -0600
@@ -330,6 +330,7 @@
 #define SSL3_FLAGS_DELAY_CLIENT_FINISHED	0x0002
 #define SSL3_FLAGS_POP_BUFFER			0x0004
 #define TLS1_FLAGS_TLS_PADDING_BUG		0x0008
+#define SSL3_FLAGS_ALLOW_UNSAFE_LEGACY_RENEGOTIATION	0x0010
 
 typedef struct ssl3_state_st
 	{
diff -u -r openssl-0.9.8h/ssl/ssl_err.c openssl-0.9.8h.patched/ssl/ssl_err.c
--- openssl-0.9.8h/ssl/ssl_err.c	2007-10-11 19:00:30.000000000 -0500
+++ openssl-0.9.8h.patched/ssl/ssl_err.c	2009-12-11 11:04:31.000000000 -0600
@@ -381,6 +381,7 @@
 {ERR_REASON(SSL_R_NO_PRIVATE_KEY_ASSIGNED),"no private key assigned"},
 {ERR_REASON(SSL_R_NO_PROTOCOLS_AVAILABLE),"no protocols available"},
 {ERR_REASON(SSL_R_NO_PUBLICKEY)          ,"no publickey"},
+{ERR_REASON(SSL_R_NO_RENEGOTIATION)      ,"no renegotiation"},
 {ERR_REASON(SSL_R_NO_SHARED_CIPHER)      ,"no shared cipher"},
 {ERR_REASON(SSL_R_NO_VERIFY_CALLBACK)    ,"no verify callback"},
 {ERR_REASON(SSL_R_NULL_SSL_CTX)          ,"null ssl ctx"},
