--- Makefile.vc9	Wed Aug 12 04:00:34 2009
+++ Makefile.vc9	Mon Dec 20 14:04:11 2010
@@ -91,7 +91,7 @@
 
 CCNODBG    = cl.exe /O2 /DNDEBUG
 CCDEBUG    = cl.exe /Od /Gm /Zi /D_DEBUG /RTC1
-CFLAGSSSL  = /DUSE_SSLEAY /I "$(OPENSSL_PATH)/inc32" /I "$(OPENSSL_PATH)/inc32/openssl"
+CFLAGSSSL  = /DUSE_SSLEAY /I "$(OPENSSL_PATH)/include" /I "$(OPENSSL_PATH)/include/openssl"
 CFLAGSZLIB = /DHAVE_ZLIB_H /DHAVE_ZLIB /DHAVE_LIBZ /I "$(ZLIB_PATH)"
 CFLAGS     = /I. /I../include /nologo /W3 /EHsc /DWIN32 /FD /c /DBUILDING_LIBCURL
 CFLAGSLIB  = /DCURL_STATICLIB
@@ -163,7 +163,7 @@
 !IF "$(CFG)" == "release-ssl"
 TARGET   = $(LIB_NAME).lib
 DIROBJ   = $(CFG)
-LFLAGSSSL = "/LIBPATH:$(OPENSSL_PATH)\out32"
+LFLAGSSSL = "/LIBPATH:$(OPENSSL_PATH)\lib"
 LNK      = $(LNKLIB) $(LFLAGSSSL) /out:$(DIROBJ)\$(TARGET)
 CC       = $(CCNODBG) $(RTLIB) $(CFLAGSSSL) $(CFLAGSLIB)
 CFGSET   = TRUE
