diff -u -r globus_toolkit-6.0.1441801527/gsi_openssh/source/Makefile.in globus_toolkit-6.0.1441801527.patched/gsi_openssh/source/Makefile.in
--- globus_toolkit-6.0.1441801527/gsi_openssh/source/Makefile.in	2015-09-09 07:55:36.777535165 -0500
+++ globus_toolkit-6.0.1441801527.patched/gsi_openssh/source/Makefile.in	2016-01-26 11:36:25.395740786 -0600
@@ -1,7 +1,7 @@
 # $Id$
 
 # uncomment if you run a non bourne compatable shell. Ie. csh
-#SHELL = @SH@
+SHELL = @SHELL@
 
 DISTPERL=perl
 DISTSED=sed
