diff -u -r globus_toolkit-6.0.1506371041/gass/copy/source/configure globus_toolkit-6.0.1506371041-patched/gass/copy/source/configure
--- globus_toolkit-6.0.1506371041/gass/copy/source/configure	2018-05-03 08:35:03.000000000 -0500
+++ globus_toolkit-6.0.1506371041-patched/gass/copy/source/configure	2018-05-03 08:36:06.000000000 -0500
@@ -8,6 +8,9 @@
 # Copyright (C) 1992-1996, 1998-2012 Free Software Foundation, Inc.
 #
 #
+if test x"${LTDLINCL}" != x; then
+    CPPFLAGS="${CPPFLAGS} ${LTDLINCL}"
+fi
 # This configure script is free software; the Free Software Foundation
 # gives unlimited permission to copy, distribute and modify it.
 ## -------------------- ##
diff -u -r globus_toolkit-6.0.1506371041/gridftp/server/src/configure globus_toolkit-6.0.1506371041-patched/gridftp/server/src/configure
--- globus_toolkit-6.0.1506371041/gridftp/server/src/configure	2018-05-03 08:35:29.000000000 -0500
+++ globus_toolkit-6.0.1506371041-patched/gridftp/server/src/configure	2018-05-03 08:36:53.000000000 -0500
@@ -8,6 +8,9 @@
 # Copyright (C) 1992-1996, 1998-2012 Free Software Foundation, Inc.
 #
 #
+if test x"${LTDLINCL}" != x; then
+    CPPFLAGS="${CPPFLAGS} ${LTDLINCL}"
+fi
 # This configure script is free software; the Free Software Foundation
 # gives unlimited permission to copy, distribute and modify it.
 ## -------------------- ##
