diff -u -r globus_toolkit-6.0.1506371041/common/source/library/globus_libc.h globus_toolkit-6.0.1506371041.patched/common/source/library/globus_libc.h
--- globus_toolkit-6.0.1506371041/common/source/library/globus_libc.h	2017-09-25 15:35:56.000000000 -0500
+++ globus_toolkit-6.0.1506371041.patched/common/source/library/globus_libc.h	2018-09-19 15:14:03.000000000 -0500
@@ -203,13 +203,13 @@
     const char *                            s2,
     globus_size_t                           n);
 
-int globus_libc_setenv(register const char *name,
-		       register const char *value,
+int globus_libc_setenv(const char *name,
+		       const char *value,
 		       int rewrite);
-void globus_libc_unsetenv(register const char *name);
+void globus_libc_unsetenv(const char *name);
 
 /* Use getenv instead */
-GLOBUS_DEPRECATED(char *globus_libc_getenv(register const char *name));
+GLOBUS_DEPRECATED(char *globus_libc_getenv(const char *name));
 
 /* Use strerror or strerror_r as needed instead */
 char *globus_libc_system_error_string(int the_error);
diff -u -r globus_toolkit-6.0.1506371041/gsi_openssh/source/openbsd-compat/openbsd-compat.h globus_toolkit-6.0.1506371041.patched/gsi_openssh/source/openbsd-compat/openbsd-compat.h
--- globus_toolkit-6.0.1506371041/gsi_openssh/source/openbsd-compat/openbsd-compat.h	2017-03-19 21:39:27.000000000 -0500
+++ globus_toolkit-6.0.1506371041.patched/gsi_openssh/source/openbsd-compat/openbsd-compat.h	2018-09-19 15:12:59.000000000 -0500
@@ -97,7 +97,7 @@
 #endif
 
 #ifndef HAVE_SETENV
-int setenv(register const char *name, register const char *value, int rewrite);
+int setenv(const char *name, const char *value, int rewrite);
 #endif
 
 #ifndef HAVE_STRMODE
