--- voms-1.8.9/src/socklib/Makefile.am	2008-12-22 11:51:08.000000000 -0600
+++ voms-1.8.9-patched/src/socklib/Makefile.am	2009-02-05 17:44:02.000000000 -0600
@@ -107,7 +107,7 @@
 libsock_server_gcc32dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC32DBG_CFLAGS)
+	$(GLOBUS_GCC32DBGPTHR_CFLAGS)
 
 libsock_server_gcc32dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc32dbgpthr.la \
@@ -159,7 +159,7 @@
 libsock_server_gcc64dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC64DBG_CFLAGS)
+	$(GLOBUS_GCC64DBGPTHR_CFLAGS)
 
 libsock_server_gcc64dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc64dbgpthr.la \
@@ -234,7 +234,7 @@
 libsock_gcc32dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC32DBG_CFLAGS)
+	$(GLOBUS_GCC32DBGPTHR_CFLAGS)
 
 libsock_gcc32dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc32dbgpthr.la \
@@ -286,7 +286,7 @@
 libsock_gcc64dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC64DBG_CFLAGS)
+	$(GLOBUS_GCC64DBGPTHR_CFLAGS)
 
 libsock_gcc64dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc64dbgpthr.la \
--- voms-1.8.9/src/socklib/Makefile.in	2008-12-22 11:51:08.000000000 -0600
+++ voms-1.8.9-patched/src/socklib/Makefile.in	2009-02-05 17:45:12.000000000 -0600
@@ -537,7 +537,7 @@
 libsock_server_gcc32dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC32DBG_CFLAGS)
+	$(GLOBUS_GCC32DBGPTHR_CFLAGS)
 
 libsock_server_gcc32dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc32dbgpthr.la \
@@ -589,7 +589,7 @@
 libsock_server_gcc64dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC64DBG_CFLAGS)
+	$(GLOBUS_GCC64DBGPTHR_CFLAGS)
 
 libsock_server_gcc64dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc64dbgpthr.la \
@@ -663,7 +663,7 @@
 libsock_gcc32dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC32DBG_CFLAGS)
+	$(GLOBUS_GCC32DBGPTHR_CFLAGS)
 
 libsock_gcc32dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc32dbgpthr.la \
@@ -715,7 +715,7 @@
 libsock_gcc64dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC64DBG_CFLAGS)
+	$(GLOBUS_GCC64DBGPTHR_CFLAGS)
 
 libsock_gcc64dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc64dbgpthr.la \
