diff -u -r org.glite.ce.blahp/src/blparser_master.c org.glite.ce.blahp.patched/src/blparser_master.c
--- org.glite.ce.blahp/src/blparser_master.c	2010-03-17 05:48:17.000000000 -0500
+++ org.glite.ce.blahp.patched/src/blparser_master.c	2010-05-26 17:00:47.000000000 -0500
@@ -34,7 +34,11 @@
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <syslog.h>
+
+#ifdef MTRACE_ON
 #include <mcheck.h>
+#endif
+
 #include <pthread.h>
 #include <wordexp.h>
 #include <signal.h>
diff -u -r org.glite.ce.blahp/src/main.c org.glite.ce.blahp.patched/src/main.c
--- org.glite.ce.blahp/src/main.c	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/main.c	2010-05-26 17:01:03.000000000 -0500
@@ -43,7 +43,10 @@
 #include <sys/select.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+
+#ifdef MTRACE_ON
 #include <mcheck.h>
+#endif
 
 #include "blahpd.h"
 #include "server.h"
diff -u -r org.glite.ce.blahp/src/main_daemon.c org.glite.ce.blahp.patched/src/main_daemon.c
--- org.glite.ce.blahp/src/main_daemon.c	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/main_daemon.c	2010-05-26 17:01:20.000000000 -0500
@@ -43,7 +43,10 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <syslog.h>
+
+#ifdef MTRACE_ON
 #include <mcheck.h>
+#endif
 
 #include "blahpd.h"
 #include "server.h"
