--- glibc-2.12-2-x86_64/nscd/nscd_helper.c	2010-05-04 06:27:23.000000000 -0500
+++ glibc-2.12-2-x86_64/nscd/nscd_helper.c.patched	2012-05-07 12:53:44.000000000 -0500
@@ -165,6 +165,14 @@
 {
   int sock;
 
+  /* Patched by jfrey on 05/07/12 (based on an earlier patch by psilord)
+   * to turn off connecting to the nscd via a unix domain socket, which
+   * in some cases is left open permanently. This open socket defers
+   * checkpointing forever.
+   */
+  errno = ENOENT;
+  return -1;
+
 #ifdef SOCK_CLOEXEC
 # ifndef __ASSUME_SOCK_CLOEXEC
   if (__have_sock_cloexec >= 0)
