--- configure.ac	2016-06-30 15:03:43.090035173 -0500
+++ configure.ac.patched	2016-06-30 15:03:45.565057886 -0500
@@ -33,10 +33,10 @@
 
 PKG_CHECK_MODULES([OPENSSL], [openssl])
 
-PKG_CHECK_MODULES([GSOAP],[gsoap >= 2.7])
-PKG_CHECK_MODULES([GSOAP_PP],[gsoap++ >= 2.7])
-PKG_CHECK_MODULES([GSOAP_SSL],[gsoapssl >= 2.7])
-PKG_CHECK_MODULES([GSOAP_SSL_PP],[gsoapssl++ >= 2.7])
+#PKG_CHECK_MODULES([GSOAP],[gsoap >= 2.7])
+#PKG_CHECK_MODULES([GSOAP_PP],[gsoap++ >= 2.7])
+#PKG_CHECK_MODULES([GSOAP_SSL],[gsoapssl >= 2.7])
+#PKG_CHECK_MODULES([GSOAP_SSL_PP],[gsoapssl++ >= 2.7])
 
 AC_CHECK_HEADER([expat.h], 
     [], 
@@ -51,7 +51,7 @@
 
 AC_SUBST(EXPAT_LIBS)
 
-AC_WSDL2H
+#AC_WSDL2H
 AC_ENABLE_DOCS
 
 # Checks for header files.
