diff -u -r gt4.2.1-all-source-installer/source-trees/xio/src/configure gt4.2.1-all-source-installer.patched/source-trees/xio/src/configure
--- gt4.2.1-all-source-installer/source-trees/xio/src/configure	2008-10-03 00:13:09.000000000 -0500
+++ gt4.2.1-all-source-installer.patched/source-trees/xio/src/configure	2009-01-20 14:21:50.000000000 -0600
@@ -3672,7 +3672,7 @@
 
 
 
-                                                                                                                                                                          ac_config_files="$ac_config_files Makefile pkgdata/Makefile pkgdata/pkg_data_src.gpt builtins/Makefile doxygen/Makefile doxygen/Doxyfile doxygen/Doxyfile-internal version.h extensions.h builtins/file/Makefile builtins/http/Makefile builtins/mode_e/Makefile builtins/ordering/Makefile builtins/queue/Makefile builtins/tcp/Makefile builtins/telnet/Makefile builtins/udp/Makefile"
+                                                                                                                                                                          ac_config_files="$ac_config_files Makefile pkgdata/Makefile pkgdata/pkg_data_src.gpt builtins/Makefile doxygen/Makefile doxygen/Doxyfile doxygen/Doxyfile-internal version.h extensions.h builtins/file/Makefile builtins/http/Makefile builtins/mode_e/Makefile builtins/ordering/Makefile builtins/queue/Makefile builtins/tcp/Makefile builtins/telnet/Makefile builtins/udp/Makefile builtins/gsi/Makefile builtins/popen/Makefile"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
 # tests run on this system so they can be shared between configure
@@ -4260,6 +4260,8 @@
   "builtins/tcp/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/tcp/Makefile" ;;
   "builtins/telnet/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/telnet/Makefile" ;;
   "builtins/udp/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/udp/Makefile" ;;
+  "builtins/gsi/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/gsi/Makefile" ;;
+  "builtins/popen/Makefile" ) CONFIG_FILES="$CONFIG_FILES builtins/popen/Makefile" ;;
   "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
   "config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
diff -u -r gt4.2.1-all-source-installer/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in gt4.2.1-all-source-installer.patched/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in
--- gt4.2.1-all-source-installer/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	2008-04-30 13:25:28.000000000 -0500
+++ gt4.2.1-all-source-installer.patched/source-trees/xio/src/pkgdata/pkg_data_src.gpt.in	2009-01-20 14:25:22.000000000 -0600
@@ -15,10 +15,28 @@
     <Source_Dependencies Type="compile" >
       <Dependency Name="globus_common" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
       <Dependency Name="globus_core" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
+      <Dependency Name="globus_gss_assist" >
+        <Version><Simple_Version Major="3"/></Version>
+      </Dependency>
+      <Dependency Name="gssapi_error">
+        <Version ><Simple_Version Major="2"/></Version>
+      </Dependency>
+      <Dependency Name="globus_gssapi_gsi">
+        <Version ><Simple_Version Major="4"/></Version>
+      </Dependency>
     </Source_Dependencies>
 
     <Source_Dependencies Type="lib_link" >
       <Dependency Name="globus_common" > <Version > <Simple_Version Major="4" /> </Version> </Dependency>
+      <Dependency Name="globus_gss_assist" >
+        <Version><Simple_Version Major="3"/></Version>
+      </Dependency>
+      <Dependency Name="gssapi_error">
+        <Version ><Simple_Version Major="2"/></Version>
+      </Dependency>
+      <Dependency Name="globus_gssapi_gsi">
+        <Version ><Simple_Version Major="4"/></Version>
+      </Dependency>
     </Source_Dependencies>
 
     <Build_Environment >
