diff -u -r src/canl_cred.c src/canl_cred.c-patched
--- src/canl_cred.c	2016-11-29 11:05:14.000000000 -0600
+++ src/canl_cred.c-patched	2018-10-15 11:43:44.458762165 -0500
@@ -145,7 +145,11 @@
 
 void pkey_dup(EVP_PKEY **to, EVP_PKEY *from)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
     EVP_PKEY_up_ref(from);
+#else
+    CRYPTO_add(&from->references,1,CRYPTO_LOCK_EVP_PKEY);
+#endif
     *to = from;
 }
 
diff -u -r src/canl_ocsp.c src/canl_ocsp.c-patched
--- src/canl_ocsp.c	2016-11-29 11:05:14.000000000 -0600
+++ src/canl_ocsp.c-patched	2018-10-15 12:06:39.990170967 -0500
@@ -354,7 +354,11 @@
 
     if (ssl){
         BIO *sbio;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
         ctx_in = SSL_CTX_new(TLS_client_method());
+#else
+	ctx_in = SSL_CTX_new(SSLv3_client_method());
+#endif
         if (ctx_in == NULL) {
             goto end;
         }
diff -u -r src/proxy/sslutils.c src/proxy/sslutils.c-patched
--- src/proxy/sslutils.c	2016-11-29 11:05:15.000000000 -0600
+++ src/proxy/sslutils.c-patched	2018-10-15 12:05:57.713756583 -0500
@@ -1224,9 +1224,15 @@
     /* transfer the public key from req to new cert */
     /* DEE? should this be a dup? */
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
     new_public_key = X509_REQ_get0_pubkey(req);
     X509_set_pubkey(*new_cert, new_public_key);
 //    req->req_info->pubkey = NULL;
+#else
+    X509_PUBKEY_free((*new_cert)->cert_info->key);
+    (*new_cert)->cert_info->key = req->req_info->pubkey;
+    req->req_info->pubkey = NULL;
+#endif
 
     /*
      * We can now add additional extentions here
@@ -4330,8 +4336,13 @@
                 goto failure;
             }
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
             ex_pcpathlen = X509_get_proxy_pathlen(cert);
             ex_pathlen = X509_get_pathlen(cert);
+#else
+	    ex_pcpathlen = cert->ex_pcpathlen;
+	    ex_pathlen = cert->ex_pathlen;
+#endif
 
             /* Log (L_DEBUG, "\tCert here is: %s\n", cert_subjectdn); */
             curr_cert_type = lcmaps_type_of_proxy(cert);
