diff -u -r src/grst_canl_x509.c src/grst_canl_x509.c-patched
--- src/grst_canl_x509.c	2017-02-25 05:29:58.000000000 -0600
+++ src/grst_canl_x509.c-patched	2018-10-15 11:35:50.117258598 -0500
@@ -209,7 +209,11 @@
 static void
 ssl_init_crypto(void)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
     OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_DIGESTS, NULL);
+#else
+    OpenSSL_add_all_digests();
+#endif
 }
 
 /* Safely initialize OpenSSL digests */
@@ -353,7 +357,11 @@
    prvkey = X509_extract_key(cert);
    if (prvkey == NULL) return GRST_RET_FAILED;
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
    ctx = EVP_MD_CTX_new();
+#else
+   ctx = EVP_MD_CTX_create();
+#endif
    if (ctx == NULL)
        return GRST_RET_FAILED;
             
@@ -369,8 +377,10 @@
 
    ret = EVP_VerifyFinal(ctx, sig, sig_len, prvkey);
 
-#if OPENSSL_VERSION_NUMBER >= 0x0090701fL
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
    EVP_MD_CTX_free(ctx);
+#elif OPENSSL_VERSION_NUMBER >= 0x0090701fL
+   EVP_MD_CTX_destroy(ctx);
 #endif
    EVP_PKEY_free(prvkey);
 
@@ -2653,7 +2663,11 @@
 
   GRSTx509SafeOpenSSLInitialization();
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
   ctx = EVP_MD_CTX_new();
+#else
+  ctx = EVP_MD_CTX_create();
+#endif
   if (ctx == NULL)
     return NULL;
 
@@ -2679,7 +2693,11 @@
 
   delegation_id[16] = '\0';
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
   EVP_MD_CTX_free(ctx);
+#else
+  EVP_MD_CTX_destroy(ctx);
+#endif
 
   return delegation_id;
 }
