--- mk1mf.pl.old	2008-12-02 17:50:21.000000000 -0600
+++ mk1mf.pl	2009-11-18 18:44:21.341435553 -0600
@@ -243,6 +243,7 @@
 $cflags.=" -DOPENSSL_NO_ENGINE"   if $no_engine;
 $cflags.=" -DOPENSSL_NO_HW"   if $no_hw;
 $cflags.=" -DOPENSSL_FIPS"    if $fips;
+$cflags.=" -D_CRT_SECURE_NO_DEPRECATE"   if ($platform eq "VC-NT");
 $cflags.= " -DZLIB" if $zlib_opt;
 $cflags.= " -DZLIB_SHARED" if $zlib_opt == 2;
 
