--- a/src/condor_contrib/campus_factory/bin/campus_factory
+++ b/src/condor_contrib/campus_factory/bin/campus_factory
@@ -19,7 +19,7 @@
 
 if [ -e $script_location/../etc/campus_factory.conf ]; then
     config=$script_location/../etc/campus_factory.conf
-elif [ -e `condor_config_val ETC`/campus_factory.conf 
+elif [ -e `condor_config_val ETC`/campus_factory.conf ]; then
 	config=`condor_config_val ETC`/campus_factory.conf
 fi
 
