diff --git a/externals/bundles/blahp/1.16.5.1/pbs_status.py b/externals/bundles/blahp/1.16.5.1/pbs_status.py
index 3ec22085a4..5fa3925fbf 100755
--- a/externals/bundles/blahp/1.16.5.1/pbs_status.py
+++ b/externals/bundles/blahp/1.16.5.1/pbs_status.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 #  File:     pbs_status.py
 #
diff --git a/nmi_tools/condor_nmi_extract b/nmi_tools/condor_nmi_extract
index 3fc95ae33f..d459c2691b 100755
--- a/nmi_tools/condor_nmi_extract
+++ b/nmi_tools/condor_nmi_extract
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 import os
 import re
 import sys
diff --git a/nmi_tools/glue/build/email_on_failure.py b/nmi_tools/glue/build/email_on_failure.py
index 5fd8a7377e..3c2d62099c 100755
--- a/nmi_tools/glue/build/email_on_failure.py
+++ b/nmi_tools/glue/build/email_on_failure.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # The purpose of this script is to email users when an automated build breaks.
 # Caveats:
diff --git a/src/annex/setup/upgrade-runtime.py b/src/annex/setup/upgrade-runtime.py
index aa80b632af..71de2da81a 100755
--- a/src/annex/setup/upgrade-runtime.py
+++ b/src/annex/setup/upgrade-runtime.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python2
+#!/usr/bin/python3
 
 from __future__ import print_function
 
diff --git a/src/azure_gahp/AzureGAHPServer.py b/src/azure_gahp/AzureGAHPServer.py
index ec89591c38..95ec0c48d2 100644
--- a/src/azure_gahp/AzureGAHPServer.py
+++ b/src/azure_gahp/AzureGAHPServer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python -u
+#!/usr/bin/env python3 -u
 
 from __future__ import print_function
 
diff --git a/src/condor_contrib/aviary/test/collector_tool.py b/src/condor_contrib/aviary/test/collector_tool.py
index 1dfa9768aa..59ff5cb3c6 100755
--- a/src/condor_contrib/aviary/test/collector_tool.py
+++ b/src/condor_contrib/aviary/test/collector_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/dag/dag-submit.py b/src/condor_contrib/aviary/test/dag/dag-submit.py
index ffb2683568..1d4e14a70c 100755
--- a/src/condor_contrib/aviary/test/dag/dag-submit.py
+++ b/src/condor_contrib/aviary/test/dag/dag-submit.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/hadoop_tool.py b/src/condor_contrib/aviary/test/hadoop_tool.py
index 7f955600be..57619462bb 100755
--- a/src/condor_contrib/aviary/test/hadoop_tool.py
+++ b/src/condor_contrib/aviary/test/hadoop_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/jobcontrol.py b/src/condor_contrib/aviary/test/jobcontrol.py
index 7b35146573..17db61510f 100755
--- a/src/condor_contrib/aviary/test/jobcontrol.py
+++ b/src/condor_contrib/aviary/test/jobcontrol.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/jobinventory.py b/src/condor_contrib/aviary/test/jobinventory.py
index 0411f988f7..20d480185d 100755
--- a/src/condor_contrib/aviary/test/jobinventory.py
+++ b/src/condor_contrib/aviary/test/jobinventory.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/jobquery.py b/src/condor_contrib/aviary/test/jobquery.py
index 1f538881ae..03591bca98 100755
--- a/src/condor_contrib/aviary/test/jobquery.py
+++ b/src/condor_contrib/aviary/test/jobquery.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/locator.py b/src/condor_contrib/aviary/test/locator.py
index 19b66e5be9..7f60ee7ed0 100755
--- a/src/condor_contrib/aviary/test/locator.py
+++ b/src/condor_contrib/aviary/test/locator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/setattr.py b/src/condor_contrib/aviary/test/setattr.py
index a4e8356bf1..e665af0a57 100755
--- a/src/condor_contrib/aviary/test/setattr.py
+++ b/src/condor_contrib/aviary/test/setattr.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/subinventory.py b/src/condor_contrib/aviary/test/subinventory.py
index 6f858eb56f..4fe60cfd37 100755
--- a/src/condor_contrib/aviary/test/subinventory.py
+++ b/src/condor_contrib/aviary/test/subinventory.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2013 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/submission_ids.py b/src/condor_contrib/aviary/test/submission_ids.py
index f512cae4ed..e78a87bf19 100755
--- a/src/condor_contrib/aviary/test/submission_ids.py
+++ b/src/condor_contrib/aviary/test/submission_ids.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/submissions.py b/src/condor_contrib/aviary/test/submissions.py
index 0ee7d9fbf5..be88d99bd8 100755
--- a/src/condor_contrib/aviary/test/submissions.py
+++ b/src/condor_contrib/aviary/test/submissions.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/aviary/test/submit.py b/src/condor_contrib/aviary/test/submit.py
index 30f84c522a..0ede8b9af1 100755
--- a/src/condor_contrib/aviary/test/submit.py
+++ b/src/condor_contrib/aviary/test/submit.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/bosco/bosco_cluster b/src/condor_contrib/bosco/bosco_cluster
index 82a040132e..ef4a54ba79 100755
--- a/src/condor_contrib/bosco/bosco_cluster
+++ b/src/condor_contrib/bosco/bosco_cluster
@@ -246,14 +246,14 @@ ssh_find_remote () {
     remote_host=$1
     findplatform_path=$2
     # if .. else .. only from Python 2.5, no RH5 support
-    #ssh uc3-sub /usr/bin/python -c \"import platform\; mydist = platform.dist\(\)\; print \'DEB6\' if mydist[0]==\'debian\' and mydist[1].startswith\(\'6.\'\) else \(\'RH5\' if mydist[1].startswith\(\'5.\'\) else \'RH6\' if mydist[1].startswith\(\'6.\'\) else \'UNKNOWN\'\) if mydist[0]==\'redhat\'  else \'UNKNOWN\'\"
-    cmd_out=`ssh $remote_host "python -c \"import sys; print sys.platform\"" 2>/dev/null`
+    #ssh uc3-sub /usr/bin/python3 -c \"import platform\; mydist = platform.dist\(\)\; print \'DEB6\' if mydist[0]==\'debian\' and mydist[1].startswith\(\'6.\'\) else \(\'RH5\' if mydist[1].startswith\(\'5.\'\) else \'RH6\' if mydist[1].startswith\(\'6.\'\) else \'UNKNOWN\'\) if mydist[0]==\'redhat\'  else \'UNKNOWN\'\"
+    cmd_out=`ssh $remote_host "python3 -c \"import sys; print sys.platform\"" 2>/dev/null`
     if [ $? -eq 0 -a "x$cmd_out" == "xdarwin" ]; then
         # assume all Mac OK
         echo "`$findplatform_path --url --force MAC`"
         return 0
     fi
-    cmd_out=`ssh $remote_host "python -c \"import platform; mydist = platform.dist(); print '%s%s' % (mydist[0], mydist[1])\"" 2>/dev/null`
+    cmd_out=`ssh $remote_host "python3 -c \"import platform; mydist = platform.dist(); print '%s%s' % (mydist[0], mydist[1])\"" 2>/dev/null`
     if [ $? -eq 0 ]; then
         # check for linux
         case "$cmd_out" in 
@@ -321,7 +321,7 @@ check_reverse_connectivity() {
     # Create tmp file for the python script to test connectivity
     tmpfile=`mktemp -t tmpreversetest.XXXXXX`
     cat << EOF > $tmpfile
-#!/usr/bin/python
+#!/usr/bin/python3
 import sys,socket
 (host,port)=sys.argv[1].split(":")
 s = socket.socket()
diff --git a/src/condor_contrib/bosco/bosco_findplatform b/src/condor_contrib/bosco/bosco_findplatform
index 62225ee435..424bb6be3d 100755
--- a/src/condor_contrib/bosco/bosco_findplatform
+++ b/src/condor_contrib/bosco/bosco_findplatform
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #Marco Mambelli - first version 10/8/2012 
 
 from __future__ import print_function
diff --git a/src/condor_contrib/bosco/bosco_ssh_start b/src/condor_contrib/bosco/bosco_ssh_start
index ead8870555..2e514393b3 100755
--- a/src/condor_contrib/bosco/bosco_ssh_start
+++ b/src/condor_contrib/bosco/bosco_ssh_start
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 from __future__ import print_function
 
diff --git a/src/condor_contrib/bosco/htsub b/src/condor_contrib/bosco/htsub
index 3edb6a9a4c..cde2f5ffb6 100755
--- a/src/condor_contrib/bosco/htsub
+++ b/src/condor_contrib/bosco/htsub
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 """
 htsub command to submit high throughput jobs via BOSCO/HTCondor:
 http://jira.opensciencegrid.org/browse/CAMPUS-54
diff --git a/src/condor_contrib/campus_factory/bin/glidein_creation b/src/condor_contrib/campus_factory/bin/glidein_creation
index 3f99a5b9a6..8067a71af9 100755
--- a/src/condor_contrib/campus_factory/bin/glidein_creation
+++ b/src/condor_contrib/campus_factory/bin/glidein_creation
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import warnings
 warnings.filterwarnings("ignore", category=DeprecationWarning)
diff --git a/src/condor_contrib/campus_factory/bin/runfactory b/src/condor_contrib/campus_factory/bin/runfactory
index a2a9c5fb0b..3af42cfbaa 100755
--- a/src/condor_contrib/campus_factory/bin/runfactory
+++ b/src/condor_contrib/campus_factory/bin/runfactory
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 from optparse import OptionParser
 
 from campus_factory.Factory import Factory
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/change_availability.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/change_availability.py
index c4212f4493..39adb4ef6e 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/change_availability.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/change_availability.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Changes your Skype online status (availability).
 #
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/chat_send_all.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/chat_send_all.py
index 08d328a976..12deb88b82 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/chat_send_all.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/chat_send_all.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import Skype4Py
 import re
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/find_chat.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/find_chat.py
index 696ed3739a..5843379ccb 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/find_chat.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/find_chat.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Displays all multi-chats and allows to open them.
 #
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/mood_to_twitter.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/mood_to_twitter.py
index 1c52255642..cd6910c413 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/mood_to_twitter.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/mood_to_twitter.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Sends your own mood messages to Twitter.
 #
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/msgSend.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/msgSend.py
index b8e311301a..76e8a383ea 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/msgSend.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/msgSend.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Starts a Skype File Transfer.
 # The open file dialog is set to the current directory.
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/sendAll.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/sendAll.py
index 8854f95159..c848b43abf 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/sendAll.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/sendAll.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Starts a Skype File Transfer.
 # The open file dialog is set to the current directory.
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_file.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_file.py
index e022b7eeed..ea819b6959 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_file.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_file.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Starts a Skype File Transfer.
 # The open file dialog is set to the current directory.
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_sms.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_sms.py
index f6e77873de..b3653e8262 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_sms.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/send_sms.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Sends a SMS with Skype.
 # Uses the Skype4Py Skype API Wrapper. See https://developer.skype.com/wiki/Skype4Py
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/show_edited_chats.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/show_edited_chats.py
index a150e7ca0a..ae199e83c5 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/show_edited_chats.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/show_edited_chats.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Shows Skype edited chat messages
 #
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/skype_api.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/skype_api.py
index 0d9414d7d7..c56f854de9 100755
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/skype_api.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/skype_api.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #
 # Some tools to use the Skype Public API under Linux.
 # Uses X11 as communication protocol. Tested with Skype version 1.4.
diff --git a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/twitter.py b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/twitter.py
index 9da7739a7d..25af549168 100644
--- a/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/twitter.py
+++ b/src/condor_contrib/condor_pigeon/src/condor_pigeon_client/skype_linux_tools/twitter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # Copyright 2007 Google Inc. All Rights Reserved.
 
diff --git a/src/condor_contrib/mgmt/qmf/test/dag/submit-dag.py b/src/condor_contrib/mgmt/qmf/test/dag/submit-dag.py
index 9cbdc734f2..5313dfad6f 100755
--- a/src/condor_contrib/mgmt/qmf/test/dag/submit-dag.py
+++ b/src/condor_contrib/mgmt/qmf/test/dag/submit-dag.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
 import os, time
diff --git a/src/condor_contrib/mgmt/qmf/test/job_server-perf.py b/src/condor_contrib/mgmt/qmf/test/job_server-perf.py
index 8ea82106c4..8e434c38cd 100755
--- a/src/condor_contrib/mgmt/qmf/test/job_server-perf.py
+++ b/src/condor_contrib/mgmt/qmf/test/job_server-perf.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2009-2011 Red Hat, Inc.
 #
diff --git a/src/condor_contrib/mgmt/qmf/test/jobcontrol.py b/src/condor_contrib/mgmt/qmf/test/jobcontrol.py
index d85b001cf1..648d0ef703 100755
--- a/src/condor_contrib/mgmt/qmf/test/jobcontrol.py
+++ b/src/condor_contrib/mgmt/qmf/test/jobcontrol.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/jobinventory.py b/src/condor_contrib/mgmt/qmf/test/jobinventory.py
index 4539b3e643..3321d52f20 100755
--- a/src/condor_contrib/mgmt/qmf/test/jobinventory.py
+++ b/src/condor_contrib/mgmt/qmf/test/jobinventory.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/jobserver.py b/src/condor_contrib/mgmt/qmf/test/jobserver.py
index 53a7ea05eb..ae67e4eda1 100755
--- a/src/condor_contrib/mgmt/qmf/test/jobserver.py
+++ b/src/condor_contrib/mgmt/qmf/test/jobserver.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/negotiator.py b/src/condor_contrib/mgmt/qmf/test/negotiator.py
index 5e7dd90c56..cc36ea5421 100755
--- a/src/condor_contrib/mgmt/qmf/test/negotiator.py
+++ b/src/condor_contrib/mgmt/qmf/test/negotiator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/qmfprobe.py b/src/condor_contrib/mgmt/qmf/test/qmfprobe.py
index ed46de22c6..b61a89687b 100755
--- a/src/condor_contrib/mgmt/qmf/test/qmfprobe.py
+++ b/src/condor_contrib/mgmt/qmf/test/qmfprobe.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2009-2011 Red Hat, Inc.
 #
diff --git a/src/condor_contrib/mgmt/qmf/test/scheduler.py b/src/condor_contrib/mgmt/qmf/test/scheduler.py
index 458e40aec2..59e5de2334 100755
--- a/src/condor_contrib/mgmt/qmf/test/scheduler.py
+++ b/src/condor_contrib/mgmt/qmf/test/scheduler.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2009-2011 Red Hat, Inc.
 #
diff --git a/src/condor_contrib/mgmt/qmf/test/setattr.py b/src/condor_contrib/mgmt/qmf/test/setattr.py
index 21d49a269b..516367e268 100755
--- a/src/condor_contrib/mgmt/qmf/test/setattr.py
+++ b/src/condor_contrib/mgmt/qmf/test/setattr.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/submissions.py b/src/condor_contrib/mgmt/qmf/test/submissions.py
index 0aa416baa7..5696ee2afb 100755
--- a/src/condor_contrib/mgmt/qmf/test/submissions.py
+++ b/src/condor_contrib/mgmt/qmf/test/submissions.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/submit-vm.py b/src/condor_contrib/mgmt/qmf/test/submit-vm.py
index 8f83185c44..a5ca599869 100755
--- a/src/condor_contrib/mgmt/qmf/test/submit-vm.py
+++ b/src/condor_contrib/mgmt/qmf/test/submit-vm.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/mgmt/qmf/test/submit.py b/src/condor_contrib/mgmt/qmf/test/submit.py
index ca35574959..5f100daf38 100755
--- a/src/condor_contrib/mgmt/qmf/test/submit.py
+++ b/src/condor_contrib/mgmt/qmf/test/submit.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/plumage/test/plumage_accounting b/src/condor_contrib/plumage/test/plumage_accounting
index d85f976581..129207bf31 100755
--- a/src/condor_contrib/plumage/test/plumage_accounting
+++ b/src/condor_contrib/plumage/test/plumage_accounting
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/plumage/test/plumage_history b/src/condor_contrib/plumage/test/plumage_history
index 3c01cf9475..b68955b58e 100755
--- a/src/condor_contrib/plumage/test/plumage_history
+++ b/src/condor_contrib/plumage/test/plumage_history
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/plumage/test/plumage_scheduler b/src/condor_contrib/plumage/test/plumage_scheduler
index 259e04cdcb..d2a38460b5 100755
--- a/src/condor_contrib/plumage/test/plumage_scheduler
+++ b/src/condor_contrib/plumage/test/plumage_scheduler
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/plumage/test/plumage_stats b/src/condor_contrib/plumage/test/plumage_stats
index 5372e650ae..be5501417b 100755
--- a/src/condor_contrib/plumage/test/plumage_stats
+++ b/src/condor_contrib/plumage/test/plumage_stats
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2011 Red Hat, Inc.
diff --git a/src/condor_contrib/plumage/test/plumage_utilization b/src/condor_contrib/plumage/test/plumage_utilization
index 6235adec1c..b008019ab5 100755
--- a/src/condor_contrib/plumage/test/plumage_utilization
+++ b/src/condor_contrib/plumage/test/plumage_utilization
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2009-2012 Red Hat, Inc.
diff --git a/src/condor_contrib/triggerd/src/condor_trigger_config.py b/src/condor_contrib/triggerd/src/condor_trigger_config.py
index 8ac5e93ce8..d5495058ac 100644
--- a/src/condor_contrib/triggerd/src/condor_trigger_config.py
+++ b/src/condor_contrib/triggerd/src/condor_trigger_config.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 #
 # Copyright 2010 Red Hat, Inc.
diff --git a/src/condor_examples/master_wrapper b/src/condor_examples/master_wrapper
index 94393c004b..3fa23cbf89 100755
--- a/src/condor_examples/master_wrapper
+++ b/src/condor_examples/master_wrapper
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 ##**************************************************************
 ##
 ## Copyright (C) 1990-2010, Condor Team, Computer Sciences Department,
diff --git a/src/condor_gridmanager/slurm_status.py b/src/condor_gridmanager/slurm_status.py
index 61e68682ff..3d695c0611 100755
--- a/src/condor_gridmanager/slurm_status.py
+++ b/src/condor_gridmanager/slurm_status.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 #  File:     slurm_status.py
 #
diff --git a/src/condor_job_router/condor_router_history b/src/condor_job_router/condor_router_history
index 2b36844b72..963b2e7bd2 100755
--- a/src/condor_job_router/condor_router_history
+++ b/src/condor_job_router/condor_router_history
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 ##**************************************************************
 ##
 ## Copyright (C) 1990-2007, Condor Team, Computer Sciences Department,
diff --git a/src/condor_scripts/condor_top b/src/condor_scripts/condor_top
index c31f14d6a7..3fbc080b57 100755
--- a/src/condor_scripts/condor_top
+++ b/src/condor_scripts/condor_top
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import htcondor
 import classad
diff --git a/src/condor_tests/BasePython/pythontest.py b/src/condor_tests/BasePython/pythontest.py
index cbbcbae1a9..da7ee65654 100755
--- a/src/condor_tests/BasePython/pythontest.py
+++ b/src/condor_tests/BasePython/pythontest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import platform
diff --git a/src/condor_tests/classad_tests.py b/src/condor_tests/classad_tests.py
index 3c289495a8..48927bf91a 100644
--- a/src/condor_tests/classad_tests.py
+++ b/src/condor_tests/classad_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import re
 import classad
diff --git a/src/condor_tests/cmd_curl_plugin.run b/src/condor_tests/cmd_curl_plugin.run
index cee51f9f58..6cd67107c2 100755
--- a/src/condor_tests/cmd_curl_plugin.run
+++ b/src/condor_tests/cmd_curl_plugin.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/src/condor_tests/cmd_curl_plugin_multifile_failure.run b/src/condor_tests/cmd_curl_plugin_multifile_failure.run
index 11c3b56a48..a9b1bded4e 100755
--- a/src/condor_tests/cmd_curl_plugin_multifile_failure.run
+++ b/src/condor_tests/cmd_curl_plugin_multifile_failure.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import htcondor
 import os
diff --git a/src/condor_tests/cmd_curl_plugin_multifile_success.py b/src/condor_tests/cmd_curl_plugin_multifile_success.py
index 8fd509c14d..ef8cafb6db 100755
--- a/src/condor_tests/cmd_curl_plugin_multifile_success.py
+++ b/src/condor_tests/cmd_curl_plugin_multifile_success.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/src/condor_tests/cmd_curl_plugin_multifile_success.run b/src/condor_tests/cmd_curl_plugin_multifile_success.run
index 532024f56d..9756227c6a 100755
--- a/src/condor_tests/cmd_curl_plugin_multifile_success.run
+++ b/src/condor_tests/cmd_curl_plugin_multifile_success.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import htcondor
 import os
diff --git a/src/condor_tests/cmd_drain_policies.run b/src/condor_tests/cmd_drain_policies.run
index c908493588..8b60f8a827 100755
--- a/src/condor_tests/cmd_drain_policies.run
+++ b/src/condor_tests/cmd_drain_policies.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/src/condor_tests/job_basic_pytest.run b/src/condor_tests/job_basic_pytest.run
index 4f82dc30cf..7a718473cc 100755
--- a/src/condor_tests/job_basic_pytest.run
+++ b/src/condor_tests/job_basic_pytest.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/src/condor_tests/job_late_materialize_py.run b/src/condor_tests/job_late_materialize_py.run
index 866d4da03b..9e7fdaffd7 100755
--- a/src/condor_tests/job_late_materialize_py.run
+++ b/src/condor_tests/job_late_materialize_py.run
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 ##**************************************************************
 ##
 ## Copyright (C) 2018, Condor Team, Computer Sciences Department,
diff --git a/src/condor_tests/python_bindings/classad_tests.py b/src/condor_tests/python_bindings/classad_tests.py
index 44d8762cf5..5b3b19fe7a 100644
--- a/src/condor_tests/python_bindings/classad_tests.py
+++ b/src/condor_tests/python_bindings/classad_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import re
diff --git a/src/condor_tests/python_bindings/delayed_update.py b/src/condor_tests/python_bindings/delayed_update.py
index b67de9fa8f..987ec69db8 100644
--- a/src/condor_tests/python_bindings/delayed_update.py
+++ b/src/condor_tests/python_bindings/delayed_update.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import sys
diff --git a/src/condor_tests/python_bindings/htcondor_tests.py b/src/condor_tests/python_bindings/htcondor_tests.py
index 3b4da86160..044e9f4a25 100755
--- a/src/condor_tests/python_bindings/htcondor_tests.py
+++ b/src/condor_tests/python_bindings/htcondor_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import re
diff --git a/src/condor_tests/python_bindings/htcondor_version_tests.py b/src/condor_tests/python_bindings/htcondor_version_tests.py
index de40976314..4ed993e3e5 100644
--- a/src/condor_tests/python_bindings/htcondor_version_tests.py
+++ b/src/condor_tests/python_bindings/htcondor_version_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import sys
diff --git a/src/condor_tests/python_bindings_classad.run b/src/condor_tests/python_bindings_classad.run
index b1c945d875..7a46be905b 100755
--- a/src/condor_tests/python_bindings_classad.run
+++ b/src/condor_tests/python_bindings_classad.run
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import classad
 import datetime
diff --git a/src/condor_tests/python_bindings_example.run b/src/condor_tests/python_bindings_example.run
index 15ccd2b059..cf1b3e343b 100644
--- a/src/condor_tests/python_bindings_example.run
+++ b/src/condor_tests/python_bindings_example.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 ##**************************************************************
 ##
diff --git a/src/condor_tests/python_bindings_jobeventlog.run b/src/condor_tests/python_bindings_jobeventlog.run
index dc800cba25..ac564faf09 100644
--- a/src/condor_tests/python_bindings_jobeventlog.run
+++ b/src/condor_tests/python_bindings_jobeventlog.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/src/condor_tests/python_collector_query.run b/src/condor_tests/python_collector_query.run
index 5bb8b19e91..b4a9ae88bc 100755
--- a/src/condor_tests/python_collector_query.run
+++ b/src/condor_tests/python_collector_query.run
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 ##**************************************************************
 ##
 ## Copyright (C) 2018, Condor Team, Computer Sciences Department,
diff --git a/src/condor_tests/python_example.run b/src/condor_tests/python_example.run
index 1ba7f1d876..ca8b39b8e7 100644
--- a/src/condor_tests/python_example.run
+++ b/src/condor_tests/python_example.run
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 ##**************************************************************
 ##
diff --git a/src/condor_tests/pythontestfiles/pythontest.py b/src/condor_tests/pythontestfiles/pythontest.py
index 69db1feb7c..eb96a3b716 100755
--- a/src/condor_tests/pythontestfiles/pythontest.py
+++ b/src/condor_tests/pythontestfiles/pythontest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import platform
diff --git a/src/condor_tools/userlog_tail.py b/src/condor_tools/userlog_tail.py
index 6417c85f81..f70845f784 100755
--- a/src/condor_tools/userlog_tail.py
+++ b/src/condor_tools/userlog_tail.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import sys;
 import htcondor;
diff --git a/src/condor_utils/condor_log_reader.py b/src/condor_utils/condor_log_reader.py
index 5a822ed51c..593a2600ab 100644
--- a/src/condor_utils/condor_log_reader.py
+++ b/src/condor_utils/condor_log_reader.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 ##**************************************************************
 ##
 ## Copyright (C) 1990-2008, Condor Team, Computer Sciences Department,
diff --git a/src/condor_utils/test_log_verify b/src/condor_utils/test_log_verify
index c932a14bac..fee56e7fb8 100755
--- a/src/condor_utils/test_log_verify
+++ b/src/condor_utils/test_log_verify
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 ##**************************************************************
 ##
 ## Copyright (C) 1990-2008, Condor Team, Computer Sciences Department,
diff --git a/src/python-bindings/tests/classad_tests.py b/src/python-bindings/tests/classad_tests.py
index 44d8762cf5..5b3b19fe7a 100644
--- a/src/python-bindings/tests/classad_tests.py
+++ b/src/python-bindings/tests/classad_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import re
diff --git a/src/python-bindings/tests/delayed_update.py b/src/python-bindings/tests/delayed_update.py
index b67de9fa8f..987ec69db8 100644
--- a/src/python-bindings/tests/delayed_update.py
+++ b/src/python-bindings/tests/delayed_update.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import sys
diff --git a/src/python-bindings/tests/htcondor_tests.py b/src/python-bindings/tests/htcondor_tests.py
index 0895bbc4c2..e3873da818 100644
--- a/src/python-bindings/tests/htcondor_tests.py
+++ b/src/python-bindings/tests/htcondor_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import re
diff --git a/src/python-bindings/tests/htcondor_version_tests.py b/src/python-bindings/tests/htcondor_version_tests.py
index de40976314..4ed993e3e5 100644
--- a/src/python-bindings/tests/htcondor_version_tests.py
+++ b/src/python-bindings/tests/htcondor_version_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os
 import sys
