--- org.glite.security.voms/configure	2009-10-06 08:20:07.000000000 -0500
+++ org.glite.security.voms-patched/configure	2011-01-28 14:08:58.000000000 -0600
@@ -15761,8 +15761,13 @@
 
 
   if test "x$found" = "xyes"; then
-    NO_GLOBUS_FLAGS="-I$with_openssl_prefix/include"
-    OPENSSL_LIBS="-L$with_openssl_prefix/lib -lcrypto -lssl"
+    if test "x$with_openssl_prefix" = "x/usr"; then
+      NO_GLOBUS_FLAGS=""
+      OPENSSL_LIBS="-lcrypto -lssl"
+    else
+      NO_GLOBUS_FLAGS="-I$with_openssl_prefix/include"
+      OPENSSL_LIBS="-L$with_openssl_prefix/lib -lcrypto -lssl"
+    fi
 
 
     { $as_echo "$as_me:$LINENO: checking for system OpenSSL version" >&5
