--- org.glite.security.voms/configure	2012-02-06 16:35:50.000000000 -0600
+++ org.glite.security.voms.patched/configure	2012-02-06 16:36:34.000000000 -0600
@@ -16176,22 +16176,22 @@
     { $as_echo "$as_me:$LINENO: checking for GLOBUS flavors" >&5
 $as_echo_n "checking for GLOBUS flavors... " >&6; }
 
-    for i in `ls $with_globus_prefix/include`; do
+    for i in `ls $with_globus_prefix/include/globus | grep -v '[.]h$'`; do
       if test "x$i" != "xldap_backend"; then
         GLOBUS_FLAVORS="$GLOBUS_FLAVORS $i"
       fi
     done
 
     if test "x$with_64bit" == "xno" ; then
-      if test -e $with_globus_prefix/include/gcc32dbg ; then
+      if test -e $with_globus_prefix/include/globus/gcc32dbg ; then
         default_flavor=gcc32dbg
-      elif test -e $with_globus_prefix/include/gcc64dbg ; then
+      elif test -e $with_globus_prefix/include/globus/gcc64dbg ; then
         default_flavor=gcc64dbg
       else
         default_flavor=""
       fi
     else
-      if test -e $with_globus_prefix/include/gcc64dbg ; then
+      if test -e $with_globus_prefix/include/globus/gcc64dbg ; then
         default_flavor=gcc64dbg
       else
         default_flavor=""
@@ -16310,7 +16310,7 @@
     fi
 
     for flavor in $GLOBUS_FLAVORS ; do
-      if ( ( ldd $with_globus_prefix/lib/libglobus_gssapi_gsi_$flavor.so |grep crypto_$flavor >/dev/null 2>&1) ); then
+      if ( ( ldd $with_globus_prefix/lib/libglobus_gssapi_gsi.so |grep crypto_$flavor >/dev/null 2>&1) ); then
         FLVR="_$flavor"
       else
         FLVR=""
@@ -16318,49 +16318,49 @@
       { $as_echo "$as_me:$LINENO: result: found flavor=$FLVR" >&5
 $as_echo "found flavor=$FLVR" >&6; }
       if test "x$flavor" = "x$with_globus_flavor" ; then
-      	GLOBUS_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+      	GLOBUS_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32" ; then
-	      GLOBUS_GCC32_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC32_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+	      GLOBUS_GCC32_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC32_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC32_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32dbg" ; then
-        GLOBUS_GCC32DBG_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC32DBG_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+        GLOBUS_GCC32DBG_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC32DBG_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC32DBG_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32dbgpthr" ; then
-        GLOBUS_GCC32DBGPTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC32DBGPTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+        GLOBUS_GCC32DBGPTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC32DBGPTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC32DPBPTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc32pthr" ; then
-        GLOBUS_GCC32PTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC32PTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+        GLOBUS_GCC32PTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC32PTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC32PTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
 
       if test "x$flavor" = "xgcc64" ; then
-	      GLOBUS_GCC64_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC64_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+	      GLOBUS_GCC64_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC64_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC64_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc64dbg" ; then
-        GLOBUS_GCC64DBG_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC64DBG_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+        GLOBUS_GCC64DBG_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC64DBG_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC64DBG_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc64dbgpthr" ; then
-        GLOBUS_GCC64DBGPTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC64DBGPTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+        GLOBUS_GCC64DBGPTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC64DBGPTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC64DBGPTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
       if test "x$flavor" = "xgcc64pthr" ; then
-        GLOBUS_GCC64PTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/$flavor"
-        GLOBUS_GCC64PTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi_$flavor -lglobus_gss_assist_$flavor"
+        GLOBUS_GCC64PTHR_CFLAGS="$NO_GLOBUS_FLAGS -I$with_globus_prefix/include/globus"
+        GLOBUS_GCC64PTHR_GSS_LIBS="$ac_globus_ldlib -lglobus_gssapi_gsi -lglobus_gss_assist"
         GLOBUS_GCC64PTHR_GSS_API_LIBS="$ac_globus_ldlib -lcrypto$FLVR -lssl$FLVR"
       fi
     done
