--- org.glite.security.voms/src/socklib/Makefile.am	2009-04-07 04:05:29.000000000 -0500
+++ org.glite.security.voms-patched/src/socklib/Makefile.am	2010-10-12 13:10:54.000000000 -0500
@@ -111,7 +111,7 @@
 libsock_server_gcc32dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC32DBG_CFLAGS)
+	$(GLOBUS_GCC32DBGPTHR_CFLAGS)
 
 libsock_server_gcc32dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc32dbgpthr.la \
@@ -163,7 +163,7 @@
 libsock_server_gcc64dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC64DBG_CFLAGS)
+	$(GLOBUS_GCC64DBGPTHR_CFLAGS)
 
 libsock_server_gcc64dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc64dbgpthr.la \
--- org.glite.security.voms/src/socklib/Makefile.in	2010-10-12 13:11:52.000000000 -0500
+++ org.glite.security.voms/src/socklib/Makefile.in-patched	2010-10-12 13:11:02.000000000 -0500
@@ -511,7 +511,7 @@
 libsock_server_gcc32dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC32DBG_CFLAGS)
+	$(GLOBUS_GCC32DBGPTHR_CFLAGS)
 
 libsock_server_gcc32dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc32dbgpthr.la \
@@ -563,7 +563,7 @@
 libsock_server_gcc64dbgpthr_la_CPPFLAGS = \
 	-I $(top_srcdir)/src/include \
 	-I $(top_srcdir)/src/sslutils \
-	$(GLOBUS_GCC64DBG_CFLAGS)
+	$(GLOBUS_GCC64DBGPTHR_CFLAGS)
 
 libsock_server_gcc64dbgpthr_la_LIBADD = \
 	$(top_builddir)/src/sslutils/libssl_utils_gcc64dbgpthr.la \
