diff -u -r org.glite.security.voms/src/api/ccapi/api_util.cc org.glite.security.voms-patched/src/api/ccapi/api_util.cc
--- org.glite.security.voms/src/api/ccapi/api_util.cc	2009-10-06 08:17:04.000000000 -0500
+++ org.glite.security.voms-patched/src/api/ccapi/api_util.cc	2011-07-05 23:59:21.000000000 -0500
@@ -402,7 +402,7 @@
   AC * ac = (AC *)data;
   STACK_OF(AC_ATTR) * atts = ac->acinfo->attrib;
   int nid = OBJ_txt2nid("idatcap");
-  int pos = X509at_get_attr_by_NID(atts, nid, -1);
+  int pos = X509at_get_attr_by_NID((STACK_OF(X509_ATTRIBUTE) *)atts, nid, -1);
 
   int nidc = OBJ_txt2nid("certseq");
   STACK_OF(X509_EXTENSION) *exts = ac->acinfo->exts;
diff -u -r org.glite.security.voms/src/include/acstack.h org.glite.security.voms-patched/src/include/acstack.h
--- org.glite.security.voms/src/include/acstack.h	2005-07-29 10:15:04.000000000 -0500
+++ org.glite.security.voms-patched/src/include/acstack.h	2011-07-05 23:47:37.000000000 -0500
@@ -19,7 +19,7 @@
 #include <openssl/safestack.h>
 
 #define IMPL_STACK(type) \
-   DECLARE_STACK_OF(type) \
+   /*DECLARE_STACK_OF(type)*/ \
    STACK_OF(type) *sk_##type##_new (int (*cmp)(const type * const *, const type * const *)) \
        { return sk_new ( (int (*)(const char * const *, const char * const *))cmp);} \
    STACK_OF(type) *sk_##type##_new_null () { return sk_new_null(); } \
diff -u -r org.glite.security.voms/src/socklib/Client.cpp org.glite.security.voms-patched/src/socklib/Client.cpp
--- org.glite.security.voms/src/socklib/Client.cpp	2009-04-07 04:05:29.000000000 -0500
+++ org.glite.security.voms-patched/src/socklib/Client.cpp	2011-07-05 21:28:59.000000000 -0500
@@ -272,7 +272,11 @@
 bool 
 GSISocketClient::Open()
 {
+#if OPENSSL_VERSION_NUMBER < 0x10000000L
   SSL_METHOD *meth = NULL;
+#else
+  const SSL_METHOD *meth = NULL;
+#endif
 
   meth = SSLv3_method();
 
