diff -u -r org.glite.security.voms/project/acinclude.m4 org.glite.security.voms-patched/project/acinclude.m4
--- org.glite.security.voms/project/acinclude.m4	2009-04-07 04:05:28.000000000 -0500
+++ org.glite.security.voms-patched/project/acinclude.m4	2010-10-15 02:08:48.000000000 -0500
@@ -920,12 +920,12 @@
 
 AC_DEFUN([PUT_PRIVATES],
 [
-        AH_BOTTOM([#ifdef __GNUC__
+        AH_BOTTOM([#ifdef __SMELL__
 #define UNUSED(z)  z __attribute__ ((unused))
 #define PRIVATE    __attribute__ ((visibility ("hidden")))
 #define PUBLIC     __attribute__ ((visibility ("default")))
 #else
-#define UNUSED
+#define UNUSED(z)  z
 #define PRIVATE
 #define PUBLIC
 #endif])])
diff -u -r org.glite.security.voms/src/autogen/config.h.in org.glite.security.voms-patched/src/autogen/config.h.in
--- org.glite.security.voms/src/autogen/config.h.in	2009-04-07 04:05:29.000000000 -0500
+++ org.glite.security.voms-patched/src/autogen/config.h.in	2010-10-15 02:08:18.000000000 -0500
@@ -371,12 +371,12 @@
 #include <new>
 #endif
 
-#ifdef __GNUC__
+#ifdef __SMELL__
 #define UNUSED(z)  z __attribute__ ((unused))
 #define PRIVATE    __attribute__ ((visibility ("hidden")))
 #define PUBLIC     __attribute__ ((visibility ("default")))
 #else
-#define UNUSED
+#define UNUSED(z)  z
 #define PRIVATE
 #define PUBLIC
 #endif
