--- tools/build/src/tools/python.jam.orig	2018-04-13 16:21:27.138928875 -0500
+++ tools/build/src/tools/python.jam	2018-04-13 16:17:44.477715149 -0500
@@ -830,7 +830,7 @@
     # Discover the presence of NumPy
     #
     debug-message "Checking for NumPy..." ;
-    local full-cmd = "import sys; sys.stderr = sys.stdout; import numpy; print(numpy.get_include())" ;
+    local full-cmd = "import sys; sys.stderr = sys.stdout; import numpyfail; print(numpy.get_include())" ;
     local full-cmd = $(interpreter-cmd)" -c \"$(full-cmd)\"" ;
     debug-message "running command '$(full-cmd)'" ;
     local result = [ SHELL $(full-cmd) : strip-eol : exit-status ] ;
