--- gt5.0.0-all-source-installer/source-trees/gram/client/source/globus_gram_client.c	2010-02-03 09:50:25.000000000 -0600
+++ gt5.0.0-all-source-installer/source-trees/gram/client/source/globus_gram_client.c.patched	2010-02-03 10:23:51.000000000 -0600
@@ -1392,6 +1392,7 @@
                                           &monitor);
     if(rc != GLOBUS_SUCCESS)
     {
+	monitor.info = NULL;
         globus_l_gram_client_monitor_destroy(&monitor);
 
         return rc;
@@ -1409,6 +1410,7 @@
     }
     globus_mutex_unlock(&monitor.mutex);
 
+    monitor.info = NULL;
     globus_l_gram_client_monitor_destroy(&monitor);
 
     return rc;
@@ -2341,6 +2343,7 @@
     globus_mutex_unlock(&monitor.mutex);
 
 error_exit:
+    monitor.info = NULL;
     globus_l_gram_client_monitor_destroy(&monitor);
 
     return rc;
@@ -4145,11 +4148,8 @@
     globus_l_gram_client_monitor_t *    monitor)
 {
 
-    if (monitor->info->job_contact != NULL)
-    {
-        globus_gram_client_job_contact_free(monitor->info->job_contact);
-        monitor->info->job_contact = NULL;
-    }
+    globus_gram_client_job_info_destroy(monitor->info);
+    free(monitor->info);
 
     globus_mutex_destroy(&monitor->mutex);
     globus_cond_destroy(&monitor->cond);
