Index: globus_gram_protocol_io.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/protocol/source/globus_gram_protocol_io.c,v
retrieving revision 1.18.2.3
diff -u -r1.18.2.3 globus_gram_protocol_io.c
--- globus_gram_protocol_io.c	4 Nov 2008 21:39:36 -0000	1.18.2.3
+++ globus_gram_protocol_io.c	16 Jun 2009 18:19:07 -0000
@@ -749,7 +749,7 @@
     }
     if(result != GLOBUS_SUCCESS)
     {
-        goto error_exit;
+        goto no_connection;
     }
     connection = globus_libc_calloc(
 		    1,
@@ -757,7 +757,7 @@
 
     if(connection == GLOBUS_NULL)
     {
-        goto error_exit;
+        goto no_connection;
     }
     connection->read_type = GLOBUS_GRAM_PROTOCOL_REQUEST;
     connection->callback = listener->callback;
@@ -823,6 +823,11 @@
   free_connection_exit:
     globus_libc_free(connection);
 
+  no_connection:
+    globus_io_tcp_register_listen(handle,
+                                  globus_l_gram_protocol_listen_callback,
+				  listener);
+
   error_exit:
     globus_mutex_unlock(&globus_i_gram_protocol_mutex);
 }
