--- gt4.2.0-all-source-installer/source-trees/common/source/library/globus_object.h	Mon Apr 18 17:59:33 2005
+++ gt4.2.0-all-source-installer.patched/source-trees/common/source/library/globus_object.h	Wed Feb  7 14:28:59 2007
@@ -34,6 +34,10 @@
   globus_object_copy_func_t const             copy_func;
   globus_object_destructor_func_t const       destructor;
   void * const                                class_data;
+#ifdef __cplusplus
+  private:
+  globus_object_type_s() : parent_type(NULL), copy_func(NULL), destructor(NULL), class_data(NULL) {};
+#endif
 } globus_object_type_t;
 
 typedef struct globus_object_s {
