diff -u -r gt4.2.0-all-source-installer/source-trees/gram/protocol/source/globus_gram_protocol_io.c gt4.2.0-all-source-installer.patch/source-trees/gram/protocol/source/globus_gram_protocol_io.c
--- gt4.2.0-all-source-installer/source-trees/gram/protocol/source/globus_gram_protocol_io.c	2006-09-22 07:42:07.000000000 -0500
+++ gt4.2.0-all-source-installer.patch/source-trees/gram/protocol/source/globus_gram_protocol_io.c	2008-09-16 13:39:16.000000000 -0500
@@ -1179,17 +1179,10 @@
 			     GLOBUS_NULL);
     }
     
-    /* rc only set on connection error.  It is an error to close a handle on
-     * a failed connect
-     */
-    if(rc == 0)
-    {
-        result = globus_io_register_close(
-	    handle,
-	    globus_l_gram_protocol_connection_close_callback,
-	    callback_arg);
-    
-    }
+    result = globus_io_register_close(
+	handle,
+	globus_l_gram_protocol_connection_close_callback,
+	callback_arg);
     
     if(result != GLOBUS_SUCCESS)
     {
