diff -u -r SRB3_2_1e/Makefile.in SRB3_2_1e.patched/Makefile.in
--- SRB3_2_1e/Makefile.in	2004-06-01 18:58:31.000000000 -0500
+++ SRB3_2_1e.patched/Makefile.in	2008-08-11 13:29:40.000000000 -0500
@@ -114,10 +114,10 @@
 SVR_CFLAGS+=-I$(buildDir)/src/include 
 
 ifdef GSI_AUTH
-SVR_CFLAGS+=-I$(AUTH_INCLUDE)
+SVR_CFLAGS+=$(AUTH_INCLUDE)
 else
 ifdef SRB_SECURE_COMM
-SVR_CFLAGS+=-I$(AUTH_INCLUDE)
+SVR_CFLAGS+=$(AUTH_INCLUDE)
 endif
 endif
 
diff -u -r SRB3_2_1e/mk/mk.config.in SRB3_2_1e.patched/mk/mk.config.in
--- SRB3_2_1e/mk/mk.config.in	2004-06-29 17:21:58.000000000 -0500
+++ SRB3_2_1e.patched/mk/mk.config.in	2008-08-11 13:32:05.000000000 -0500
@@ -678,23 +678,23 @@
 # LIB_GSI_AUTH specifies where the GSI auth library is located.
 GLOBUS_LOCATION=@GSIHOME@
 GSI_INSTALL_TYPE=@GSI_INSTALL_TYPE@
-AUTH_INCLUDE += $(GLOBUS_LOCATION)/include/$(GSI_INSTALL_TYPE)
+AUTH_INCLUDE += -I$(GLOBUS_LOCATION)/include/$(GSI_INSTALL_TYPE)
 LIB_GSI_AUTH = \
-        -L $(GLOBUS_LOCATION)/lib \
+        -L$(GLOBUS_LOCATION)/lib \
         -lglobus_gss_assist_$(GSI_INSTALL_TYPE) \
+	-lglobus_gsi_cert_utils_$(GSI_INSTALL_TYPE) \
+	-lglobus_proxy_ssl_$(GSI_INSTALL_TYPE) \
 	-lglobus_gssapi_gsi_$(GSI_INSTALL_TYPE) \
-        -lglobus_gsi_credential_$(GSI_INSTALL_TYPE) \
         -lglobus_gsi_proxy_core_$(GSI_INSTALL_TYPE) \
+	-lglobus_gsi_credential_$(GSI_INSTALL_TYPE) \
 	-lglobus_gsi_callback_$(GSI_INSTALL_TYPE) \
-	-lglobus_oldgaa_$(GSI_INSTALL_TYPE) \
 	-lglobus_gsi_sysconfig_$(GSI_INSTALL_TYPE) \
-	-lglobus_gsi_cert_utils_$(GSI_INSTALL_TYPE) \
-	-lglobus_openssl_error_$(GSI_INSTALL_TYPE) \
 	-lglobus_openssl_$(GSI_INSTALL_TYPE) \
-	-lglobus_proxy_ssl_$(GSI_INSTALL_TYPE) \
-	-lssl_$(GSI_INSTALL_TYPE) \
-	-lcrypto_$(GSI_INSTALL_TYPE) \
-	-lglobus_common_$(GSI_INSTALL_TYPE)
+	-lglobus_openssl_error_$(GSI_INSTALL_TYPE) \
+	-lglobus_oldgaa_$(GSI_INSTALL_TYPE) \
+	-lglobus_callout_$(GSI_INSTALL_TYPE) \
+	-lglobus_common_$(GSI_INSTALL_TYPE) \
+	-lltdl_$(GSI_INSTALL_TYPE)
 endif
 # End GSI_AUTH config
 
@@ -706,9 +706,9 @@
 ifdef SRB_SECURE_COMM
 OPENSSL_LOCATION=@OPENSSLHOME@
 ifdef OPENSSL_LOCATION
-AUTH_INCLUDE += $(OPENSSL_LOCATION)/include
-LIB_GSI_AUTH = \
-        -L $(OPENSSL_LOCATION)/lib \
+AUTH_INCLUDE += -I$(OPENSSL_LOCATION)/include
+LIB_GSI_AUTH += \
+        -L$(OPENSSL_LOCATION)/lib \
 	-lssl \
 	-lcrypto
 else
diff -u -r SRB3_2_1e/test/GSItest/Makefile SRB3_2_1e.patched/test/GSItest/Makefile
--- SRB3_2_1e/test/GSItest/Makefile	2004-03-02 12:53:15.000000000 -0600
+++ SRB3_2_1e.patched/test/GSItest/Makefile	2008-08-11 13:26:25.000000000 -0500
@@ -30,19 +30,21 @@
 GSI_LIBS = \
 	-L $(GLOBUS_LOCATION)/lib \
 	-lglobus_gss_assist_$(GSI_INSTALL_TYPE) \
+	-lglobus_gsi_cert_utils_$(GSI_INSTALL_TYPE) \
+	-lglobus_proxy_ssl_$(GSI_INSTALL_TYPE) \
 	-lglobus_gssapi_gsi_$(GSI_INSTALL_TYPE) \
-	-lglobus_gsi_credential_$(GSI_INSTALL_TYPE) \
 	-lglobus_gsi_proxy_core_$(GSI_INSTALL_TYPE) \
+	-lglobus_gsi_credential_$(GSI_INSTALL_TYPE) \
 	-lglobus_gsi_callback_$(GSI_INSTALL_TYPE) \
-	-lglobus_oldgaa_$(GSI_INSTALL_TYPE) \
-	-lglobus_gsi_sysconfig_$(GSI_INSTALL_TYPE) \
-	-lglobus_gsi_cert_utils_$(GSI_INSTALL_TYPE) \
-	-lglobus_openssl_error_$(GSI_INSTALL_TYPE) \
-	-lglobus_openssl_$(GSI_INSTALL_TYPE) \
-	-lglobus_proxy_ssl_$(GSI_INSTALL_TYPE) \
 	-lssl_$(GSI_INSTALL_TYPE) \
 	-lcrypto_$(GSI_INSTALL_TYPE) \
-	-lglobus_common_$(GSI_INSTALL_TYPE)
+	-lglobus_gsi_sysconfig_$(GSI_INSTALL_TYPE) \
+	-lglobus_openssl_$(GSI_INSTALL_TYPE) \
+	-lglobus_openssl_error_$(GSI_INSTALL_TYPE) \
+	-lglobus_oldgaa_$(GSI_INSTALL_TYPE) \
+	-lglobus_callout_$(GSI_INSTALL_TYPE) \
+	-lglobus_common_$(GSI_INSTALL_TYPE) \
+	-lltdl_$(GSI_INSTALL_TYPE)
 
 ifndef MY_DCE
 CC=cc
