diff -r -u krb5-1.4.3-p0.orig/src/appl/gssftp/ftpd/ftpcmd.y krb5-1.4.3-p0.patched/src/appl/gssftp/ftpd/ftpcmd.y
--- krb5-1.4.3-p0.orig/src/appl/gssftp/ftpd/ftpcmd.y	2004-12-03 19:39:07.000000000 -0600
+++ krb5-1.4.3-p0.patched/src/appl/gssftp/ftpd/ftpcmd.y	2011-05-26 16:01:01.536685466 -0500
@@ -968,10 +968,10 @@
 #include <arpa/telnet.h>
 
 /*
- * getline - a hacked up version of fgets to ignore TELNET escape codes.
+ * xgetline - a hacked up version of fgets to ignore TELNET escape codes.
  */
 char *
-getline(s, n, iop)
+xgetline(s, n, iop)
 	char *s;
 	int n;
 	register FILE *iop;
@@ -1213,13 +1213,13 @@
 		case CMD:
 			(void) signal(SIGALRM, toolong);
 			(void) alarm((unsigned) timeout);
-			if (getline(cbuf, sizeof(cbuf)-1, stdin) == NULL) {
+			if (xgetline(cbuf, sizeof(cbuf)-1, stdin) == NULL) {
 				reply(221, "You could at least say goodbye.");
 				dologout(0);
 			}
 			(void) alarm(0);
 
-			/* If getline() finds an error, the string is null */
+			/* If xgetline() finds an error, the string is null */
 			if (*cbuf == '\0')
 				continue;
 
diff -r -u krb5-1.4.3-p0.orig/src/appl/gssftp/ftpd/ftpd.c krb5-1.4.3-p0.patched/src/appl/gssftp/ftpd/ftpd.c
--- krb5-1.4.3-p0.orig/src/appl/gssftp/ftpd/ftpd.c	2005-01-21 16:46:47.000000000 -0600
+++ krb5-1.4.3-p0.patched/src/appl/gssftp/ftpd/ftpd.c	2011-05-26 16:00:28.959690454 -0500
@@ -176,7 +176,7 @@
 extern	char *home;		/* pointer to home directory for glob */
 extern	FILE *ftpd_popen(), *fopen(), *freopen();
 extern	int  ftpd_pclose(), fclose();
-extern	char *getline();
+extern	char *xgetline();
 extern	char cbuf[];
 extern	off_t restart_point;
 
@@ -2139,7 +2139,7 @@
 	if (!transflag)
 		return;
 	cp = tmpline;
-	if (getline(cp, sizeof(tmpline), stdin) == NULL) {
+	if (xgetline(cp, sizeof(tmpline), stdin) == NULL) {
 		reply(221, "You could at least say goodbye.");
 		dologout(0);
 	}
diff -r -u krb5-1.4.3-p0.orig/src/appl/gssftp/ftpd/ftpd_var.h krb5-1.4.3-p0.patched/src/appl/gssftp/ftpd/ftpd_var.h
--- krb5-1.4.3-p0.orig/src/appl/gssftp/ftpd/ftpd_var.h	2001-10-09 22:18:22.000000000 -0500
+++ krb5-1.4.3-p0.patched/src/appl/gssftp/ftpd/ftpd_var.h	2011-05-26 16:00:15.344695477 -0500
@@ -89,7 +89,7 @@
 
 /* ftpcmd.y */
 void upper(char *);
-char *getline(char *, int, FILE *);
+char *xgetline(char *, int, FILE *);
 #endif /* FTPD_VAR_H__ */
 
 /* popen.c */
