--- appl/bsd/Makefile	Thu May  4 17:36:11 2006
+++ appl/bsd/Makefile.freebsd	Thu May  4 17:42:10 2006
@@ -180,7 +180,7 @@
 INSTALL_SCRIPT=${INSTALL}
 INSTALL_DATA=${INSTALL} -m 644
 INSTALL_SHLIB=@INSTALL_SHLIB@
-INSTALL_SETUID=$(INSTALL) $(INSTALL_STRIP) -m 4755 -o root
+INSTALL_SETUID=$(INSTALL) $(INSTALL_STRIP) -m 4755
 ## This is needed because autoconf will sometimes define ${prefix} to be
 ## ${prefix}.
 prefix=/data/condor/externals/install/krb5-1.4.3
