diff -r -u krb5-1.4.3-p0.orig/src/include/fake-addrinfo.h krb5-1.4.3-p0.patched/src/include/fake-addrinfo.h
--- krb5-1.4.3-p0.orig/src/include/fake-addrinfo.h	2005-06-01 14:24:32.000000000 -0500
+++ krb5-1.4.3-p0.patched/src/include/fake-addrinfo.h	2011-05-26 14:34:27.635844891 -0500
@@ -164,6 +164,17 @@
 #define HAVE_GETNAMEINFO 1
 #endif
 
+/* Modern machines have changed how getaddrinfo is defined from RFC2553 to
+ * RFC 3943, but often didn't provide any backwards compatibility.
+ * So, given the knowledge from this post:
+ * http://osdir.com/ml/encryption.kerberos.devel/2005-02/msg00025.html
+ * I'm going to use that code. -psilord 05/26/2011
+ *
+ * See git track #1998 for the ticket which generated this patch.
+*/
+#if defined(EAI_NONAME) && !defined(EAI_NODATA)
+#define EAI_NODATA EAI_NONAME
+#endif
 
 /* Do we actually have *any* systems we care about that don't provide
    either getaddrinfo or one of these two flavors of
