--- krb5-1.4.3\src\include\win-mac.h	Wed Mar 08 22:48:32 2006
+++ krb5-1.4.3-zkm-patched\src\include\win-mac.h	Wed Mar 08 22:49:38 2006
@@ -57,11 +57,11 @@
 typedef unsigned int	u_int;
 typedef unsigned short	u_short;
 typedef unsigned char	u_char;
-typedef unsigned int    uint32_t;
+// typedef unsigned int    uint32_t;
 typedef int             int32_t;
 #endif /* KRB5_SYSTYPES__ */
 
-#define MAXHOSTNAMELEN  64
+// #define MAXHOSTNAMELEN  64
 #ifndef MAXPATHLEN
 #define MAXPATHLEN      256            /* Also for Windows temp files */
 #endif
