--- org.glite.ce.blahp.patched/config/blah.config.template	2011-12-13 15:32:40.000000000 -0600
+++ org.glite.ce.blahp/config/blah.config.template	2011-12-13 15:35:51.000000000 -0600
@@ -1,5 +1,5 @@
 #Supported batch systems
-supported_lrms=pbs,lsf
+supported_lrms=pbs,lsf,sge
 
 #DGAS logfile 
 BLAHPD_ACCOUNTING_INFO_LOG=
@@ -96,6 +96,13 @@
 lsf_BLPport2=
 
 #
+# SGE Configuration
+#
+sge_root=$SGE_ROOT
+sge_cellname=$SGE_CELL
+
+
+#
 #LSF Updater
 #
 #number of logs to be read by bhist (default:3)
