--- src/scripts/blah_common_submit_functions.sh	2018-02-07 16:17:43.976747509 -0600
+++ src/scripts/blah_common_submit_functions.sh.patched	2018-02-07 16:32:42.950400721 -0600
@@ -748,7 +748,7 @@
 function bls_set_up_local_and_extra_args ()
 {
   if [ -r $bls_local_submit_attributes_file ] ; then
-      echo \#\!/bin/sh > $bls_opt_tmp_req_file
+      echo \#\!/bin/bash > $bls_opt_tmp_req_file
       if [ ! -z $bls_opt_req_file ] ; then
           cat $bls_opt_req_file >> $bls_opt_tmp_req_file
       fi
