--- src/scripts/condor_submit.sh	2016-06-21 15:06:14.868679434 -0500
+++ src/scripts/condor_submit.sh.patched	2016-07-07 16:29:59.068839269 -0500
@@ -223,12 +223,15 @@
 
 ### This appears to only be necessary if Condor is passing arguments
 ### with the "new_esc_format"
+### If the first and last characters are '"', then assume that is the case.
 # # NOTE: The arguments we are given are specially escaped for a shell,
 # # so to get them back into Condor format we need to remove all the
 # # extra quotes. We do this by replacing '" "' with ' ' and stripping
 # # the leading and trailing "s.
-# arguments="$(echo $arguments | sed -e 's/\" \"/ /g')"
-# arguments=${arguments:1:$((${#arguments}-2))}
+if echo $arguments | grep -q '^".*"$' ; then
+  arguments="$(echo $arguments | sed -e 's/\" \"/ /g')"
+  arguments=${arguments:1:$((${#arguments}-2))}
+fi
 
 cat > $submit_file << EOF
 universe = vanilla
