diff -u -r org.glite.ce.blahp/src/scripts/condor_status.sh org.glite.ce.blahp-patched/src/scripts/condor_status.sh
--- org.glite.ce.blahp/src/scripts/condor_status.sh	2013-01-10 10:33:01.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/condor_status.sh	2013-01-10 10:52:33.000000000 -0600
@@ -248,6 +248,15 @@
 	fi
     fi
 
+    # Caching of condor_q output doesn't appear to work properly in
+    # HTCondor builds of the blahp. So do an explicit condor_q for
+    # this job before trying condor_history, which can take a long time.
+    line=$(echo $FORMAT | xargs $condor_binpath/condor_q $target $id)
+    if  [ -n "$line" ] ; then
+	echo "0$(make_ad $job "$line")"
+	exit 0
+    fi
+
     ### WARNING: This is troubling because the remote history file
     ### might just happen to be in the same place as a local history
     ### file, in which case condor_history is going to be looking at
@@ -259,7 +268,7 @@
     #   instead of using -f.
     history_file=$($condor_binpath/condor_config_val $target -schedd history)
     if [ "$?" == "0" ]; then
-	line=$(echo $FORMAT | _condor_HISTORY="$history_file" xargs $condor_binpath/condor_history -backwards $id)
+	line=$(echo $FORMAT | _condor_HISTORY="$history_file" xargs $condor_binpath/condor_history -backwards -match 1 $id)
 	if  [ ! -z "$line" ] ; then
 	    echo "0$(make_ad $job "$line")"
 	    exit 0
