diff -u -r org.glite.ce.blahp/src/scripts/condor_submit.sh org.glite.ce.blahp-patched/src/scripts/condor_submit.sh
--- org.glite.ce.blahp/src/scripts/condor_submit.sh	2016-12-15 15:29:59.343721729 -0600
+++ org.glite.ce.blahp-patched/src/scripts/condor_submit.sh	2016-12-15 15:31:42.499692275 -0600
@@ -351,7 +351,11 @@
 if [ "x$job_registry" == "x" ]; then
     if [ -r "$proxy_file" -a -f "$proxy_file" ] ; then
         [ -d "$proxy_dir" ] || mkdir $proxy_dir
-        ln -s $proxy_file $proxy_dir/$jobID.proxy.norenew
+        if [ "${proxy_file:0:1}" == "/" ] ; then
+            ln -s $proxy_file $proxy_dir/$jobID.proxy.norenew
+        else
+            ln -s `pwd`/$proxy_file $proxy_dir/$jobID.proxy.norenew
+        fi
     fi
 fi
 
