diff -u -r org.glite.ce.blahp/src/scripts/condor_status.sh org.glite.ce.blahp-patched/src/scripts/condor_status.sh
--- org.glite.ce.blahp/src/scripts/condor_status.sh	2011-03-04 06:56:04.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/condor_status.sh	2012-07-17 14:04:47.000000000 -0500
@@ -254,9 +254,12 @@
     ### the history of an unexpected queue.
 
     # We can possibly get the location of the history file and check it.
+    # NOTE: In Condor 7.7.6-7.8.1, the -f option to condor_history was
+    #   broken. To work around that, we set HISTORY via the environment
+    #   instead of using -f.
     history_file=$($condor_binpath/condor_config_val $target -schedd history)
     if [ "$?" == "0" ]; then
-	line=$(echo $FORMAT | xargs $condor_binpath/condor_history -f $history_file -backwards $id)
+	line=$(echo $FORMAT | _condor_HISTORY="$history_file" xargs $condor_binpath/condor_history -backwards $id)
 	if  [ ! -z "$line" ] ; then
 	    echo "0$(make_ad $job "$line")"
 	    exit 0
