--- org.glite.ce.blahp/src/scripts/condor_submit.sh	2016-04-19 10:40:25.111499726 -0500
+++ org.glite.ce.blahp/src/scripts/condor_submit.sh.patched	2016-04-19 11:16:30.505347353 -0500
@@ -47,13 +47,13 @@
 # script debug flag: currently unused
 debug=no
 
-# number of MPI nodes: currently unused
-mpinodes=0
+# number of MPI nodes: interpretted as a core count for vanilla universe
+mpinodes=1
 
 # Name of local requirements file: currently unused
 req_file=""
 
-while getopts "a:i:o:de:j:n:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:m:" arg 
+while getopts "a:i:o:de:j:n:N:z:h:S:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:m:" arg 
 do
     case "$arg" in
     a) xtra_args="$OPTARG" ;;
@@ -66,6 +66,10 @@
     V) environment="$OPTARG";;
     c) command="$OPTARG" ;;
     n) mpinodes="$OPTARG" ;;
+    N) hostsmpsize="$OPTARG";;
+    z) wholenodes="$OPTARG";;
+    h) hostnumber="$OPTARG";;
+    S) smpgranularity="$OPTARG";;
     w) workdir="$OPTARG";;
     x) proxy_file="$OPTARG" ;;
     u) proxy_subject="$OPTARG" ;;
@@ -247,6 +251,7 @@
 fi
 
 cat >> $submit_file << EOF
+request_cpus = $mpinodes
 # We insist on new style quoting in Condor
 arguments = $arguments
 input = $stdin
