diff -u -r org.glite.ce.blahp/config/blah.config.template org.glite.ce.blahp.patched/config/blah.config.template
--- org.glite.ce.blahp/config/blah.config.template	2012-02-02 10:58:36.000000000 -0600
+++ org.glite.ce.blahp.patched/config/blah.config.template	2012-02-02 10:59:15.000000000 -0600
@@ -15,17 +15,17 @@
 blah_delegate_renewed_proxies=no
 
 #Path where PBS executables are located 
-pbs_binpath=/usr/pbs/bin
+pbs_binpath=`which qsub 2>/dev/null|sed 's|/[^/]*$||'`
 
 #Path where the PBS logs are located ($pbs_spoolpath/server_logs)
-pbs_spoolpath=/usr/spool/PBS
+#pbs_spoolpath=/usr/spool/PBS
 
 #If it is set to yes blah does not check the jobid in the logfiles
 pbs_nochecksubmission=
 
 #If it is set to yes blah does NOT use log files to get job status, 
 #but uses only standard LRMS query (qstat)  
-pbs_nologaccess=
+pbs_nologaccess=yes
 
 #If it is set to no blah scripts for PBS will not try to read 
 #locally from the logs if BLParser is not present
@@ -61,7 +61,7 @@
 ####
 
 #Path where LSF executables are located 
-lsf_binpath=/usr/local/lsf/bin
+lsf_binpath=`which bsub 2>/dev/null|sed 's|/[^/]*$||'`
 
 #Path where the LSF conf file is located ($lsf_confpath/lsf.conf)
 lsf_confpath=/etc
@@ -124,7 +124,7 @@
 bnotifier_pidfile=/var/tmp/cream_tomcat_bnotifier.pid
 
 #condor bin location
-condor_binpath=/opt/condor-c/bin
+condor_binpath=`which condor_submit 2>/dev/null|sed 's|/[^/]*$||'`
 
 #Registry file location
 job_registry=
@@ -163,7 +163,7 @@
 alldone_interval=3600
 
 #path to condor_config
-export CONDOR_CONFIG="/opt/condor-c/etc/condor_config"
+#export CONDOR_CONFIG="/opt/condor-c/etc/condor_config"
 
 #max number of concurrent threads to serve commands (default = 500)
 #blah_max_threaded_cmds=100
