diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp.patched/src/server.c
--- org.glite.ce.blahp/src/server.c	2011-09-15 07:39:39.000000000 -0500
+++ org.glite.ce.blahp.patched/src/server.c	2012-07-03 17:57:35.000000000 -0500
@@ -3467,6 +3467,12 @@
 			memcpy(result + j, CONVARG_DBLQUOTESC, CONVARG_DBLQUOTESC_LEN);
 			j += CONVARG_DBLQUOTESC_LEN;
 		}
+		/* Must escape a few meta-characters for wordexp */
+		else if ((original[i] == '(') || (original[i] == ')') || (original[i] == '&'))
+		{
+			result[j++] = '\\';
+			result[j++] = original[i];
+		}
 		else
 		{	/* plain copy from the original */
 			result[j++] = original[i];
