--- org.glite.ce.blahp/src/classad_c_helper.C	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/classad_c_helper.C	2012-01-09 09:10:40.664337402 -0600
@@ -285,7 +285,7 @@
     ExprList *et_value;
     et_value = ExprList::MakeExprList(et_ads);
 
-    if (ad->Insert (name, et_value)) return C_CLASSAD_NO_ERROR; 
+    if (ad->Insert (name, (ExprTree* &)et_value)) return C_CLASSAD_NO_ERROR; 
     else                             return C_CLASSAD_INSERT_FAILED;
    }
 
