diff -u -r org.glite.ce.blahp/src/job_registry.c org.glite.ce.blahp.patched/src/job_registry.c
--- org.glite.ce.blahp/src/job_registry.c	2012-03-07 03:52:47.000000000 -0600
+++ org.glite.ce.blahp.patched/src/job_registry.c	2015-10-19 16:42:54.530781672 -0500
@@ -294,7 +294,7 @@
      }
     if (ret == 0) break;
 
-    if (cur.cdate < oldest_creation_date)
+    if (cur.cdate < oldest_creation_date && (cur.status != RUNNING && cur.status != IDLE && cur.status != HELD))
      {
       job_registry_unlink_proxy(jra, &cur);
       continue;
