--- src/scripts/blah_common_submit_functions.sh	2019-03-07 15:28:24.287322532 -0600
+++ src/scripts/blah_common_submit_functions.sh.patched	2019-03-08 11:17:34.206039743 -0600
@@ -632,9 +632,12 @@
     echo "new_home=\${old_home}/$run_dir">>$bls_tmp_file
   fi
 
-  echo "mkdir \$new_home">>$bls_tmp_file
-  echo "trap 'cd \$old_home; rm -rf \$new_home; exit 255' 1 2 3 15 24" >> $bls_tmp_file
-  echo "trap 'cd \$old_home; rm -rf \$new_home' 0" >> $bls_tmp_file
+  echo 'mkdir "$new_home"'>>$bls_tmp_file
+  echo 'job_wait_cleanup () { wait "$job_pid"; cd "$old_home"; rm -rf "$new_home"; }' >> $bls_tmp_file
+  echo 'on_signal () { kill -$1 "$job_pid"; job_wait_cleanup; exit 255; }' >> $bls_tmp_file
+  echo 'trap_sigs () { for sig; do trap "on_signal $sig" $sig; done; }' >> $bls_tmp_file
+  echo 'trap_sigs HUP INT QUIT TERM XCPU' >> $bls_tmp_file
+  echo 'trap job_wait_cleanup EXIT' >> $bls_tmp_file
 
   echo "# Copy into new home any shared input sandbox file" >> $bls_tmp_file
   bls_fl_subst_and_dump inputcopy "cp \"@@F_LOCAL\" \"\$new_home/@@F_REMOTE\" &> /dev/null" $bls_tmp_file
