diff -u -r org.glite.ce.blahp/src/condor_create_log.c org.glite.ce.blahp-patched/src/condor_create_log.c
--- org.glite.ce.blahp/src/condor_create_log.c	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp-patched/src/condor_create_log.c	2012-11-12 13:56:34.051269784 -0600
@@ -78,6 +78,11 @@
                 acc_loc = DEFAULT_GLITE_LOCATION;
         }
 
+/* In the Condor build of the blahp, we can find all the libraries we need
+ * via the RUNPATH. Setting LD_LIBRARY_PATH can muck up the command line
+ * tools for the local batch system.
+ */
+#if 0
         acc_needed_libs = make_message("%s/lib:%s/externals/lib:%s/lib:/opt/lcg/lib", acc_loc, acc_loc,
 			                getenv("GLOBUS_LOCATION") ? getenv("GLOBUS_LOCATION") : "/opt/globus");
         acc_old_ld_lib=getenv("LD_LIBRARY_PATH");
@@ -88,6 +93,7 @@
                 setenv("LD_LIBRARY_PATH",acc_new_ld_lib,1);
         }else
                  setenv("LD_LIBRARY_PATH",acc_needed_libs,1);
+#endif
 
         blah_config_handle = config_read(NULL);
         if (blah_config_handle == NULL) 
diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp-patched/src/server.c
--- org.glite.ce.blahp/src/server.c	2011-09-15 07:39:39.000000000 -0500
+++ org.glite.ce.blahp-patched/src/server.c	2012-11-12 13:52:33.750893748 -0600
@@ -342,6 +342,11 @@
 		tmp_dir  = DEFAULT_TEMP_DIR;
 	}
 
+/* In the Condor build of the blahp, we can find all the libraries we need
+ * via the RUNPATH. Setting LD_LIBRARY_PATH can muck up the command line
+ * tools for the local batch system.
+ */
+#if 0
 	needed_libs = make_message("%s/lib:%s/externals/lib:%s/lib:/opt/lcg/lib", result, result, getenv("GLOBUS_LOCATION") ? getenv("GLOBUS_LOCATION") : "/opt/globus");
 	old_ld_lib=getenv("LD_LIBRARY_PATH");
 	if(old_ld_lib)
@@ -357,6 +362,7 @@
 	}
 	else
 	 	 setenv("LD_LIBRARY_PATH",needed_libs,1);
+#endif
 	
 	blah_script_location = strdup(blah_config_handle->bin_path);
 	blah_version = make_message(RCSID_VERSION, VERSION, "poly,new_esc_format");
