diff -u -r org.glite.ce.blahp/src/scripts/lsf_submit.sh org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh
--- org.glite.ce.blahp/src/scripts/lsf_submit.sh	2011-03-04 06:56:04.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh	2013-03-15 14:02:38.000000000 -0500
@@ -178,6 +178,9 @@
 
 jobID=`echo "$bsub_out" | awk -F" " '{ print $2 }' | sed "s/>//" |sed "s/<//"`
 
+# Remove an empty line in the string
+jobID=`echo $jobID`
+
 if [ "x$jobID" == "x" ] ; then
         rm -f $bls_tmp_file
         exit 1
