diff -u -r org.glite.ce.blahp/src/scripts/lsf_submit.sh org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh
--- org.glite.ce.blahp/src/scripts/lsf_submit.sh	2011-03-04 06:56:04.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh	2015-03-13 16:09:37.000000000 -0500
@@ -51,6 +51,9 @@
 
 lsf_base_path=`cat $conffile|grep LSB_SHAREDIR| awk -F"=" '{ print $2 }'`
 
+lsf_confdir=`cat $conffile|grep LSF_CONFDIR| awk -F"=" '{ print $2 }'`
+[ -f ${lsf_confdir}/profile.lsf ] && . ${lsf_confdir}/profile.lsf
+
 lsf_clustername=`${lsf_binpath}/lsid | grep 'My cluster name is'|awk -F" " '{ print $5 }'`
 logpath=$lsf_base_path/$lsf_clustername/logdir
 
