--- org.glite.ce.blahp/src/scripts/lsf_status.sh	2011-03-04 06:56:04.000000000 -0600
+++ org.glite.ce.blahp/src/scripts/lsf_status.sh.patched	2014-09-16 14:36:49.990340256 -0500
@@ -225,11 +225,11 @@
 	if [ "$cliretcode" == "1" -o "x$lsf_BLParser" != "xyes" ] ; then
 		result=""
 		usedBLParser="no"
-		datefile=blahdate_$RANDOM$RANDOM$RANDOM
+		datefile=/tmp/blahdate_$RANDOM$RANDOM$RANDOM
 		touch $datefile;chmod 600 $datefile
 
 		if [ $? -ne 0 ]; then
-   			echo 'Error creating temporary file'
+   			echo '1ERROR: Could not create temporary file'
    			datefile=""
 			echo "1ERROR: Job not found"
 			break
