diff -u -r org.glite.ce.blahp/config/Makefile.am org.glite.ce.blahp.patched/config/Makefile.am
--- org.glite.ce.blahp/config/Makefile.am	2011-02-07 09:49:47.000000000 -0600
+++ org.glite.ce.blahp.patched/config/Makefile.am	2011-12-13 15:42:51.000000000 -0600
@@ -15,13 +15,13 @@
 #
 
 wletcwmdir = $(sysconfdir)
-wletcwm_DATA = blah.config.template blparser.conf.template
+wletcwm_DATA = batch_gahp.config.template blparser.conf.template
 
-#install-data-hook:
-#        /bin/cp $(DESTDIR)$(sysconfdir)/blah.config.template $(DESTDIR)$(sysconfdir)/blah.config
+install-data-hook:
+	/bin/cp $(DESTDIR)$(sysconfdir)/batch_gahp.config.template $(DESTDIR)$(sysconfdir)/batch_gahp.config
 
 
-EXTRA_DIST = blah.config.template \
+EXTRA_DIST = batch_gahp.config.template \
              blparser.conf.template \
              glite-ce-blparser \
              glite-ce-blahparser \
diff -u -r org.glite.ce.blahp/config/glite-ce-blahparser org.glite.ce.blahp.patched/config/glite-ce-blahparser
--- org.glite.ce.blahp/config/glite-ce-blahparser	2011-03-11 09:36:15.000000000 -0600
+++ org.glite.ce.blahp.patched/config/glite-ce-blahparser	2011-12-13 15:44:49.000000000 -0600
@@ -34,15 +34,15 @@
 fi
 
 if [ ! -r "$BLAHPD_CONFIG_LOCATION" ]; then
-  if [ -r "/etc/blah.config" ]; then
-    export BLAHPD_CONFIG_LOCATION="/etc/blah.config"
-  elif [ -r "${BLAHPD_LOCATION:-/opt/glite}/etc/blah.config" ]; then
-    export BLAHPD_CONFIG_LOCATION="${BLAHPD_LOCATION:-/opt/glite}/etc/blah.config"
+  if [ -r "/etc/batch_gahp.config" ]; then
+    export BLAHPD_CONFIG_LOCATION="/etc/batch_gahp.config"
+  elif [ -r "${BLAHPD_LOCATION:-/opt/glite}/etc/batch_gahp.config" ]; then
+    export BLAHPD_CONFIG_LOCATION="${BLAHPD_LOCATION:-/opt/glite}/etc/batch_gahp.config"
   fi
 fi
 
 if [ ! -r "$BLAHPD_CONFIG_LOCATION" ]; then
-  echo "$0: Cannot read BLAH config file (\$BLAHPD_CONFIG_LOCATION, or /etc/blah.config, or /opt/glite/etc/blah.config)"
+  echo "$0: Cannot read BLAH config file (\$BLAHPD_CONFIG_LOCATION, or /etc/batch_gahp.config, or /opt/glite/etc/batch_gahp.config)"
   exit 1
 fi
 
diff -u -r org.glite.ce.blahp/src/Makefile.am org.glite.ce.blahp.patched/src/Makefile.am
--- org.glite.ce.blahp/src/Makefile.am	2010-11-05 08:28:44.000000000 -0500
+++ org.glite.ce.blahp.patched/src/Makefile.am	2011-12-13 15:46:34.000000000 -0600
@@ -41,18 +41,18 @@
 GLOBUS_EXECS =
 endif
 
-bin_PROGRAMS = blahpd blahpd_daemon BLClient BLParserLSF BLParserPBS test_condor_logger blah_job_registry_add blah_job_registry_lkup blah_job_registry_scan_by_subject BUpdaterCondor BNotifier BUpdaterLSF BUpdaterPBS BUpdaterSGE $(GLOBUS_EXECS)  blparser_master blah_check_config blah_job_registry_dump
+bin_PROGRAMS = batch_gahp batch_gahp_daemon BLClient BLParserLSF BLParserPBS test_condor_logger blah_job_registry_add blah_job_registry_lkup blah_job_registry_scan_by_subject BUpdaterCondor BNotifier BUpdaterLSF BUpdaterPBS BUpdaterSGE $(GLOBUS_EXECS)  blparser_master blah_check_config blah_job_registry_dump
 noinst_PROGRAMS = test_job_registry_create test_job_registry_purge test_job_registry_update test_job_registry_access
 
 common_sources = console.c job_status.c resbuffer.c server.c commands.c classad_binary_op_unwind.C classad_c_helper.C proxy_hashcontainer.c config.c job_registry.c blah_utils.c env_helper.c mapped_exec.c md5.c 
 
-blahpd_SOURCES = main.c $(common_sources)
+batch_gahp_SOURCES = main.c $(common_sources)
 
-blahpd_daemon_SOURCES = main_daemon.c $(common_sources)
+batch_gahp_daemon_SOURCES = main_daemon.c $(common_sources)
 
-blahpd_LDADD = $(CLASSAD_LIBS)
+batch_gahp_LDADD = $(CLASSAD_LIBS)
 
-blahpd_daemon_LDADD = $(blahpd_LDADD)
+batch_gahp_daemon_LDADD = $(batch_gahp_LDADD)
 
 BPRclient_SOURCES = BPRclient.c BPRcomm.c tokens.c
 
diff -u -r org.glite.ce.blahp/src/config.h org.glite.ce.blahp.patched/src/config.h
--- org.glite.ce.blahp/src/config.h	2010-03-17 08:19:34.000000000 -0500
+++ org.glite.ce.blahp.patched/src/config.h	2011-12-13 15:47:03.000000000 -0600
@@ -56,7 +56,7 @@
 int config_test_boolean(const config_entry *entry);
 void config_free(config_handle *handle);
 
-#define CONFIG_FILE_BASE "blah.config"
+#define CONFIG_FILE_BASE "batch_gahp.config"
 
 #define CONFIG_SKIP_WHITESPACE_FWD(c) while ((*(c) == ' ')  || (*(c) == '\t') || \
                                   (*(c) == '\n') || (*(c) == '\r') ) (c)++;
diff -u -r org.glite.ce.blahp/src/scripts/blah_load_config.sh org.glite.ce.blahp.patched/src/scripts/blah_load_config.sh
--- org.glite.ce.blahp/src/scripts/blah_load_config.sh	2011-06-22 07:03:42.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/blah_load_config.sh	2011-12-13 15:48:58.000000000 -0600
@@ -36,10 +36,10 @@
 
 if [ -r "$BLAHPD_CONFIG_LOCATION" ]; then
   . $BLAHPD_CONFIG_LOCATION
-elif [ -r "${GLITE_LOCATION:-/opt/glite}/etc/blah.config" ]; then
-  . ${GLITE_LOCATION:-/opt/glite}/etc/blah.config
-elif [ -r "${BLAHPD_LOCATION}/etc/blah.config" ]; then
-#   This will default to /etc/blah.config if BLAHPD_LOCATION is unset.
-  . ${BLAHPD_LOCATION}/etc/blah.config
+elif [ -r "${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config" ]; then
+  . ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config
+elif [ -r "${BLAHPD_LOCATION}/etc/batch_gahp.config" ]; then
+#   This will default to /etc/batch_gahp.config if BLAHPD_LOCATION is unset.
+  . ${BLAHPD_LOCATION}/etc/batch_gahp.config
 fi
 
diff -u -r org.glite.ce.blahp/src/scripts/sge_cancel.sh org.glite.ce.blahp.patched/src/scripts/sge_cancel.sh
--- org.glite.ce.blahp/src/scripts/sge_cancel.sh	2011-06-09 06:21:01.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/sge_cancel.sh	2011-12-13 15:50:02.000000000 -0600
@@ -20,7 +20,7 @@
 #
 
 
-[ -f ${GLITE_LOCATION:-/opt/glite}/etc/blah.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/blah.config
+[ -f ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config
 
 if [ -z "$sge_rootpath" ]; then sge_rootpath="/usr/local/sge/pro"; fi
 if [ -r "$sge_rootpath/${sge_cellname:-default}/common/settings.sh" ]
diff -u -r org.glite.ce.blahp/src/scripts/sge_hold.sh org.glite.ce.blahp.patched/src/scripts/sge_hold.sh
--- org.glite.ce.blahp/src/scripts/sge_hold.sh	2011-06-09 06:21:01.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/sge_hold.sh	2011-12-13 15:50:23.000000000 -0600
@@ -20,7 +20,7 @@
 #
 
 
-[ -f ${GLITE_LOCATION:-/opt/glite}/etc/blah.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/blah.config
+[ -f ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config
 
 if [ -z "$sge_rootpath" ]; then sge_rootpath="/usr/local/sge/pro"; fi
 if [ -r "$sge_rootpath/${sge_cellname:-default}/common/settings.sh" ]
diff -u -r org.glite.ce.blahp/src/scripts/sge_resume.sh org.glite.ce.blahp.patched/src/scripts/sge_resume.sh
--- org.glite.ce.blahp/src/scripts/sge_resume.sh	2011-06-09 06:21:01.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/sge_resume.sh	2011-12-13 15:50:50.000000000 -0600
@@ -20,7 +20,7 @@
 #
 
 
-[ -f ${GLITE_LOCATION:-/opt/glite}/etc/blah.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/blah.config
+[ -f ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config
 
 if [ -z "$sge_rootpath" ]; then sge_rootpath="/usr/local/sge/pro"; fi
 if [ -r "$sge_rootpath/${sge_cellname:-default}/common/settings.sh" ]
diff -u -r org.glite.ce.blahp/src/scripts/sge_status.sh org.glite.ce.blahp.patched/src/scripts/sge_status.sh
--- org.glite.ce.blahp/src/scripts/sge_status.sh	2011-06-09 06:21:02.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/sge_status.sh	2011-12-13 15:51:13.000000000 -0600
@@ -20,7 +20,7 @@
 #
 
 
-[ -f ${GLITE_LOCATION:-/opt/glite}/etc/blah.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/blah.config
+[ -f ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config ] && . ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config
 
 usage_string="Usage: $0 [-w] [-n]"
 
diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp.patched/src/server.c
--- org.glite.ce.blahp/src/server.c	2011-09-15 07:39:39.000000000 -0500
+++ org.glite.ce.blahp.patched/src/server.c	2011-12-13 15:52:03.000000000 -0600
@@ -315,7 +315,7 @@
 	blah_config_handle = config_read(NULL);
 	if (blah_config_handle == NULL)
 	{
-		fprintf(stderr, "Cannot access blah.config file in default locations ($BLAHPD_CONFIG_LOCATION, or $GLITE_LOCATION/etc or $BLAHPD_LOCATION/etc): ");
+		fprintf(stderr, "Cannot access batch_gahp.config file in default locations ($BLAHPD_CONFIG_LOCATION, or $GLITE_LOCATION/etc or $BLAHPD_LOCATION/etc): ");
 		perror("");
 		exit(MALLOC_ERROR);
 	}
