--- org.glite.ce.blahp/src/scripts/pbs_status.sh	2016-04-19 13:17:52.631313130 -0500
+++ org.glite.ce.blahp-patched/src/scripts/pbs_status.sh	2016-06-13 11:08:38.398204947 -0500
@@ -33,6 +33,10 @@
 
 . `dirname $0`/blah_load_config.sh
 
+if [ -x ${blah_bin_directory}/pbs_status.py ] ; then
+    exec ${blah_bin_directory}/pbs_status.py "$@"
+fi
+
 if [ "x$job_registry" != "x" ] ; then
    ${blah_bin_directory}/blah_job_registry_lkup $@
    exit 0
--- org.glite.ce.blahp/src/scripts/Makefile.am     2011-09-09 04:58:16.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/Makefile.am     2016-06-13 11:14:50.282538970 -0500
@@ -32,7 +32,7 @@
 
 
 bin_SCRIPTS = blah_load_config.sh blah_common_submit_functions.sh \
-  pbs_cancel.sh pbs_status.sh pbs_submit.sh pbs_hold.sh pbs_resume.sh \
+  pbs_cancel.sh pbs_status.sh pbs_submit.sh pbs_hold.sh pbs_resume.sh pbs_status.py \
   lsf_cancel.sh lsf_status.sh lsf_submit.sh lsf_hold.sh lsf_resume.sh \
   condor_cancel.sh condor_status.sh condor_submit.sh condor_hold.sh condor_resume.sh \
   sge_cancel.sh sge_helper sge_resume.sh sge_submit.sh sge_filestaging \
