diff -u -r org.glite.ce.blahp/src/scripts/pbs_submit.sh org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh
--- org.glite.ce.blahp/src/scripts/pbs_submit.sh	2015-05-11 16:48:59.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh	2015-05-11 16:53:16.000000000 -0500
@@ -207,6 +207,12 @@
 
 # The job id is actually the first numbers in the string (slurm support)
 jobID=`echo $jobID | awk 'match($0,/[0-9]+/){print substr($0, RSTART, RLENGTH)}'`
+if [ "X$jobID" == "X" ]; then
+	rm -f $bls_tmp_file
+	echo "Error: job id missing" >&2
+	echo Error # for the sake of waiting fgets in blahpd
+	exit 1
+fi
 
 if [ "x$pbs_nologaccess" != "xyes" -a "x$pbs_nochecksubmission" != "xyes" ]; then
 
