diff -u -r org.glite.ce.blahp/src/scripts/pbs_status.sh org.glite.ce.blahp-patched/src/scripts/pbs_status.sh
--- org.glite.ce.blahp/src/scripts/pbs_status.sh	2011-03-04 06:56:04.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/pbs_status.sh	2012-07-23 15:48:21.000000000 -0500
@@ -196,6 +196,9 @@
 
 END {
         if (current_js ~ "Q")  {jobstatus = 1}
+        if (current_js ~ "W")  {jobstatus = 1}
+        if (current_js ~ "S")  {jobstatus = 1}
+        if (current_js ~ "T")  {jobstatus = 1}
         if (current_js ~ "R")  {jobstatus = 2}
         if (current_js ~ "E")  {jobstatus = 2}
         if (current_js ~ "C")  {jobstatus = 4}
