diff -u -r org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh org.glite.ce.blahp.patched/src/scripts/blah_common_submit_functions.sh
--- org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh	2013-04-01 17:39:56.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/blah_common_submit_functions.sh	2013-04-02 08:39:40.000000000 -0500
@@ -272,7 +272,7 @@
   ###############################################################
   # Parse parameters
   ###############################################################
-  while getopts "a:i:o:e:c:s:v:V:dw:q:n:N:z:h:S:r:p:l:x:u:j:T:I:O:R:C:D:" arg 
+  while getopts "a:i:o:e:c:s:v:V:dw:q:n:N:z:h:S:r:p:l:x:u:j:T:I:O:R:C:D:m:" arg 
   do
       case "$arg" in
       a) bls_opt_xtra_args="$OPTARG" ;;
@@ -303,6 +303,7 @@
       R) bls_opt_outputflstringremap="$OPTARG" ;;
       C) bls_opt_req_file="$OPTARG";;
       D) bls_opt_run_dir="$OPTARG";;
+      m) bls_opt_req_mem="$OPTARG";;
       -) break ;;
       ?) echo $usage_string
          exit 1 ;;
diff -u -r org.glite.ce.blahp/src/scripts/condor_submit.sh org.glite.ce.blahp.patched/src/scripts/condor_submit.sh
--- org.glite.ce.blahp/src/scripts/condor_submit.sh	2013-04-01 17:39:56.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/condor_submit.sh	2013-04-02 08:38:06.000000000 -0500
@@ -53,7 +53,7 @@
 # Name of local requirements file: currently unused
 req_file=""
 
-while getopts "a:i:o:de:j:n:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:" arg 
+while getopts "a:i:o:de:j:n:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:m:" arg 
 do
     case "$arg" in
     a) xtra_args="$OPTARG" ;;
@@ -78,6 +78,7 @@
     R) remaps="$OPTARG" ;;
     C) req_file="$OPTARG" ;;
     D) run_dir="$OPTARG" ;;
+    m) req_mem="$OPTARG" ;;
     -) break ;;
     ?) echo $usage_string
        exit 1 ;;
@@ -240,6 +241,11 @@
   echo -e $xtra_args >> $submit_file
 fi
 
+if [ "x$req_mem" != "x" ]
+then
+  echo "request_memory = $req_mem" >> $submit_file
+fi
+
 cat >> $submit_file << EOF
 # We insist on new style quoting in Condor
 arguments = $arguments
diff -u -r org.glite.ce.blahp/src/scripts/pbs_submit.sh org.glite.ce.blahp.patched/src/scripts/pbs_submit.sh
--- org.glite.ce.blahp/src/scripts/pbs_submit.sh	2013-04-01 17:39:33.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/pbs_submit.sh	2013-04-02 08:40:55.000000000 -0500
@@ -111,6 +111,14 @@
 #local batch system-specific file output must be added to the submit file
 bls_local_submit_attributes_file=${blah_bin_directory}/pbs_local_submit_attributes.sh
 
+if [ "x$bls_opt_req_mem" != "x" ]
+then
+  # Different schedulers require different memory checks
+  echo "#PBS -l mem=${bls_opt_req_mem}mb" >> $bls_tmp_file
+  echo "#PBS -l pmem=${bls_opt_req_mem}mb" >> $bls_tmp_file
+  echo "#PBS -l pvmem=${bls_opt_req_mem}mb" >> $bls_tmp_file
+fi
+
 bls_set_up_local_and_extra_args
 
 # Write PBS directives according to command line options
diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp.patched/src/server.c
--- org.glite.ce.blahp/src/server.c	2013-04-01 17:39:56.000000000 -0500
+++ org.glite.ce.blahp.patched/src/server.c	2013-04-02 08:32:30.000000000 -0500
@@ -1201,7 +1201,8 @@
 	    (set_cmd_bool_option  (&command, cad, "StageCmd",   "-s", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "ClientJobId","-j", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "JobDirectory","-D", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
-	    (set_cmd_string_option(&command, cad, "BatchExtraSubmitArgs", "-a", SINGLE_QUOTE) == C_CLASSAD_OUT_OF_MEMORY))
+	    (set_cmd_string_option(&command, cad, "BatchExtraSubmitArgs", "-a", SINGLE_QUOTE) == C_CLASSAD_OUT_OF_MEMORY) ||
+	    (set_cmd_int_option(&command, cad, "RequestMemory", "-m", INT_NOQUOTE) == C_CLASSAD_OUT_OF_MEMORY))
 //	    (set_cmd_string_option(&command, cad, "Args",      	"--", SINGLE_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY))
 	{
 		/* PUSH A FAILURE */
