diff -u -r org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh org.glite.ce.blahp-patched/src/scripts/blah_common_submit_functions.sh
--- org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh	2011-06-22 07:03:42.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/blah_common_submit_functions.sh	2012-11-15 16:23:25.000000000 -0600
@@ -272,7 +272,7 @@
   ###############################################################
   # Parse parameters
   ###############################################################
-  while getopts "a:i:o:e:c:s:v:V:dw:q:n:N:z:h:S:r:p:l:x:u:j:T:I:O:R:C:" arg 
+  while getopts "a:i:o:e:c:s:v:V:dw:q:n:N:z:h:S:r:p:l:x:u:j:T:I:O:R:C:D:" arg 
   do
       case "$arg" in
       a) bls_opt_xtra_args="$OPTARG" ;;
@@ -302,6 +302,7 @@
       O) bls_opt_outputflstring="$OPTARG" ;;
       R) bls_opt_outputflstringremap="$OPTARG" ;;
       C) bls_opt_req_file="$OPTARG";;
+      D) bls_opt_run_dir="$OPTARG";;
       -) break ;;
       ?) echo $usage_string
          exit 1 ;;
@@ -624,10 +625,15 @@
   
   echo "old_home=\`pwd\`">>$bls_tmp_file
   # Set the temporary home (including cd'ing into it)
+  if [ "x$bls_opt_run_dir" != "x" ] ; then
+    run_dir="$bls_opt_run_dir"
+  else
+    run_dir="home_$bls_tmp_name"
+  fi
   if [ -n "$blah_wn_temporary_home_dir" ] ; then
-    echo "new_home=${blah_wn_temporary_home_dir}/home_$bls_tmp_name">>$bls_tmp_file
+    echo "new_home=${blah_wn_temporary_home_dir}/$run_dir">>$bls_tmp_file
   else
-    echo "new_home=\${old_home}/home_$bls_tmp_name">>$bls_tmp_file
+    echo "new_home=\${old_home}/$run_dir">>$bls_tmp_file
   fi
 
   echo "mkdir \$new_home">>$bls_tmp_file
diff -u -r org.glite.ce.blahp/src/scripts/condor_submit.sh org.glite.ce.blahp-patched/src/scripts/condor_submit.sh
--- org.glite.ce.blahp/src/scripts/condor_submit.sh	2011-03-04 06:56:04.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/condor_submit.sh	2012-11-16 14:19:34.000000000 -0600
@@ -53,7 +53,7 @@
 # Name of local requirements file: currently unused
 req_file=""
 
-while getopts "a:i:o:de:j:n:v:V:c:w:x:u:q:r:s:T:I:O:R:C:" arg 
+while getopts "a:i:o:de:j:n:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:" arg 
 do
     case "$arg" in
     a) xtra_args="$OPTARG" ;;
@@ -77,6 +77,7 @@
     O) outputflstring="$OPTARG" ;;
     R) remaps="$OPTARG" ;;
     C) req_file="$OPTARG" ;;
+    D) run_dir="$OPTARG" ;;
     -) break ;;
     ?) echo $usage_string
        exit 1 ;;
diff -u -r org.glite.ce.blahp/src/server.c org.glite.ce.blahp-patched/src/server.c
--- org.glite.ce.blahp/src/server.c	2011-09-15 07:39:39.000000000 -0500
+++ org.glite.ce.blahp-patched/src/server.c	2012-11-15 16:23:48.000000000 -0600
@@ -1191,6 +1191,7 @@
 	    (set_cmd_int_option   (&command, cad, "HostSMPSize", "-N", INT_NOQUOTE)  == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_bool_option  (&command, cad, "StageCmd",   "-s", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "ClientJobId","-j", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
+	    (set_cmd_string_option(&command, cad, "JobDirectory","-D", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "BatchExtraSubmitArgs", "-a", SINGLE_QUOTE) == C_CLASSAD_OUT_OF_MEMORY))
 //	    (set_cmd_string_option(&command, cad, "Args",      	"--", SINGLE_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY))
 	{
