--- org.glite.ce.blahp/src/server.c	2016-03-25 19:41:52.396108343 +0000
+++ org.glite.ce.blahp.patched/src/server.c	2016-04-26 18:55:43.839629858 +0000
@@ -1201,6 +1201,8 @@
 	    (set_cmd_bool_option  (&command, cad, "StageCmd",   "-s", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "ClientJobId","-j", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "JobDirectory","-D", NO_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY) ||
+	    (set_cmd_string_option(&command, cad, "BatchProject", "-A", NO_QUOTE) == C_CLASSAD_OUT_OF_MEMORY) ||
+	    (set_cmd_int_option(&command, cad, "BatchRuntime", "-t", INT_NOQUOTE) == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_string_option(&command, cad, "BatchExtraSubmitArgs", "-a", SINGLE_QUOTE) == C_CLASSAD_OUT_OF_MEMORY) ||
 	    (set_cmd_int_option(&command, cad, "RequestMemory", "-m", INT_NOQUOTE) == C_CLASSAD_OUT_OF_MEMORY))
 //	    (set_cmd_string_option(&command, cad, "Args",      	"--", SINGLE_QUOTE)      == C_CLASSAD_OUT_OF_MEMORY))
--- org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh	2016-03-25 19:41:52.357422214 +0000
+++ org.glite.ce.blahp.patched/src/scripts/blah_common_submit_functions.sh	2016-04-26 18:58:09.549426129 +0000
@@ -272,7 +272,7 @@
   ###############################################################
   # Parse parameters
   ###############################################################
-  while getopts "a:i:o:e:c:s:v:V:dw:q:n:N:z:h:S:r:p:l:x:u:j:T:I:O:R:C:D:m:" arg 
+  while getopts "a:i:o:e:c:s:v:V:dw:q:n:N:z:h:S:r:p:l:x:u:j:T:I:O:R:C:D:m:A:t:" arg 
   do
       case "$arg" in
       a) bls_opt_xtra_args="$OPTARG" ;;
@@ -304,6 +304,8 @@
       C) bls_opt_req_file="$OPTARG";;
       D) bls_opt_run_dir="$OPTARG";;
       m) bls_opt_req_mem="$OPTARG";;
+      A) bls_opt_project="$OPTARG";;
+      t) bls_opt_runtime="$OPTARG";;
       -) break ;;
       ?) echo $usage_string
          exit 1 ;;
--- org.glite.ce.blahp/src/scripts/lsf_submit.sh	2017-10-13 10:22:51.263416732 -0500
+++ org.glite.ce.blahp.patched/src/scripts/lsf_submit.sh	2017-10-16 15:19:14.722272459 -0500
@@ -142,6 +142,14 @@
 fi
 # --- End of MPI directives
 
+if [ "x$bls_opt_project" != "x" ] ; then
+  echo "#BSUB -P $bls_opt_project" >> $bls_tmp_file
+fi
+
+if [ "x$bls_opt_runtime" != "x" ] ; then
+  echo "#BSUB -W $((bls_opt_runtime / 60))" >> $bls_tmp_file
+fi
+
 #local batch system-specific file output must be added to the submit file
 bls_local_submit_attributes_file=${blah_bin_directory}/lsf_local_submit_attributes.sh
 
--- org.glite.ce.blahp/src/scripts/pbs_submit.sh	2017-10-13 10:22:51.266416762 -0500
+++ org.glite.ce.blahp.patched/src/scripts/pbs_submit.sh	2017-10-16 15:19:38.098498549 -0500
@@ -108,6 +108,14 @@
   echo "#PBS -e $pbs_std_storage" >> $bls_tmp_file
 fi
 
+if [ "x$bls_opt_project" != "x" ] ; then
+  echo "#PBS -A $bls_opt_project" >> $bls_tmp_file
+fi
+
+if [ "x$bls_opt_runtime" != "x" ] ; then
+  echo "#PBS -l walltime=$bls_opt_runtime" >> $bls_tmp_file
+fi
+
 #local batch system-specific file output must be added to the submit file
 bls_local_submit_attributes_file=${blah_bin_directory}/pbs_local_submit_attributes.sh
 
--- org.glite.ce.blahp/src/scripts/sge_submit.sh	2011-09-02 04:05:50.000000000 -0500
+++ org.glite.ce.blahp.patched/src/scripts/sge_submit.sh	2017-10-16 15:19:58.482695350 -0500
@@ -63,6 +63,14 @@
 #\$ -S /bin/bash
 end_of_preamble
 
+if [ "x$bls_opt_project" != "x" ] ; then
+  echo "#\$ -A $bls_opt_project" >> $bls_tmp_file
+fi
+
+if [ "x$bls_opt_runtime" != "x" ] ; then
+  echo "#\$ -l h_rt=$bls_opt_runtime" >> $bls_tmp_file
+fi
+
 #local batch system-specific file output must be added to the submit file
 local_submit_attributes_file=${GLITE_LOCATION:-/opt/glite}/bin/sge_local_submit_attributes.sh
 if [ -r $local_submit_attributes_file ] ; then
--- org.glite.ce.blahp/src/scripts/condor_submit.sh	2017-10-13 10:22:51.290416996 -0500
+++ org.glite.ce.blahp.patched/src/scripts/condor_submit.sh	2017-10-20 10:53:34.389263817 -0500
@@ -53,7 +53,7 @@
 # Name of local requirements file: currently unused
 req_file=""
 
-while getopts "a:i:o:de:j:n:N:z:h:S:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:m:" arg 
+while getopts "a:i:o:de:j:n:N:z:h:S:v:V:c:w:x:u:q:r:s:T:I:O:R:C:D:m:A:t:" arg 
 do
     case "$arg" in
     a) xtra_args="$OPTARG" ;;
@@ -83,6 +83,8 @@
     C) req_file="$OPTARG" ;;
     D) run_dir="$OPTARG" ;;
     m) req_mem="$OPTARG" ;;
+    A) project="$OPTARG" ;;
+    t) runtime="$OPTARG" ;;
     -) break ;;
     ?) echo $usage_string
        exit 1 ;;
@@ -253,6 +255,11 @@
   echo "request_memory = $req_mem" >> $submit_file
 fi
 
+if [ "x$runtime" != "x" ]
+then
+  echo "periodic_remove = JobStatus == 2 && time() - JobCurrentStartExecutingDate > $runtime" >> $submit_file
+fi
+
 cat >> $submit_file << EOF
 request_cpus = $mpinodes
 # We insist on new style quoting in Condor
