diff -u -r org.glite.ce.blahp/config/blah.config.template org.glite.ce.blahp.patched/config/blah.config.template
--- org.glite.ce.blahp/config/blah.config.template	2012-01-24 16:59:13.000000000 -0600
+++ org.glite.ce.blahp.patched/config/blah.config.template	2012-01-24 17:00:20.000000000 -0600
@@ -180,6 +180,7 @@
 #Colon-separated list of paths that are shared among batch system
 #head and worker nodes.
 #blah_shared_directories=/home:/users
+blah_shared_directories=/
 
 #By default the job temporary work directory is created as a subdirectory
 #of wherever the batch system is configured to land the job. 
