diff -u -r org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh org.glite.ce.blahp-patched/src/scripts/blah_common_submit_functions.sh
--- org.glite.ce.blahp/src/scripts/blah_common_submit_functions.sh	2011-06-22 07:03:42.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/blah_common_submit_functions.sh	2012-08-15 14:15:39.000000000 -0500
@@ -498,6 +498,7 @@
   
   # Setup stdin, stdout & stderr
   if [ ! -z "$bls_opt_stdin" ] ; then
+      if [ "${bls_opt_stdin:0:1}" != "/" ] ; then bls_opt_stdin=${bls_opt_workdir}/${bls_opt_stdin} ; fi
       if [ -f "$bls_opt_stdin" ] ; then
           bls_test_shared_dir "$bls_opt_stdin"
           if [ "x$bls_is_in_shared_dir" == "xyes" ] ; then
@@ -512,22 +513,22 @@
       fi
   fi
   if [ ! -z "$bls_opt_stdout" ] ; then
+      if [ "${bls_opt_stdout:0:1}" != "/" ] ; then bls_opt_stdout=${bls_opt_workdir}/${bls_opt_stdout} ; fi
       bls_test_shared_dir "$bls_opt_stdout"
       if [ "x$bls_is_in_shared_dir" == "xyes" ] ; then
           bls_arguments="$bls_arguments > \"$bls_opt_stdout\""
       else
-          if [ "${bls_opt_stdout:0:1}" != "/" ] ; then bls_opt_stdout=${bls_opt_workdir}/${bls_opt_stdout} ; fi
           bls_unique_stdout_name="${blah_wn_outputsandbox}out_${bls_tmp_name}_`basename $bls_opt_stdout`"
           bls_arguments="$bls_arguments > \"$bls_unique_stdout_name\""
           bls_fl_add_value outputsand "$bls_opt_stdout" "$bls_unique_stdout_name"
       fi
   fi
   if [ ! -z "$bls_opt_stderr" ] ; then
+      if [ "${bls_opt_stderr:0:1}" != "/" ] ; then bls_opt_stderr=${bls_opt_workdir}/${bls_opt_stderr} ; fi
       bls_test_shared_dir "$bls_opt_stderr"
       if [ "x$bls_is_in_shared_dir" == "xyes" ] ; then
           bls_arguments="$bls_arguments 2> \"$bls_opt_stderr\""
       else
-          if [ "${bls_opt_stderr:0:1}" != "/" ] ; then bls_opt_stderr=${bls_opt_workdir}/${bls_opt_stderr} ; fi
           if [ "$bls_opt_stderr" == "$bls_opt_stdout" ]; then
               bls_arguments="$bls_arguments 2>&1"
           else
@@ -547,6 +548,7 @@
       exec 4< "$bls_opt_inputflstring"
       while read xfile <&4 ; do
           if [ ! -z $xfile  ] ; then
+               if [ "${xfile:0:1}" != "/" ] ; then xfile=${bls_opt_workdir}/${xfile} ; fi
                bls_test_shared_dir "$xfile"
                if [ "x$bls_is_in_shared_dir" == "xyes" ] ; then
                    bls_fl_add_value inputcopy "$xfile" "./`basename ${xfile}`"
@@ -574,28 +576,19 @@
                   read xfileremap <&6
               fi
 
-              bls_test_shared_dir "$xfile"
+              if [ -z $xfileremap ] ; then
+                xfileremap="$xfile"
+              fi
+              if [ "${xfileremap:0:1}" != "/" ] ; then
+                xfileremap=${bls_opt_workdir}/${xfileremap}
+              fi
+              bls_test_shared_dir "$xfileremap"
               if [ "x$bls_is_in_shared_dir" != "xyes" ] ; then
-                  if [ "${xfile:0:1}" != "/" ] ; then
-                       xfile_base="`basename ${xfile}`"
-                       xfile_transfer="${blah_wn_outputsandbox}${xfile_base}.$uni_ext"
-                  else
-                       xfile_transfer="$xfile"
-                  fi
-                  if [ ! -z $xfileremap ] ; then
-                      if [ "${xfileremap:0:1}" != "/" ] ; then
-                          bls_fl_add_value outputsand "${bls_opt_workdir}/${xfileremap}" "$xfile_transfer" "$xfile"
-                      else
-                          bls_test_shared_dir "$xfileremap"
-                          if [ "x$bls_is_in_shared_dir" == "xyes" ] ; then
-                              bls_fl_add_value outputmove "${xfileremap}" "$xfile"
-                          else
-                              bls_fl_add_value outputsand "${xfileremap}" "$xfile_transfer" "$xfile"
-                          fi
-                      fi
-                  else
-                      bls_fl_add_value outputsand "${bls_opt_workdir}/${xfile}" "$xfile_transfer" "$xfile"
-                  fi
+                  xfile_base="`basename ${xfile}`"
+                  xfile_transfer="${blah_wn_outputsandbox}${xfile_base}.$uni_ext"
+                  bls_fl_add_value outputsand "$xfileremap" "$xfile_transfer" "$xfile"
+              else
+                  bls_fl_add_value outputmove "$xfileremap" "$xfile"
               fi
           fi
       done
