diff -u -r org.glite.ce.blahp/src/scripts/pbs_status.sh org.glite.ce.blahp-patched/src/scripts/pbs_status.sh
--- org.glite.ce.blahp/src/scripts/pbs_status.sh	2013-02-01 10:37:29.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/pbs_status.sh	2013-02-01 10:42:03.000000000 -0600
@@ -176,11 +176,14 @@
     current_job = ""
     current_wn = ""
     current_js = ""
+    exitcode = "-1"
 }
 
 /Job Id:/ {
     current_job = substr($0, index($0, ":") + 2)
-    current_job = substr(current_job, 1, index(current_job, ".")-1)
+    end = index(current_job, ".")
+    if ( end == 0 ) { end = length(current_job) + 1 }
+    current_job = substr(current_job, 1, end)
     print "[BatchJobId=\"" current_job "\";"
 }
 /exec_host =/ {
@@ -212,6 +215,9 @@
 	}
 	print "JobStatus=" jobstatus ";"
 	if (jobstatus == 4) {
+		if (exitcode == "-1") {
+			exitcode = "0"
+		}
 		print "ExitCode=" exitcode ";"
 	}
 	print "]"
diff -u -r org.glite.ce.blahp/src/scripts/pbs_submit.sh org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh
--- org.glite.ce.blahp/src/scripts/pbs_submit.sh	2013-02-01 10:38:17.000000000 -0600
+++ org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh	2013-02-01 10:42:20.000000000 -0600
@@ -197,6 +197,9 @@
 	exit 1
 fi
 
+# The job id is actually the first numbers in the string (slurm support)
+jobID=`echo $jobID | awk 'match($0,/[0-9]+/){print substr($0, RSTART, RLENGTH)}'`
+
 if [ "x$pbs_nologaccess" != "xyes" -a "x$pbs_nochecksubmission" != "xyes" ]; then
 
 # Don't trust qsub retcode, it could have crashed
