diff -u -r org.glite.ce.blahp/src/scripts/condor_submit.sh org.glite.ce.blahp-patched/src/scripts/condor_submit.sh
--- org.glite.ce.blahp/src/scripts/condor_submit.sh	2012-10-16 11:22:01.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/condor_submit.sh	2011-03-04 06:56:04.000000000 -0600
@@ -1,4 +1,4 @@
-#!/bin/bash -l
+#!/bin/bash
 #
 # 	File:     condor_submit.sh
 # 	Author:   Giuseppe Fiorentino (giuseppe.fiorentino@mi.infn.it)
diff -u -r org.glite.ce.blahp/src/scripts/lsf_submit.sh org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh
--- org.glite.ce.blahp/src/scripts/lsf_submit.sh	2012-10-16 11:22:08.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/lsf_submit.sh	2011-03-04 06:56:04.000000000 -0600
@@ -1,4 +1,4 @@
-#!/bin/bash -l
+#!/bin/bash
 #
 # File:     lsf_submit.sh
 # Author:   David Rebatto (david.rebatto@mi.infn.it)
diff -u -r org.glite.ce.blahp/src/scripts/pbs_submit.sh org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh
--- org.glite.ce.blahp/src/scripts/pbs_submit.sh	2012-10-16 11:22:14.000000000 -0500
+++ org.glite.ce.blahp-patched/src/scripts/pbs_submit.sh	2012-03-08 05:29:09.000000000 -0600
@@ -1,4 +1,4 @@
-#!/bin/bash -l
+#!/bin/bash
 #
 # File:     pbs_submit.sh
 # Author:   David Rebatto (david.rebatto@mi.infn.it)
