diff -ru a/voms-2.1.0-rc0/src/sslutils/proxycertinfo.c b/voms-2.1.0-rc0/src/sslutils/proxycertinfo.c
--- a/voms-2.1.0-rc0/src/sslutils/proxycertinfo.c	2017-01-25 12:48:50.000000000 -0600
+++ b/voms-2.1.0-rc0/src/sslutils/proxycertinfo.c	2017-10-24 14:37:13.208312687 -0500
@@ -346,7 +346,10 @@
 
   char const* pci_v3_sn =  "proxyCertInfo_V3";
   char const* pci_v3_ln =  "Proxy Certificate Information (V3)";
-  int const v3nid = OBJ_create(PROXYCERTINFO_OLD_OID, pci_v3_sn, pci_v3_ln);
+  int v3nid = OBJ_txt2nid(PROXYCERTINFO_OLD_OID);
+  if (v3nid == 0) {
+    v3nid = OBJ_create(PROXYCERTINFO_OLD_OID, pci_v3_sn, pci_v3_ln);
+  }
   assert(v3nid != 0 && "OBJ_create failed");
 
   if (X509V3_EXT_get_nid(v3nid) == NULL) {
Only in b/voms-2.1.0-rc0/src/sslutils: .proxycertinfo.c.un~
